// @fonoster/common/dist/voice/voice.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamEvent = exports.StreamContent = exports.ERROR = exports.END = exports.DATA = void 0;
const DATA = "data";
exports.DATA = DATA;
const END = "end";
exports.END = END;
const ERROR = "error";
exports.ERROR = ERROR;
var StreamEvent;
(function (StreamEvent) {
    StreamEvent["DATA"] = "data";
    StreamEvent["END"] = "end";
    StreamEvent["ERROR"] = "error";
})(StreamEvent || (exports.StreamEvent = StreamEvent = {}));
var StreamContent;
(function (StreamContent) {
    StreamContent["ANSWER_REQUEST"] = "answerRequest";
    StreamContent["ANSWER_RESPONSE"] = "answerResponse";
    StreamContent["HANGUP_REQUEST"] = "hangupRequest";
    StreamContent["HANGUP_RESPONSE"] = "hangupResponse";
    StreamContent["PLAY_REQUEST"] = "playRequest";
    StreamContent["PLAY_RESPONSE"] = "playResponse";
    StreamContent["PLAY_DTMF_REQUEST"] = "playDtmfRequest";
    StreamContent["PLAY_DTMF_RESPONSE"] = "playDtmfResponse";
    StreamContent["PLAYBACK_CONTROL_REQUEST"] = "playbackControlRequest";
    StreamContent["PLAYBACK_CONTROL_RESPONSE"] = "playbackControlResponse";
    StreamContent["MUTE_REQUEST"] = "muteRequest";
    StreamContent["MUTE_RESPONSE"] = "muteResponse";
    StreamContent["UNMUTE_REQUEST"] = "unmuteRequest";
    StreamContent["UNMUTE_RESPONSE"] = "unmuteResponse";
    StreamContent["GATHER_REQUEST"] = "gatherRequest";
    StreamContent["GATHER_RESPONSE"] = "gatherResponse";
    StreamContent["SAY_REQUEST"] = "sayRequest";
    StreamContent["SAY_RESPONSE"] = "sayResponse";
    StreamContent["RECORD_REQUEST"] = "recordRequest";
    StreamContent["RECORD_RESPONSE"] = "recordResponse";
    StreamContent["DIAL_REQUEST"] = "dialRequest";
    StreamContent["DIAL_RESPONSE"] = "dialResponse";
    StreamContent["START_STREAM_REQUEST"] = "startStreamRequest";
    StreamContent["START_STREAM_RESPONSE"] = "startStreamResponse";
    StreamContent["STOP_STREAM_REQUEST"] = "stopStreamRequest";
    StreamContent["STOP_STREAM_RESPONSE"] = "stopStreamResponse";
    StreamContent["STOP_SAY_REQUEST"] = "stopSayRequest";
    StreamContent["STOP_SAY_RESPONSE"] = "stopSayResponse";
    StreamContent["STREAM_PAYLOAD"] = "streamPayload";
    StreamContent["START_STREAM_GATHER_REQUEST"] = "startStreamGatherRequest";
    StreamContent["START_STREAM_GATHER_RESPONSE"] = "startStreamGatherResponse";
    StreamContent["STOP_STREAM_GATHER_REQUEST"] = "stopStreamGatherRequest";
    StreamContent["STOP_STREAM_GATHER_RESPONSE"] = "stopStreamGatherResponse";
    StreamContent["STREAM_GATHER_PAYLOAD"] = "streamGatherPayload";
})(StreamContent || (exports.StreamContent = StreamContent = {}));
