// @fonoster/common/dist/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WELCOME_DEMO_SPECIAL_LOCAL_ADDRESS = exports.AUTOPILOT_INTERNAL_ADDRESS = exports.AUTOPILOT_SPECIAL_LOCAL_ADDRESS = exports.ROUTR_DEFAULT_PEER_AOR = exports.APP_REF_HEADER = exports.INFLUXDB_CALLS_BUCKET = exports.CALL_DETAIL_RECORD_MEASUREMENT = exports.CALL_EXTENSION = exports.CALL_CONTEXT = exports.STASIS_APP_NAME = exports.GRPC_SERVING_STATUS = exports.GRPC_NOT_SERVING_STATUS = void 0;
exports.GRPC_NOT_SERVING_STATUS = "NOT_SERVING";
exports.GRPC_SERVING_STATUS = "SERVING";
exports.STASIS_APP_NAME = "mediacontroller";
exports.CALL_CONTEXT = "local-ctx-common";
exports.CALL_EXTENSION = "start";
exports.CALL_DETAIL_RECORD_MEASUREMENT = "cdr";
exports.INFLUXDB_CALLS_BUCKET = "calls";
exports.APP_REF_HEADER = "x-app-ref";
exports.ROUTR_DEFAULT_PEER_AOR = "sip:voice@default";
exports.AUTOPILOT_SPECIAL_LOCAL_ADDRESS = "autopilot.fonoster.local";
exports.AUTOPILOT_INTERNAL_ADDRESS = "autopilot:50061";
exports.WELCOME_DEMO_SPECIAL_LOCAL_ADDRESS = "welcome.demo.fonoster.local";
