// @fonoster/common/dist/messages.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VALID_UUID = exports.VALID_URL = exports.VALID_LANGUAGE_CODE = exports.VALID_DTMF = exports.VALID_DATE = exports.POSITIVE_INTEGER_MESSAGE = exports.MUST_BE_A_STRING = exports.MUST_BE_A_SINGLE_CHARACTER = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
exports.MUST_BE_A_SINGLE_CHARACTER = "Must be a single character";
exports.MUST_BE_A_STRING = "Must be a string";
exports.POSITIVE_INTEGER_MESSAGE = "Must be a positive number";
exports.VALID_DATE = "The date must be a valid ISO 8601";
exports.VALID_DTMF = "Must be a valid DTMF";
exports.VALID_LANGUAGE_CODE = "Must be a valid language code";
exports.VALID_URL = "Must be a valid URL";
exports.VALID_UUID = "Must be a valid UUID";
