// @fonoster/common/dist/assistants/tools/toolSchema.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.toolSchema = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const zod_1 = require("zod");
const Messages = __importStar(require("../../messages"));
const sendHttpRequest_1 = require("../../utils/sendHttpRequest");
const propertySchema_1 = require("./propertySchema");
const toolSchema = zod_1.z.object({
    name: zod_1.z.string(),
    description: zod_1.z.string(),
    parameters: zod_1.z.object({
        type: zod_1.z.enum(["object", "array"]),
        properties: zod_1.z.record(propertySchema_1.propertySchema),
        required: zod_1.z.array(zod_1.z.string()).optional()
    }),
    requestStartMessage: zod_1.z.string().optional(),
    operation: zod_1.z.object({
        method: zod_1.z
            .nativeEnum(sendHttpRequest_1.AllowedHttpMethod, {
            message: "Invalid method"
        })
            .default(sendHttpRequest_1.AllowedHttpMethod.GET),
        url: zod_1.z.string().url({ message: Messages.VALID_URL }),
        waitForResponse: zod_1.z.boolean().default(true),
        headers: zod_1.z
            .record(zod_1.z.string(), zod_1.z.string())
            .refine((headers) => !Object.keys(headers || {}).some((key) => key.toLowerCase() === "accept" ||
            key.toLowerCase() === "content-type"), {
            message: "Headers cannot include 'accept' or 'content-type' as they are set internally"
        })
            .optional()
    })
});
exports.toolSchema = toolSchema;
