// @fonoster/sdk/dist/node/client/AbstractClient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AbstractClient = void 0;
const identity_pb_1 = require("../generated/node/identity_pb");
const makeRpcRequest_1 = require("./makeRpcRequest");
class AbstractClient {
    accessKeyId;
    _accessToken;
    _refreshToken;
    _idToken;
    identityClient;
    constructor(config) {
        this.accessKeyId = config.accessKeyId;
        this.identityClient = config.identityClient;
        this._accessToken = "";
        this._refreshToken = "";
        this._idToken = "";
    }
    async login(username, password, twoFactorCode) {
        const { refreshToken, accessToken, idToken } = await (0, makeRpcRequest_1.makeRpcRequest)({
            method: this.identityClient.exchangeCredentials.bind(this.identityClient),
            requestPBObjectConstructor: identity_pb_1.ExchangeCredentialsRequest,
            metadata: {},
            request: {
                username,
                password,
                twoFactorCode
            }
        });
        this._refreshToken = refreshToken;
        this._accessToken = accessToken;
        this._idToken = idToken;
    }
    logout() {
        this._refreshToken = "";
        this._accessToken = "";
        this._idToken = "";
    }
    async loginWithRefreshToken(refreshToken) {
        const { accessToken, refreshToken: newRefreshToken, idToken } = await (0, makeRpcRequest_1.makeRpcRequest)({
            method: this.identityClient.exchangeRefreshToken.bind(this.identityClient),
            requestPBObjectConstructor: identity_pb_1.ExchangeRefreshTokenRequest,
            metadata: {},
            request: {
                refreshToken
            }
        });
        this._refreshToken = newRefreshToken;
        this._accessToken = accessToken;
        this._idToken = idToken;
    }
    async loginWithApiKey(accessKeyId, accessKeySecret) {
        const { refreshToken, accessToken } = await (0, makeRpcRequest_1.makeRpcRequest)({
            method: this.identityClient.exchangeApiKey.bind(this.identityClient),
            requestPBObjectConstructor: identity_pb_1.ExchangeApiKeyRequest,
            metadata: {},
            request: {
                accessKeyId,
                accessKeySecret
            }
        });
        this._refreshToken = refreshToken;
        this._accessToken = accessToken;
    }
    async loginWithOauth2Code(provider, code) {
        const { refreshToken, accessToken, idToken } = await (0, makeRpcRequest_1.makeRpcRequest)({
            method: this.identityClient.exchangeOauth2Code.bind(this.identityClient),
            requestPBObjectConstructor: identity_pb_1.ExchangeOauth2CodeRequest,
            metadata: {},
            request: {
                provider,
                code
            },
            enumMapping: [["provider", identity_pb_1.ExchangeOauth2CodeRequest.Oauth2Provider]]
        });
        this._refreshToken = refreshToken;
        this._accessToken = accessToken;
        this._idToken = idToken;
    }
    setAccessToken(accessToken) {
        this._accessToken = accessToken;
    }
    async sendVerificationCode(contactType, value) {
        return (0, makeRpcRequest_1.makeRpcRequest)({
            method: this.identityClient.sendVerificationCode.bind(this.identityClient),
            requestPBObjectConstructor: identity_pb_1.SendVerificationCodeRequest,
            metadata: {},
            request: {
                contactType,
                value
            },
            enumMapping: [["contactType", identity_pb_1.ContactType]]
        });
    }
    async verifyCode(request) {
        return (0, makeRpcRequest_1.makeRpcRequest)({
            method: this.identityClient.verifyCode.bind(this.identityClient),
            requestPBObjectConstructor: identity_pb_1.VerifyCodeRequest,
            metadata: {},
            request,
            enumMapping: [["contactType", identity_pb_1.ContactType]]
        });
    }
    async refreshToken() {
        return this.loginWithRefreshToken(this._refreshToken);
    }
    getAccessKeyId() {
        return this.accessKeyId;
    }
    setAccessKeyId(accessKeyId) {
        this.accessKeyId = accessKeyId;
    }
    getAccessToken() {
        return this._accessToken;
    }
    getRefreshToken() {
        return this._refreshToken;
    }
    getIdToken() {
        return this._idToken;
    }
}
exports.AbstractClient = AbstractClient;
