// @fonoster/sdk/dist/node/client/Client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Client = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const grpc_js_1 = require("@grpc/grpc-js");
const acls_grpc_pb_1 = require("../generated/node/acls_grpc_pb");
const agents_grpc_pb_1 = require("../generated/node/agents_grpc_pb");
const applications_grpc_pb_1 = require("../generated/node/applications_grpc_pb");
const calls_grpc_pb_1 = require("../generated/node/calls_grpc_pb");
const credentials_grpc_pb_1 = require("../generated/node/credentials_grpc_pb");
const domains_grpc_pb_1 = require("../generated/node/domains_grpc_pb");
const identity_grpc_pb_1 = require("../generated/node/identity_grpc_pb");
const numbers_grpc_pb_1 = require("../generated/node/numbers_grpc_pb");
const secrets_grpc_pb_1 = require("../generated/node/secrets_grpc_pb");
const trunks_grpc_pb_1 = require("../generated/node/trunks_grpc_pb");
const AbstractClient_1 = require("./AbstractClient");
const TokenRefresherNode_1 = require("./TokenRefresherNode");
const DEFAULT_ENDPOINT = "api.fonoster.com";
class Client extends AbstractClient_1.AbstractClient {
    endpoint;
    tokenRefresherInterceptor;
    channelCredentials;
    constructor(config = {
        endpoint: DEFAULT_ENDPOINT,
        accessKeyId: "",
        allowInsecure: false,
        withoutInterceptors: false
    }) {
        const channelCredentials = config.allowInsecure
            ? grpc_js_1.credentials.createInsecure()
            : grpc_js_1.credentials.createSsl();
        super({
            accessKeyId: config.accessKeyId,
            identityClient: new identity_grpc_pb_1.IdentityClient(config.endpoint ?? DEFAULT_ENDPOINT, channelCredentials)
        });
        this.channelCredentials = channelCredentials;
        this.endpoint = config?.endpoint ?? DEFAULT_ENDPOINT;
        this.tokenRefresherInterceptor = config.withoutInterceptors
            ? null
            : new TokenRefresherNode_1.TokenRefresherNode(this).createInterceptor().bind(this);
    }
    getMetadata() {
        const metadata = new grpc_js_1.Metadata();
        metadata.set("token", this.getAccessToken());
        metadata.set("accessKeyId", this.getAccessKeyId());
        return metadata;
    }
    getApplicationsClient() {
        return new applications_grpc_pb_1.ApplicationsClient(this.endpoint, this.channelCredentials, {
            interceptors: [this.tokenRefresherInterceptor]
        });
    }
    getCallsClient() {
        return new calls_grpc_pb_1.CallsClient(this.endpoint, this.channelCredentials, {
            interceptors: [this.tokenRefresherInterceptor]
        });
    }
    getIdentityClient() {
        return new identity_grpc_pb_1.IdentityClient(this.endpoint, this.channelCredentials);
    }
    getSecretsClient() {
        return new secrets_grpc_pb_1.SecretsClient(this.endpoint, this.channelCredentials, {
            interceptors: [this.tokenRefresherInterceptor]
        });
    }
    getAgentsClient() {
        return new agents_grpc_pb_1.AgentsClient(this.endpoint, this.channelCredentials, {
            interceptors: [this.tokenRefresherInterceptor]
        });
    }
    getAclsClient() {
        return new acls_grpc_pb_1.AclsClient(this.endpoint, this.channelCredentials, {
            interceptors: [this.tokenRefresherInterceptor]
        });
    }
    getDomainsClient() {
        return new domains_grpc_pb_1.DomainsClient(this.endpoint, this.channelCredentials, {
            interceptors: [this.tokenRefresherInterceptor]
        });
    }
    getTrunksClient() {
        return new trunks_grpc_pb_1.TrunksClient(this.endpoint, this.channelCredentials, {
            interceptors: [this.tokenRefresherInterceptor]
        });
    }
    getNumbersClient() {
        return new numbers_grpc_pb_1.NumbersClient(this.endpoint, this.channelCredentials, {
            interceptors: [this.tokenRefresherInterceptor]
        });
    }
    getCredentialsClient() {
        return new credentials_grpc_pb_1.CredentialsServiceClient(this.endpoint, this.channelCredentials, {
            interceptors: [this.tokenRefresherInterceptor]
        });
    }
}
exports.Client = Client;
