// @fonoster/sdk/dist/node/client/TokenRefresherWeb.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenRefresherWeb = void 0;
const isJwtExpired_1 = require("./isJwtExpired");
class TokenRefresherWeb {
    client;
    constructor(client) {
        this.client = client;
    }
    intercept(request, invoker) {
        class InterceptedStream {
            refresher;
            stream;
            constructor(stream, refresher) {
                this.stream = stream;
                this.refresher = refresher;
            }
            on(eventType, callback) {
                const token = this.refresher.client.getAccessToken();
                if ((0, isJwtExpired_1.isJwtExpired)(token)) {
                    this.refresher.client.refreshToken().then(() => { });
                }
                if (eventType == "data") {
                    this.stream.on(eventType, callback);
                }
                else if (eventType == "error") {
                    this.stream.on("error", callback);
                }
                else if (eventType == "metadata") {
                    this.stream.on("metadata", callback);
                }
                else if (eventType == "status") {
                    this.stream.on("status", callback);
                }
                else if (eventType == "end") {
                    this.stream.on("end", callback);
                }
                return this;
            }
        }
        return new InterceptedStream(invoker(request), this);
    }
}
exports.TokenRefresherWeb = TokenRefresherWeb;
