// @fonoster/common/dist/assistants/conversationSettingsSchema.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.conversationSettingsSchema = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const zod_1 = require("zod");
const Messages = __importStar(require("../messages"));
const ZERO_OR_GREATER_THAN_ZERO = "Must be a number greater than or equal to 0";
const NUMBER_BETWEEN_0_AND_1 = "Must be a number between 0 and 1";
const MAX_SPEECH_WAIT_TIMEOUT = 0;
const IDLE_OPTIONS_TIMEOUT = 30000;
const IDLE_OPTIONS_MAX_TIMEOUT_COUNT = 2;
const VAD_ACTIVATION_THRESHOLD = 0.4;
const VAD_DEACTIVATION_THRESHOLD = 0.25;
const VAD_DEBOUNCE_FRAMES = 4;
const conversationSettingsSchema = zod_1.z.object({
    firstMessage: zod_1.z.string().optional(),
    systemPrompt: zod_1.z.string(),
    goodbyeMessage: zod_1.z.string(),
    systemErrorMessage: zod_1.z.string(),
    initialDtmf: zod_1.z
        .string()
        .regex(/^[0-9*#]+$/, { message: Messages.VALID_DTMF })
        .optional(),
    maxSessionDuration: zod_1.z
        .number()
        .int()
        .positive()
        .default(30 * 60 * 1000), // 30 minutes
    maxSpeechWaitTimeout: zod_1.z
        .number()
        .int({ message: Messages.POSITIVE_INTEGER_MESSAGE })
        .gte(0, { message: ZERO_OR_GREATER_THAN_ZERO })
        .default(MAX_SPEECH_WAIT_TIMEOUT),
    transferOptions: zod_1.z
        .object({
        phoneNumber: zod_1.z.string(),
        message: zod_1.z.string(),
        timeout: zod_1.z
            .number()
            .int({ message: Messages.POSITIVE_INTEGER_MESSAGE })
            .positive({ message: Messages.POSITIVE_INTEGER_MESSAGE })
            .default(30000)
    })
        .optional(),
    idleOptions: zod_1.z.object({
        message: zod_1.z.string(),
        timeout: zod_1.z
            .number()
            .int({ message: Messages.POSITIVE_INTEGER_MESSAGE })
            .positive({ message: Messages.POSITIVE_INTEGER_MESSAGE })
            .default(IDLE_OPTIONS_TIMEOUT),
        maxTimeoutCount: zod_1.z
            .number()
            .int({ message: Messages.POSITIVE_INTEGER_MESSAGE })
            .positive({ message: Messages.POSITIVE_INTEGER_MESSAGE })
            .default(IDLE_OPTIONS_MAX_TIMEOUT_COUNT)
    }),
    allowUserBargeIn: zod_1.z.boolean().optional().default(true),
    vad: zod_1.z
        .object({
        pathToModel: zod_1.z.string().optional(),
        activationThreshold: zod_1.z
            .number()
            .max(1)
            .min(0)
            .positive({ message: NUMBER_BETWEEN_0_AND_1 }),
        deactivationThreshold: zod_1.z
            .number()
            .max(1)
            .min(0)
            .positive({ message: NUMBER_BETWEEN_0_AND_1 }),
        debounceFrames: zod_1.z
            .number()
            .int({ message: Messages.POSITIVE_INTEGER_MESSAGE })
            .positive({ message: Messages.POSITIVE_INTEGER_MESSAGE })
    })
        .default({
        activationThreshold: VAD_ACTIVATION_THRESHOLD,
        deactivationThreshold: VAD_DEACTIVATION_THRESHOLD,
        debounceFrames: VAD_DEBOUNCE_FRAMES
    })
});
exports.conversationSettingsSchema = conversationSettingsSchema;
