// @fonoster/common/dist/assistants/languageModelConfigSchema.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.languageModelConfigSchema = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const zod_1 = require("zod");
const Messages = __importStar(require("../messages"));
const LanguageModelProvider_1 = require("./LanguageModelProvider");
const tools_1 = require("./tools");
const NUMBER_BETWEEN_0_AND_2 = "Must be a number between 0 and 2";
const languageModelConfigSchema = zod_1.z.object({
    provider: zod_1.z.nativeEnum(LanguageModelProvider_1.LanguageModelProvider, {
        message: "Invalid language model provider."
    }),
    apiKey: zod_1.z.string().optional(),
    model: zod_1.z.string(),
    temperature: zod_1.z
        .number()
        .max(2, { message: NUMBER_BETWEEN_0_AND_2 })
        .min(0, { message: NUMBER_BETWEEN_0_AND_2 }),
    maxTokens: zod_1.z
        .number()
        .int({ message: Messages.POSITIVE_INTEGER_MESSAGE })
        .positive({ message: Messages.POSITIVE_INTEGER_MESSAGE }),
    baseUrl: zod_1.z
        .string()
        .url({
        message: Messages.VALID_URL
    })
        .optional(),
    knowledgeBase: zod_1.z
        .array(zod_1.z.object({
        type: zod_1.z.enum(["s3"]),
        title: zod_1.z.string(),
        document: zod_1.z.string().regex(/\.pdf$/, {
            message: "Document must be a pdf file"
        })
    }))
        .default([]),
    tools: zod_1.z.array(tools_1.toolSchema).default([])
});
exports.languageModelConfigSchema = languageModelConfigSchema;
