// @fonoster/common/dist/assistants/testCasesSchema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.testCasesSchema = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const types_1 = require("@fonoster/types");
const zod_1 = require("zod");
const testCasesSchema = zod_1.z.object({
    evalsLanguageModel: zod_1.z.object({
        provider: zod_1.z.enum(["openai"], {
            message: "Invalid language model provider. Only OpenAI is supported for evals."
        }),
        model: zod_1.z.string(),
        apiKey: zod_1.z.string().optional()
    }),
    evalsSystemPrompt: zod_1.z.string().optional(),
    scenarios: zod_1.z.array(zod_1.z.object({
        ref: zod_1.z.string(),
        description: zod_1.z.string(),
        telephonyContext: zod_1.z.object({
            callDirection: zod_1.z.nativeEnum(types_1.CallDirection),
            ingressNumber: zod_1.z.string(),
            callerNumber: zod_1.z.string(),
            metadata: zod_1.z.record(zod_1.z.string(), zod_1.z.string()).optional()
        }),
        conversation: zod_1.z.array(zod_1.z.object({
            userInput: zod_1.z.string(),
            expected: zod_1.z.object({
                text: zod_1.z.object({
                    type: zod_1.z.enum(["EXACT", "SIMILAR"]),
                    response: zod_1.z.string()
                }),
                tools: zod_1.z
                    .array(zod_1.z.object({
                    tool: zod_1.z.string(),
                    parameters: zod_1.z.record(zod_1.z.string(), zod_1.z.any())
                }))
                    .optional()
            })
        }))
    }))
});
exports.testCasesSchema = testCasesSchema;
