// @fonoster/sdk/dist/node/Acls.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Acls = void 0;
const makeRpcRequest_1 = require("./client/makeRpcRequest");
const acls_pb_1 = require("./generated/node/acls_pb");
/**
 * @classdesc Fonoster Acls, part of the Fonoster SIP Proxy subsystem,
 * allows you to create, update, retrieve, and delete Access Control Lists (ACLs) rules for your deployment.
 * Note that an active Fonoster deployment is required.
 *
 * @example
 * const SDK = require("@fonoster/sdk");
 *
 * async function main(request) {
 *   const apiKey = "your-api-key";
 *   const apiSecret = "your-api-secret"
 *   const accessKeyId = "WO00000000000000000000000000000000";
 *
 *   try {
 *     const client = SDK.Client({ accessKeyId });
 *     await client.loginWithApiKey(apiKey, apiSecret);
 *
 *     const acls = new SDK.Acls(client);
 *     const response = await acls.createAcl(request);
 *
 *     console.log(response); // successful response
 *   } catch (e) {
 *     console.error(e); // an error occurred
 *   }
 * }
 *
 * const request = {
 *   name: "My ACL",
 *   allow: ["47.132.130.31"] // Allow only this IP
 * };
 *
 * main(request);
 */
class Acls {
    client;
    /**
     * Constructs a new Acls object.
     *
     * @param {FonosterClient} client - Client object with underlying implementations to make requests to Fonoster's API
     * @see AbstractClient
     * @see FonosterClient
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Creates a new Acl in the Workspace.
     *
     * @param {CreateAclRequest} request - The request object that contains the necessary information to create a new Acl
     * @param {string} request.name - The name of the Acl
     * @param {string[]} request.allow - The list of IPs to allow
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the created Acl
     * @example
     * const acls = new SDK.Acls(client); // Existing client object
     *
     * const request = {
     *   name: "My ACL",
     *   allow: ["47.132.130.31"] // Allow only this IP
     * };
     *
     * acls
     *   .createAcl(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async createAcl(request) {
        const client = this.client.getAclsClient();
        const createAclRequest = new acls_pb_1.CreateAclRequest();
        createAclRequest.setName(request.name);
        createAclRequest.setAllowList(request.allow);
        return new Promise((resolve, reject) => {
            client.createAcl(createAclRequest, this.client.getMetadata(), (err, response) => {
                if (err) {
                    reject(err);
                    return;
                }
                resolve(response.toObject());
            });
        });
    }
    /**
     * Retrieves an existing Acl in the Workspace.
     *
     * @param {string} ref - The reference of the Acl to retrieve
     * @return {Promise<Acl>} - The response object that contains the Acl information
     * @example
     * const acls = new SDK.Acls(client); // Existing client object
     *
     * const ref = "00000000-0000-0000-0000-000000000000";
     *
     * acls
     *   .getAcl(ref)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async getAcl(ref) {
        const updateAclRequest = new acls_pb_1.GetAclRequest();
        updateAclRequest.setRef(ref);
        const client = this.client.getAclsClient();
        return new Promise((resolve, reject) => {
            client.getAcl(updateAclRequest, this.client.getMetadata(), (err, response) => {
                if (err) {
                    reject(err);
                    return;
                }
                const obj = response.toObject();
                resolve({
                    ...obj,
                    allow: obj.allowList,
                    createdAt: new Date(obj.createdAt * 1000),
                    updatedAt: new Date(obj.updatedAt * 1000)
                });
            });
        });
    }
    /**
     * Updates an existing Acl in the Workspace.
     *
     * @param {UpdateAclRequest} request - The request object that contains the necessary information to update an existing Acl
     * @param {string} request.ref - The reference of the Acl to update
     * @param {string} request.name - The name of the Acl
     * @param {string[]} request.allow - The list of IPs to allow
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the updated Acl
     * @example
     * const acls = new SDK.Acls(client); // Existing client object
     *
     * const request = {
     *   ref: "00000000-0000-0000-0000-000000000000",
     *   name: "My ACL",
     *   allow: ["47.132.130.31"] // Allow only this IP
     * };
     *
     * acls
     *   .updateAcl(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async updateAcl(request) {
        const client = this.client.getAclsClient();
        const updateAclRequest = new acls_pb_1.UpdateAclRequest();
        updateAclRequest.setRef(request.ref);
        updateAclRequest.setName(request.name);
        updateAclRequest.setAllowList(request.allow);
        return new Promise((resolve, reject) => {
            client.updateAcl(updateAclRequest, this.client.getMetadata(), (err, response) => {
                if (err) {
                    reject(err);
                    return;
                }
                resolve(response.toObject());
            });
        });
    }
    /**
     * Retrieves a list of Acls from a Workspace.
     *
     * @param {ListAclsRequest} request - The request object that contains the necessary information to retrieve a list of Acls
     * @param {number} request.pageSize - The number of Acls to retrieve
     * @param {string} request.pageToken - The token to retrieve the next page of Acls
     * @return {Promise<ListAclsResponse>} - The response object that contains the list of Acls
     * @example
     * const acls = new SDK.Acls(client); // Existing client object
     *
     * const request = {
     *   pageSize: 10,
     *   pageToken: "00000000-0000-0000-0000-000000000000"
     * };
     *
     * acls
     *   .listAcls(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async listAcls(request) {
        const client = this.client.getAclsClient();
        const listAclsRequest = new acls_pb_1.ListAclsRequest();
        listAclsRequest.setPageSize(request.pageSize);
        listAclsRequest.setPageToken(request.pageToken);
        return new Promise((resolve, reject) => {
            client.listAcls(listAclsRequest, this.client.getMetadata(), (err, response) => {
                if (err) {
                    reject(err);
                    return;
                }
                const items = response.getItemsList().map((item) => {
                    const obj = item.toObject();
                    return {
                        ...obj,
                        allow: obj.allowList,
                        createdAt: new Date(obj.createdAt * 1000),
                        updatedAt: new Date(obj.updatedAt * 1000)
                    };
                });
                resolve({
                    items: items,
                    nextPageToken: response.getNextPageToken()
                });
            });
        });
    }
    /**
     * Deletes an existing Acl from Fonoster.
     * Note that this operation is irreversible.
     *
     * @param {string} ref - The reference of the Acl to delete
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the deleted Acl
     * @example
     * const acls = new SDK.Acls(client); // Existing client object
     *
     * const ref = "00000000-0000-0000-0000-000000000000";
     *
     * acls
     *   .deleteAcl(ref)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async deleteAcl(ref) {
        const client = this.client.getAclsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.deleteAcl.bind(client),
            requestPBObjectConstructor: acls_pb_1.DeleteAclRequest,
            metadata: this.client.getMetadata(),
            request: { ref }
        });
    }
}
exports.Acls = Acls;
