// @fonoster/sdk/dist/node/Agents.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Agents = void 0;
const makeRpcRequest_1 = require("./client/makeRpcRequest");
const agents_pb_1 = require("./generated/node/agents_pb");
/**
 * @classdesc Fonoster Agents, part of the Fonoster SIP Proxy subsystem,
 * allows you to create, update, retrieve, and delete SIP Agents for your deployment.
 * Note that an active Fonoster deployment is required.
 *
 * @example
 * const SDK = require("@fonoster/sdk");
 *
 * async function main(request) {
 *   const apiKey = "your-api-key";
 *   const apiSecret = "your-api-secret"
 *   const accessKeyId = "WO00000000000000000000000000000000";
 *
 *   try {
 *     const client = SDK.Client({ accessKeyId });
 *     await client.loginWithApiKey(apiKey, apiSecret);
 *
 *     const agents = new SDK.Agents(client);
 *     const response = await agents.createAgent(request);
 *
 *     console.log(response); // successful response
 *   } catch (e) {
 *     console.error(e); // an error occurred
 *   }
 * }
 *
 * const request = {
 *   name: "John Doe",
 *   username: "1001",
 *   privacy: "PRIVATE",
 *   enabled: true,
 *   maxContacts: 3
 *   domainRef: "00000000-0000-0000-0000-000000000000"
 * };
 *
 * main(request);
 */
class Agents {
    client;
    /**
     * Constructs a new Agents object.
     *
     * @param {FonosterClient} client - Client object with underlying implementations to make requests to Fonoster's API
     * @see AbstractClient
     * @see FonosterClient
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Creates a new Agent in the Workspace.
     *
     * @param {CreateAgentRequest} request - The request object that contains the necessary information to create a new Agent
     * @param {string} request.name - The name of the Agent
     * @param {string} request.username - The username of the Agent
     * @param {Privacy} request.privacy - The privacy of the Agent
     * @param {boolean} request.enabled - The status of the Agent
     * @param {number} request.maxContacts - The maximum number of contacts the Agent can have
     * @param {string} request.domainRef - The reference of the Domain to associate the Agent
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the created Agent
     * @example
     * const agents = new SDK.Agents(client); // Existing client object
     *
     * const request = {
     *   name: "John Doe",
     *   username: "1001",
     *   privacy: "PRIVATE",
     *   enabled: true,
     *   maxContacts: 3
     *   domainRef: "00000000-0000-0000-0000-000000000000"
     * };
     *
     * agents
     *   .createAgent(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async createAgent(request) {
        const client = this.client.getAgentsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.createAgent.bind(client),
            requestPBObjectConstructor: agents_pb_1.CreateAgentRequest,
            metadata: this.client.getMetadata(),
            request,
            enumMapping: [["privacy", agents_pb_1.Privacy]]
        });
    }
    /**
     * Retrieves an existing Agent in the Workspace.
     *
     * @param {string} ref - The reference of the Agent to retrieve
     * @return {Promise<Acl>} - The response object that contains the Agent information
     * @example
     * const agents = new SDK.Agents(client); // Existing client object
     *
     * const ref = "00000000-0000-0000-0000-000000000000";
     *
     * agents
     *   .getAgent(ref)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async getAgent(ref) {
        const client = this.client.getAgentsClient();
        const response = await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.getAgent.bind(client),
            requestPBObjectConstructor: agents_pb_1.GetAgentRequest,
            metadata: this.client.getMetadata(),
            request: { ref },
            enumMapping: [["privacy", agents_pb_1.Privacy]]
        });
        const credentials = response?.credentials?.toObject();
        const domain = response?.domain?.toObject();
        return response
            ? {
                ...response,
                credentials,
                domain
            }
            : null;
    }
    /**
     * Updates an existing Agent in the Workspace.
     *
     * @param {UpdateAgentRequest} request - The request object that contains the necessary information to update an existing Agent
     * @param {string} request.ref - The reference of the Agent to update
     * @param {string} request.name - The name of the Agent
     * @param {Privacy} request.privacy - The privacy of the Agent
     * @param {boolean} request.enabled - The status of the Agent
     * @param {number} request.maxContacts - The maximum number of contacts the Agent can have
     * @param {string} request.domainRef - The reference of the Domain to associate the Agent
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the updated Agent
     * @example
     * const agents = new SDK.Agents(client); // Existing client object
     *
     * const request = {
     *   ref: "00000000-0000-0000-0000-000000000000",
     *   name: "John Doe",
     *   privacy: "PRIVATE",
     *   enabled: true,
     *   maxContacts: 3
     *   domainRef: "00000000-0000-0000-0000-000000000000"
     * };
     *
     * agents
     *   .updateAgent(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async updateAgent(request) {
        const client = this.client.getAgentsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.updateAgent.bind(client),
            requestPBObjectConstructor: agents_pb_1.UpdateAgentRequest,
            metadata: this.client.getMetadata(),
            request,
            enumMapping: [["privacy", agents_pb_1.Privacy]]
        });
    }
    /**
     * Retrieves a list of Agents from a Workspace.
     *
     * @param {ListAgentsRequest} request - The request object that contains the necessary information to retrieve a list of Agents
     * @param {number} request.pageSize - The number of Agents to retrieve
     * @param {string} request.pageToken - The token to retrieve the next page of Agents
     * @return {Promise<ListAgentsResponse>} - The response object that contains the list of Agents
     * @example
     * const agents = new SDK.Agents(client); // Existing client object
     *
     * const request = {
     *  pageSize: 10,
     *  pageToken: "00000000-0000-0000-0000-000000000000"
     * };
     *
     * agents
     *   .listAgents(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async listAgents(request) {
        const client = this.client.getAgentsClient();
        const response = await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.listAgents.bind(client),
            requestPBObjectConstructor: agents_pb_1.ListAgentsRequest,
            metadata: this.client.getMetadata(),
            request,
            enumMapping: [["privacy", agents_pb_1.Privacy]],
            repeatableObjectMapping: [["itemsList", agents_pb_1.Agent]]
        });
        // Manually extract nested objects for each agent
        const processedItems = response.items?.map((agent) => {
            const credentials = agent?.credentials?.toObject();
            const domain = agent?.domain?.toObject();
            return {
                ...agent,
                credentials,
                domain
            };
        });
        return {
            ...response,
            items: processedItems || []
        };
    }
    /**
     * Deletes an existing Agent from Fonoster.
     * Note that this operation is irreversible.
     *
     * @param {string} ref - The reference of the Agent to delete
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the deleted Agent
     * @example
     * const agents = new SDK.Agents(client); // Existing client object
     *
     * const ref = "00000000-0000-0000-0000-000000000000";
     *
     * agents
     *   .deleteAgent(ref)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async deleteAgent(ref) {
        const applicationsClient = this.client.getAgentsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: applicationsClient.deleteAgent.bind(applicationsClient),
            requestPBObjectConstructor: agents_pb_1.DeleteAgentRequest,
            metadata: this.client.getMetadata(),
            request: { ref }
        });
    }
}
exports.Agents = Agents;
