// @fonoster/sdk/dist/node/Applications.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Applications = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const types_1 = require("@fonoster/types");
const empty_pb_1 = require("google-protobuf/google/protobuf/empty_pb");
const struct_pb_1 = require("google-protobuf/google/protobuf/struct_pb");
const makeRpcRequest_1 = require("./client/makeRpcRequest");
const applications_pb_1 = require("./generated/node/applications_pb");
const utils_1 = require("./utils");
/**
 * @classdesc Fonoster Applications, part of the Fonoster Voice Subsystem,
 * allow you to create, update, retrieve, and delete Voice Applications.
 * Note that an active Fonoster deployment is required.
 *
 * @example
 *
 * const SDK = require("@fonoster/sdk");
 *
 * async function main(request) {
 *   const apiKey = "your-api-key";
 *   const apiSecret = "your-api-secret"
 *   const accessKeyId = "WO00000000000000000000000000000000";
 *
 *   try {
 *     const client = SDK.Client({ accessKeyId });
 *     await client.loginWithApiKey(apiKey, apiSecret);
 *
 *     const apps = new SDK.Applications(client);
 *     const response = await apps.createApplication(request);
 *
 *     console.log(response); // successful response
 *   } catch (e) {
 *     console.error(e); // an error occurred
 *   }
 * }
 *
 * const request = {
 *   name: "My application",
 *   type: "EXTERNAL",
 *   endpoint: "welcome.demo.fonoster.local", // Built-in demo application
 *   speechToText: {
 *     productRef: "stt.deepgram",
 *     config: {
 *       model: "nova-2",
 *       languageCode: "en-US"
 *     }
 *   },
 *   textToSpeech: {
 *     productRef: "tts.elevenlabs",
 *     config: {
 *       voice: "lrTWbMInQjSJ9q5ywFKP"
 *     }
 *   }
 * };
 *
 * main(request);
 */
class Applications {
    client;
    /**
     * Constructs a new Applications object.
     *
     * @param {FonosterClient} client - Client object with underlying implementations to make requests to Fonoster's API
     * @see AbstractClient
     * @see FonosterClient
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Creates a new Application in Fonoster. The only required fields are the name and type of the application.
     *
     * @param {CreateApplicationRequest} request - The request object that contains the necessary information to create a new application
     * @param {string} request.name - The name of the application
     * @param {ApplicationType} request.type - The type of application (e.g., EXTERNAL)
     * @param {string} request.endpoint - The endpoint where the application is hosted
     * @param {SpeechToText} request.speechToText - The speech-to-text configuration
     * @param {string} request.speechToText.productRef - The product reference of the speech-to-text engine (e.g., stt.deepgram)
     * @param {object} request.speechToText.config - The configuration object for the speech-to-text engine (e.g., { model: "nova-2", languageCode: "en-US" })
     * @param {TextToSpeech} request.textToSpeech - The text-to-speech configuration
     * @param {string} request.textToSpeech.productRef - The product reference of the text-to-speech engine (e.g., tts.elevenlabs)
     * @param {object} request.textToSpeech.config - The configuration object for the text-to-speech engine (e.g., { voice: "lrTWbMInQjSJ9q5ywFKP" })
     * @param {Intelligence} request.intelligence - The intelligence configuration
     * @param {string} request.intelligence.productRef - The product reference of the intelligence engine (e.g., llm.groq)
     * @param {object} request.intelligence.config - The configuration object for the intelligence engine
     * @return {Promise<CreateAppResponse>} - The response object that contains the reference to the newly created application
     * @example
     * const apps = new SDK.Applications(client); // Existing client object
     *
     * const request = {
     *   name: "My application",
     *   type: "EXTERNAL",
     *   endpoint: "welcome.demo.fonoster.local", // Built-in demo application
     *   speechToText: {
     *     productRef: "stt.deepgram",
     *     config: {
     *       languageCode: "en-US"
     *     }
     *   },
     *   textToSpeech: {
     *     productRef: "tts.elevenlabs",
     *     config: {
     *       voice: "lrTWbMInQjSJ9q5ywFKP"
     *     }
     *   }
     * };
     *
     * apps
     *   .createApplication(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async createApplication(request) {
        const reqWithStructOverride = (0, utils_1.buildStructOverride)(request);
        const applicationsClient = this.client.getApplicationsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: applicationsClient.createApplication.bind(applicationsClient),
            requestPBObjectConstructor: applications_pb_1.CreateApplicationRequest,
            metadata: this.client.getMetadata(),
            request: reqWithStructOverride,
            enumMapping: [["type", applications_pb_1.ApplicationType]],
            objectMapping: [
                ["textToSpeech", applications_pb_1.ProductContainer],
                ["speechToText", applications_pb_1.ProductContainer],
                ["intelligence", applications_pb_1.ProductContainer]
            ]
        });
    }
    /**
     * Retrieves an existing Application in the Workspace.
     *
     * @param {string} ref - The reference of the Application to retrieve
     * @return {Promise<Application>} - The response object that contains the Application information
     * @example
     * const apps = new SDK.Applications(client); // Existing client object
     *
     * const ref = "00000000-0000-0000-0000-000000000000";
     *
     * apps
     *   .getApplication(ref)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async getApplication(ref) {
        const applicationsClient = this.client.getApplicationsClient();
        const response = await (0, makeRpcRequest_1.makeRpcRequest)({
            method: applicationsClient.getApplication.bind(applicationsClient),
            requestPBObjectConstructor: applications_pb_1.GetApplicationRequest,
            metadata: this.client.getMetadata(),
            request: { ref },
            enumMapping: [["type", applications_pb_1.ApplicationType]],
            objectMapping: [
                ["textToSpeech", applications_pb_1.ProductContainer],
                ["speechToText", applications_pb_1.ProductContainer],
                ["intelligence", applications_pb_1.ProductContainer]
            ]
        });
        return (0, utils_1.buildStructOverrideReverse)(response);
    }
    /**
     * Updates an existing application in Fonoster.
     *
     * @param {UpdateApplicationRequest} request - The request object that contains the necessary information to update an application
     * @param {string} request.ref - The reference of the application to update
     * @param {string} request.name - The name of the application
     * @param {string} request.endpoint - The endpoint where the application is hosted
     * @param {SpeechToText} request.speechToText - The speech-to-text configuration
     * @param {string} request.speechToText.productRef - The product reference of the speech-to-text engine (e.g., stt.deepgram)
     * @param {object} request.speechToText.config - The configuration object for the speech-to-text engine (e.g., { model: "nova-2", languageCode: "en-US" })
     * @param {TextToSpeech} request.textToSpeech - The text-to-speech configuration
     * @param {string} request.textToSpeech.productRef - The product reference of the text-to-speech engine (e.g., tts.elevenlabs)
     * @param {object} request.textToSpeech.config - The configuration object for the text-to-speech engine (e.g., { voice: "lrTWbMInQjSJ9q5ywFKP" })
     * @param {Intelligence} request.intelligence - The intelligence configuration
     * @param {string} request.intelligence.productRef - The product reference of the intelligence engine (e.g., llm.groq)
     * @param {object} request.intelligence.config - The configuration object for the intelligence engine
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the updated application
     * @example
     * const apps = new SDK.Applications(client); // Existing client object
     *
     * const request = {
     *   ref: "00000000-0000-0000-0000-000000000000",
     *   name: "My application",
     *   endpoint: "welcome.demo.fonoster.local", // Built-in demo application
     * };
     *
     * apps
     *   .updateApplication(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async updateApplication(request) {
        const reqWithStructOverride = (0, utils_1.buildStructOverride)(request);
        const applicationsClient = this.client.getApplicationsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: applicationsClient.updateApplication.bind(applicationsClient),
            requestPBObjectConstructor: applications_pb_1.UpdateApplicationRequest,
            metadata: this.client.getMetadata(),
            request: reqWithStructOverride,
            enumMapping: [["type", applications_pb_1.ApplicationType]],
            objectMapping: [
                ["textToSpeech", applications_pb_1.ProductContainer],
                ["speechToText", applications_pb_1.ProductContainer],
                ["intelligence", applications_pb_1.ProductContainer]
            ]
        });
    }
    /**
     * Retrieves a list of Applications from Fonoster.
     *
     * @param {ListApplicationsRequest} request - The request object that contains the necessary information to retrieve a list of Applications
     * @param {number} request.pageSize - The number of Applications to retrieve
     * @param {string} request.pageToken - The token to retrieve the next page of Applications
     * @return {Promise<ListApplicationsResponse>} - The response object that contains the list of Applications
     * @example
     * const apps = new SDK.Applications(client); // Existing client object
     *
     * const request = {
     *   pageSize: 10,
     *   pageToken: "00000000-0000-0000-0000-000000000000"
     * };
     *
     * apps
     *   .listApplications(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async listApplications(request) {
        const applicationsClient = this.client.getApplicationsClient();
        const response = await (0, makeRpcRequest_1.makeRpcRequest)({
            method: applicationsClient.listApplications.bind(applicationsClient),
            requestPBObjectConstructor: applications_pb_1.ListApplicationsRequest,
            metadata: this.client.getMetadata(),
            request,
            enumMapping: [["type", applications_pb_1.ApplicationType]],
            objectMapping: [
                ["speechToText", applications_pb_1.ProductContainer],
                ["textToSpeech", applications_pb_1.ProductContainer],
                ["intelligence", applications_pb_1.ProductContainer]
            ],
            repeatableObjectMapping: [["itemsList", applications_pb_1.Application]]
        });
        return {
            items: response.items.map(utils_1.buildStructOverrideReverse),
            nextPageToken: response.nextPageToken
        };
    }
    /**
     * Deletes an existing Application from Fonoster.
     * Note that this operation is irreversible.
     *
     * @param {string} ref - The reference of the Application to delete
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the deleted application
     * @example
     * const apps = new SDK.Applications(client); // Existing client object
     *
     * const ref = "00000000-0000-0000-0000-000000000000";
     *
     * apps
     *   .deleteApplication(ref)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async deleteApplication(ref) {
        const applicationsClient = this.client.getApplicationsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: applicationsClient.deleteApplication.bind(applicationsClient),
            requestPBObjectConstructor: applications_pb_1.DeleteApplicationRequest,
            metadata: this.client.getMetadata(),
            request: { ref }
        });
    }
    /**
     * Evaluates the intelligence of an application.
     *
     * @param {EvaluateIntelligenceRequest} request - The request object that contains the necessary information to evaluate the intelligence of an application
     * @param {string} request.intelligence.productRef - The product reference of the intelligence engine (e.g., llm.groq)
     * @param {object} request.intelligence.config - The configuration object for the intelligence engine
     * @return {AsyncGenerator<EvaluateIntelligenceEvent>} - Stream of evaluation events (step results, scenario summaries, or errors)
     * @example
     * const apps = new SDK.Applications(client);
     * for await (const event of apps.evaluateIntelligence(request)) {
     *   if (event.type === "stepResult") console.log(event.stepResult);
     *   if (event.type === "scenarioSummary") console.log(event.scenarioRef, event.overallPassed);
     *   if (event.type === "evalError") console.error(event.message);
     * }
     */
    evaluateIntelligence(request) {
        const applicationsClient = this.client.getApplicationsClient();
        const requestPB = new applications_pb_1.EvaluateIntelligenceRequest();
        const productContainer = new applications_pb_1.ProductContainer();
        productContainer.setProductRef(request.intelligence.productRef);
        productContainer.setConfig(struct_pb_1.Struct.fromJavaScript(request.intelligence.config));
        requestPB.setIntelligence(productContainer);
        const metadata = this.client.getMetadata();
        const call = applicationsClient.evaluateIntelligence(requestPB, metadata);
        return this.evaluateIntelligenceStreamGenerator(call);
    }
    async *evaluateIntelligenceStreamGenerator(call) {
        const queue = [];
        let done = false;
        let streamError = null;
        call.on("data", (chunk) => {
            const event = this.mapEvalEventFromPb(chunk);
            if (event)
                queue.push(event);
        });
        call.on("end", () => {
            done = true;
        });
        call.on("error", (err) => {
            streamError = err instanceof Error ? err : new Error(String(err));
            done = true;
        });
        while (!done || queue.length > 0) {
            if (streamError)
                throw streamError;
            if (queue.length > 0) {
                const next = queue.shift();
                yield next;
            }
            else {
                await new Promise((r) => setTimeout(r, 50));
            }
        }
    }
    mapEvalEventFromPb(msg) {
        if (msg.getStepResult?.()) {
            const sr = msg.getStepResult();
            const report = sr.getReport();
            const toolList = report.getToolEvaluationsList?.() ?? [];
            return {
                type: "stepResult",
                scenarioRef: sr.getScenarioRef(),
                stepResult: {
                    humanInput: report.getHumanInput(),
                    expectedResponse: report.getExpectedResponse(),
                    aiResponse: report.getAiResponse(),
                    evaluationType: report.getEvaluationType() === 0
                        ? types_1.ExpectedTextType.EXACT
                        : types_1.ExpectedTextType.SIMILAR,
                    passed: report.getPassed(),
                    errorMessage: report.getErrorMessage() || undefined,
                    toolEvaluations: toolList.map((t) => ({
                        expectedTool: t.getExpectedTool(),
                        actualTool: t.getActualTool(),
                        passed: t.getPassed(),
                        expectedParameters: t.getExpectedParameters?.()?.toJavaScript() ?? undefined,
                        actualParameters: t.getActualParameters?.()?.toJavaScript() ?? undefined,
                        errorMessage: t.getErrorMessage?.() || undefined
                    }))
                }
            };
        }
        if (msg.getScenarioSummary?.()) {
            const ss = msg.getScenarioSummary();
            return {
                type: "scenarioSummary",
                scenarioRef: ss.getScenarioRef(),
                overallPassed: ss.getOverallPassed()
            };
        }
        if (msg.getEvalError?.()) {
            const e = msg.getEvalError();
            return { type: "evalError", message: e.getMessage() };
        }
        return null;
    }
    /**
     * Creates an Ephemeral Agent token for test calls.
     *
     * @return {Promise<TestTokenResponse>} - The response object containing the ephemeral agent token and related info
     * @example
     * const apps = new SDK.Applications(client);
     * const response = await apps.createTestToken();
     * console.log(response.token); // JWT token for test call
     */
    async createTestToken() {
        const applicationsClient = this.client.getApplicationsClient();
        const response = await (0, makeRpcRequest_1.makeRpcRequest)({
            method: applicationsClient.createTestToken.bind(applicationsClient),
            requestPBObjectConstructor: empty_pb_1.Empty,
            metadata: this.client.getMetadata(),
            request: {}
        });
        return response;
    }
}
exports.Applications = Applications;
