// @fonoster/sdk/dist/node/Calls.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Calls = void 0;
const struct_pb_1 = require("google-protobuf/google/protobuf/struct_pb");
const makeRpcRequest_1 = require("./client/makeRpcRequest");
const calls_pb_1 = require("./generated/node/calls_pb");
const utils_1 = require("./utils");
/**
 * @classdesc Fonoster Calls, part of the Fonoster Media subsystem,
 * allows you to create, list, and track calls in your deployment.
 * Note that an active Fonoster deployment is required.
 *
 * @example
 *
 * const SDK = require("@fonoster/sdk");
 *
 * async function main(request) {
 *   const apiKey = "your-api-key";
 *   const apiSecret = "your-api-secret"
 *   const accessKeyId = "WO00000000000000000000000000000000";
 *
 *   try {
 *     const client = SDK.Client({ accessKeyId });
 *     await client.loginWithApiKey(apiKey, apiSecret);
 *
 *     const calls = new SDK.Calls(client);
 *     const response = await calls.createCall(request);
 *
 *     console.log(response); // successful response
 *   } catch (e) {
 *     console.error(e); // an error occurred
 *   }
 * }
 *
 * const request = {
 *   from: "+18287854037",
 *   to: "+17853178070",
 *   appRef: "00000000-0000-0000-0000-000000000000"
 * };
 *
 * main(request);
 */
class Calls {
    client;
    /**
     * Constructs a new Calls object.
     *
     * @param {FonosterClient} client - Client object with underlying implementations to make requests to Fonoster's API
     * @see AbstractClient
     * @see FonosterClient
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Creates a new Call in the Workspace.
     *
     * @param {CreateCallRequest} request - The request object that contains the necessary information to create a new Call
     * @param {string} request.from - The number that originated the call
     * @param {string} request.to - The number that received the call
     * @param {string} request.appRef - The reference of the App that will handle the call
     * @param {number} request.timeout - The time in seconds to wait for the call to be answered. Default is 60 seconds
     * @param {Record<string, string>} request.metadata - Optional metadata to be sent to the App. For Autopilot applications, this is added to the context of the conversation.
     * @return {{ref: string, statusStream: AsyncGenerator<{ status: DialStatus }>}} - The response object that contains the Call reference and a stream of status updates
     * @see DialStatus
     * @example
     * const calls = new SDK.Calls(client); // Existing client object
     *
     * const request = {
     *   from: "+18287854037",
     *   to: "+17853178070",
     *   appRef: "00000000-0000-0000-0000-000000000000",
     *   timeout: 30,
     *   metadata: {
     *     "name": "John Doe",
     *     "preferredLanguage": "en-US"
     *   }
     * };
     *
     * const response = await calls.createCall(request);
     * const { ref, statusStream } = response;
     *
     * console.log(ref); // Call reference
     *
     * for await (const status of statusStream) {
     *  console.log(status); // Streamed status
     * }
     */
    async createCall(request) {
        const client = this.client.getCallsClient();
        const response = await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.createCall.bind(client),
            requestPBObjectConstructor: calls_pb_1.CreateCallRequest,
            metadata: this.client.getMetadata(),
            request: {
                ...request,
                metadata: struct_pb_1.Struct.fromJavaScript(request.metadata)
            }
        });
        const trackCallRequest = new calls_pb_1.TrackCallRequest();
        trackCallRequest.setRef(response.ref);
        const call = client.trackCall(trackCallRequest, this.client.getMetadata());
        async function* statusStreamGenerator() {
            const queue = [];
            let done = false;
            call.on("data", (response) => {
                const data = response.toObject();
                queue.push(data);
            });
            call.on("end", () => {
                done = true;
            });
            call.on("error", () => {
                done = true;
                throw new Error("An error occurred while tracking the call");
            });
            // eslint-disable-next-line no-loops/no-loops
            while (!done) {
                if (queue.length > 0) {
                    const data = queue.shift();
                    if (!data) {
                        return;
                    }
                    yield { status: (0, utils_1.dialStatusToString)(data.status) };
                }
                else {
                    await new Promise((resolve) => setTimeout(resolve, 50));
                }
            }
        }
        const statusStream = statusStreamGenerator();
        return { ref: response.ref, statusStream };
    }
    /**
     * Retrieves an existing Call in the Workspace.
     *
     * @param {string} ref - The reference of the Call to retrieve
     * @return {Promise<CallDetailRecord>} - The response object that contains the Call detail
     * @example
     * const calls = new SDK.Calls(client); // Existing client object
     *
     * const ref = "00000000-0000-0000-0000-000000000000";
     *
     * calls
     *   .getCall(ref)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async getCall(ref) {
        const client = this.client.getCallsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.getCall.bind(client),
            requestPBObjectConstructor: calls_pb_1.GetCallRequest,
            metadata: this.client.getMetadata(),
            request: { ref },
            enumMapping: [
                ["type", calls_pb_1.CallType],
                ["status", calls_pb_1.CallStatus],
                ["direction", calls_pb_1.CallDirection]
            ]
        });
    }
    /**
     * Retrieves a list of Calls from a Workspace.
     *
     * @param {ListCallsRequest} request - The request object that contains the necessary information to retrieve a list of Calls
     * @param {number} request.pageSize - The number of Calls to retrieve
     * @param {string} request.pageToken - The token to retrieve the next page of Calls
     * @return {Promise<ListCallsResponse>} - The response object that contains the list of Calls
     * @example
     * const calls = new SDK.Calls(client); // Existing client object
     *
     * const request = {
     *   pageSize: 10,
     *   pageToken: "00000000-0000-0000-0000-000000000000"
     * };
     *
     * calls
     *   .listCalls(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async listCalls(request) {
        const client = this.client.getCallsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.listCalls.bind(client),
            requestPBObjectConstructor: calls_pb_1.ListCallsRequest,
            metadata: this.client.getMetadata(),
            request,
            enumMapping: [
                ["type", calls_pb_1.CallType],
                ["status", calls_pb_1.CallStatus],
                ["direction", calls_pb_1.CallDirection]
            ],
            repeatableObjectMapping: [["itemsList", calls_pb_1.Call]]
        });
    }
}
exports.Calls = Calls;
