// @fonoster/sdk/dist/node/Credentials.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Credentials = void 0;
const makeRpcRequest_1 = require("./client/makeRpcRequest");
const credentials_pb_1 = require("./generated/node/credentials_pb");
/**
 * @classdesc Fonoster Credentials, part of the Fonoster SIP Proxy subsystem,
 * allows you to create, update, retrieve, and delete SIP Credentials for your deployment.
 * Note that an active Fonoster deployment is required.
 *
 * @example
 *
 * const SDK = require("@fonoster/sdk");
 *
 * async function main(request) {
 *   const apiKey = "your-api-key";
 *   const apiSecret = "your-api-secret"
 *   const accessKeyId = "WO00000000000000000000000000000000";
 *
 *   try {
 *     const client = SDK.Client({ accessKeyId });
 *     await client.loginWithApiKey(apiKey, apiSecret);
 *
 *     const credentials = new SDK.Credentials(client);
 *     const response = await apiKeys.createCredentials(request);
 *
 *     console.log(response); // successful response
 *   } catch (e) {
 *     console.error(e); // an error occurred
 *   }
 * }
 *
 * const request = {
 *   name: "My Credentials",
 *   username: "myusername",
 *   password: "mysecret"
 * };
 *
 * main(request);
 */
class Credentials {
    client;
    /**
     * Constructs a new Credentials object.
     *
     * @param {FonosterClient} client - Client object with underlying implementations to make requests to Fonoster's API
     * @see AbstractClient
     * @see FonosterClient
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Creates a new set of Credentials in the Workspace.
     *
     * @param {CreateCredentialsRequest} request - The request object that contains the necessary information to create a new set of Credentials
     * @param {string} request.name - The name of the Credentials
     * @param {string} request.username - The username of the Credentials
     * @param {string} request.password - The password of the Credentials
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the created Credentials
     * @example
     * const credentials = new SDK.Credentials(client); // Existing client object
     *
     * const request = {
     *   name: "My Credentials",
     *   username: "myusername",
     *   password: "mysecret"
     * };
     *
     * credentials
     *   .createCredentials(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async createCredentials(request) {
        const client = this.client.getCredentialsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.createCredentials.bind(client),
            requestPBObjectConstructor: credentials_pb_1.CreateCredentialsRequest,
            metadata: this.client.getMetadata(),
            request
        });
    }
    /**
     * Retrieves an existing set of Credentials in the Workspace.
     *
     * @param {string} ref - The reference of the Credentials to retrieve
     * @return {Promise<Acl>} - The response object that contains the Credentials
     * @example
     * const credentials = new SDK.Credentials(client); // Existing client object
     *
     * const ref = "00000000-0000-0000-0000-000000000000";
     *
     * credentials
     *   .getCredentials(ref)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async getCredentials(ref) {
        const client = this.client.getCredentialsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.getCredentials.bind(client),
            requestPBObjectConstructor: credentials_pb_1.GetCredentialsRequest,
            metadata: this.client.getMetadata(),
            request: { ref }
        });
    }
    /**
     * Updates an existing set of Credentials in the Workspace.
     *
     * @param {UpdateCredentialsRequest} request - The request object that contains the necessary information to update an existing set of Credentials
     * @param {string} request.ref - The reference of the Credentials to update
     * @param {string} request.name - The name of the Credentials
     * @param {string} request.password - The password of the Credentials
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the updated Credentials
     * @example
     * const credentials = new SDK.Credentials(client); // Existing client object
     *
     * const request = {
     *   ref: "00000000-0000-0000-0000-000000000000",
     *   name: "My Credentials",
     *   password: "mysecret"
     * };
     *
     * credentials
     *    .updateCredentials(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async updateCredentials(request) {
        const client = this.client.getCredentialsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.updateCredentials.bind(client),
            requestPBObjectConstructor: credentials_pb_1.UpdateCredentialsRequest,
            metadata: this.client.getMetadata(),
            request
        });
    }
    /**
     * Retrieves a list of Credentials from a Workspace.
     *
     * @param {ListCredentialsRequest} request - The request object that contains the necessary information to retrieve a list of Credentials
     * @param {number} request.pageSize - The number of Credentials to retrieve
     * @param {string} request.pageToken - The token to retrieve the next page of Credentials
     * @return {Promise<ListCredentialsResponse>} - The response object that contains the list of Credentials
     * @example
     * const credentials = new SDK.Credentials(client); // Existing client object
     *
     * const request = {
     *   pageSize: 10,
     *   pageToken: "00000000-0000-0000-0000-000000000000"
     * };
     *
     * credentials
     *   .listCredentials(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async listCredentials(request) {
        const client = this.client.getCredentialsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.listCredentials.bind(client),
            requestPBObjectConstructor: credentials_pb_1.ListCredentialsRequest,
            metadata: this.client.getMetadata(),
            request,
            repeatableObjectMapping: [["itemsList", credentials_pb_1.Credentials]]
        });
    }
    /**
     * Deletes an existing set of Credentials from Fonoster.
     * Note that this operation is irreversible.
     *
     * @param {string} ref - The reference of the Credentials to delete
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the deleted Credentials
     * @example
     * const credentials = new SDK.Credentials(client); // Existing client object
     *
     * const ref = "00000000-0000-0000-0000-000000000000";
     *
     * credentials
     *   .deleteCredentials(ref)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async deleteCredentials(ref) {
        const applicationsClient = this.client.getCredentialsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: applicationsClient.deleteCredentials.bind(applicationsClient),
            requestPBObjectConstructor: credentials_pb_1.DeleteCredentialsRequest,
            metadata: this.client.getMetadata(),
            request: { ref }
        });
    }
}
exports.Credentials = Credentials;
