// @fonoster/sdk/dist/node/Secrets.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Secrets = void 0;
const makeRpcRequest_1 = require("./client/makeRpcRequest");
const secrets_pb_1 = require("./generated/node/secrets_pb");
/**
 * @classdesc Fonoster Secrets, part of the Fonoster Core,
 * allows you to create, update, retrieve, and delete Secrets for your deployment.
 * Note that an active Fonoster deployment is required.
 *
 * @example
 *
 * const SDK = require("@fonoster/sdk");
 *
 * async function main(request) {
 *   const apiKey = "your-api-key";
 *   const apiSecret = "your-api-secret"
 *   const accessKeyId = "WO00000000000000000000000000000000";
 *
 *   try {
 *     const client = SDK.Client({ accessKeyId });
 *     await client.loginWithApiKey(apiKey, apiSecret);
 *
 *     const secrets = new SDK.Secrets(client);
 *     const response = await secrets.creteSecret(request);
 *
 *     console.log(response); // successful response
 *   } catch (e) {
 *     console.error(e); // an error occurred
 *   }
 * }
 *
 * const request = {
 *   name: "FRIENDLY_NAME",
 *   secret: "mysecret"
 * };
 *
 * main(request);
 */
class Secrets {
    client;
    /**
     * Constructs a new Secrets object.
     *
     * @param {FonosterClient} client - Client object with underlying implementations to make requests to Fonoster's API
     * @see AbstractClient
     * @see FonosterClient
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Creates a new Secret in the Workspace.
     *
     * @param {CreateSecretRequest} request - The request object that contains the necessary information to create a new Secret
     * @param {string} request.name - The name of the Secret
     * @param {string} request.secret - The secret of the Secret
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the created Secret
     * @example
     * const secrets = new SDK.Secrets(client); // Existing client object
     *
     * const request = {
     *   name: "FRIENDLY_NAME",
     *   secret: "mysecret"
     * };
     *
     * secrets
     *   .createSecret(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async createSecret(request) {
        const client = this.client.getSecretsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.createSecret.bind(client),
            requestPBObjectConstructor: secrets_pb_1.CreateSecretRequest,
            metadata: this.client.getMetadata(),
            request
        });
    }
    /**
     * Retrieves an existing Secret in the Workspace.
     *
     * @param {string} ref - The reference of the Secret to retrieve
     * @return {Promise<Acl>} - The response object that contains the Secret
     * @example
     * const secrets = new SDK.Secrets(client); // Existing client object
     *
     * const ref = "00000000-0000-0000-0000-000000000000";
     *
     * secrets
     *   .getSecret(ref)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async getSecret(ref) {
        const client = this.client.getSecretsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.getSecret.bind(client),
            requestPBObjectConstructor: secrets_pb_1.GetSecretRequest,
            metadata: this.client.getMetadata(),
            request: { ref }
        });
    }
    /**
     * Updates an existing Secret in the Workspace.
     *
     * @param {UpdateSecretRequest} request - The request object that contains the necessary information to update an existing Secret
     * @param {string} request.ref - The reference of the Secret to update
     * @param {string} request.name - The name of the Secret
     * @param {string} request.secret - The secret of the Secret
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the updated Secret
     * @example
     * const secrets = new SDK.Secrets(client); // Existing client object
     *
     * const request = {
     *   ref: "00000000-0000-0000-0000-000000000000",
     *   secret: "mysecret"
     * };
     *
     * secrets
     *   .updateSecret(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async updateSecret(request) {
        const client = this.client.getSecretsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.updateSecret.bind(client),
            requestPBObjectConstructor: secrets_pb_1.UpdateSecretRequest,
            metadata: this.client.getMetadata(),
            request
        });
    }
    /**
     * Retrieves a list of Secrets from a Workspace.
     *
     * @param {ListSecretsRequest} request - The request object that contains the necessary information to retrieve a list of Secrets
     * @param {number} request.pageSize - The secret of Secrets to retrieve
     * @param {string} request.pageToken - The token to retrieve the next page of Secrets
     * @return {Promise<ListSecretsResponse>} - The response object that contains the list of Secrets
     * @example
     * const secrets = new SDK.Secrets(client); // Existing client object
     *
     * const request = {
     *   pageSize: 10,
     *   pageToken: "00000000-0000-0000-0000-000000000000"
     * };
     *
     * secrets
     *   .listSecrets(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async listSecrets(request) {
        const client = this.client.getSecretsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.listSecrets.bind(client),
            requestPBObjectConstructor: secrets_pb_1.ListSecretsRequest,
            metadata: this.client.getMetadata(),
            request,
            repeatableObjectMapping: [["itemsList", secrets_pb_1.Secret]]
        });
    }
    /**
     * Deletes an existing Secret from Fonoster.
     * Note that this operation is irreversible.
     *
     * @param {string} ref - The reference of the Secret to delete
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the deleted Secret
     * @example
     * const secrets = new SDK.Secrets(client); // Existing client object
     *
     * const ref = "00000000-0000-0000-0000-000000000000";
     *
     * secrets
     *   .deleteSecret(ref)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async deleteSecret(ref) {
        const applicationsClient = this.client.getSecretsClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: applicationsClient.deleteSecret.bind(applicationsClient),
            requestPBObjectConstructor: secrets_pb_1.DeleteSecretRequest,
            metadata: this.client.getMetadata(),
            request: { ref }
        });
    }
}
exports.Secrets = Secrets;
