// @fonoster/sdk/dist/node/Trunks.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Trunks = void 0;
const makeRpcRequest_1 = require("./client/makeRpcRequest");
const trunks_pb_1 = require("./generated/node/trunks_pb");
/**
 * @classdesc Fonoster Trunks, part of the Fonoster SIP Proxy subsystem,
 * allows you to create, update, retrieve, and delete SIP Trunks for your deployment.
 * Note that an active Fonoster deployment is required.
 *
 * @example
 *
 * const SDK = require("@fonoster/sdk");
 *
 * async function main(request) {
 *   const apiKey = "your-api-key";
 *   const apiSecret = "your-api-secret"
 *   const accessKeyId = "WO00000000000000000000000000000000";
 *
 *   try {
 *     const client = SDK.Client({ accessKeyId });
 *     await client.loginWithApiKey(apiKey, apiSecret);
 *
 *     const trunks = new SDK.Trunks(client);
 *     const response = await trunks.createTrunk(request);
 *
 *     console.log(response); // successful response
 *   } catch (e) {
 *     console.error(e); // an error occurred
 *   }
 * }
 *
 * const request = {
 *   name: "My Trunk",
 *   inboundUri: "sip.company.fonoster.io"
 * };
 *
 * main(request);
 */
class Trunks {
    client;
    /**
     * Constructs a new Trunks object.
     *
     * @param {FonosterClient} client - Client object with underlying implementations to make requests to Fonoster's API
     * @see AbstractClient
     * @see FonosterClient
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Creates a new Trunk in the Workspace.
     *
     * @param {CreateTrunkRequest} request - The request object that contains the necessary information to create a new Trunk
     * @param {string} request.name - The name of the Trunk
     * @param {string} request.inboundUri - The inboundUri of the Trunk
     * @param {boolean} request.sendRegister - The sendRegister of the Trunk
     * @param {string} request.accessControlListRef - The accessControlListRef of the Trunk
     * @param {string} request.inboundCredentialsRef - The inboundCredentialsRef of the Trunk
     * @param {string} request.outboundCredentialsRef - The outboundCredentialsRef of the Trunk
     * @param {TrunkUri[]} request.uris - The uris of the Trunk
     * @param {string} request.uris[].host - The host of the Trunk
     * @param {number} request.uris[].port - The port of the Trunk
     * @param {Transport} request.uris[].transport - The transport of the Trunk
     * @param {string} request.uris[].user - Optional user of the Trunk
     * @param {number} request.uris[].weight - Optional weight of the Trunk
     * @param {number} request.uris[].priority - Optional priority of the Trunk
     * @param {boolean} request.uris[].enabled - Optional enabled of the Trunk
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the created Trunk
     * @example
     * const trunks = new SDK.Trunks(client); // Existing client object
     *
     * const request = {
     *   name: "My Trunk",
     *   inboundUri: "sip.company.fonoster.io"
     *   sendRegister: true
     *   uris: [{
     *     host: "sip.company.fonoster.io",
     *     port: 5060,
     *     transport: "UDP",
     *     user: "user",
     *     weight: 0,
     *     priority: 0,
     *     enabled: true
     *   }]
     * };
     *
     * trunks
     *   .createTrunk(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async createTrunk(request) {
        const client = this.client.getTrunksClient();
        const createTrunkRequest = new trunks_pb_1.CreateTrunkRequest();
        createTrunkRequest.setName(request.name);
        createTrunkRequest.setInboundUri(request.inboundUri);
        createTrunkRequest.setSendRegister(request.sendRegister);
        createTrunkRequest.setAccessControlListRef(request.accessControlListRef);
        createTrunkRequest.setInboundCredentialsRef(request.inboundCredentialsRef);
        createTrunkRequest.setOutboundCredentialsRef(request.outboundCredentialsRef);
        request.uris?.forEach((uri) => {
            const current = new trunks_pb_1.TrunkURI();
            current.setHost(uri.host);
            current.setPort(uri.port);
            current.setTransport(uri.transport);
            current.setUser(uri.user);
            current.setWeight(uri.weight);
            current.setPriority(uri.priority);
            current.setEnabled(uri.enabled);
            createTrunkRequest.addUris(current);
        });
        return new Promise((resolve, reject) => {
            client.createTrunk(createTrunkRequest, this.client.getMetadata(), (err, response) => {
                if (err) {
                    reject(err);
                    return;
                }
                resolve(response.toObject());
            });
        });
    }
    /**
     * Retrieves an existing Trunk in the Workspace.
     *
     * @param {string} ref - The reference of the Trunk to retrieve
     * @return {Promise<Acl>} - The response object that contains the Trunk
     * @example
     * const trunks = new SDK.Trunks(client); // Existing client object
     *
     * const ref = "00000000-0000-0000-0000-000000000000";
     *
     * trunks
     *   .getTrunk(ref)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async getTrunk(ref) {
        const client = this.client.getTrunksClient();
        const getTrunkRequest = new trunks_pb_1.GetTrunkRequest();
        getTrunkRequest.setRef(ref);
        return new Promise((resolve, reject) => {
            client.getTrunk(getTrunkRequest, this.client.getMetadata(), (err, response) => {
                if (err) {
                    reject(err);
                    return;
                }
                const obj = response.toObject();
                const outObj = {
                    ...obj,
                    accessControlListRef: obj.accessControlList?.ref,
                    inboundCredentialsRef: obj.inboundCredentials?.ref,
                    outboundCredentialsRef: obj.outboundCredentials?.ref,
                    uris: obj.urisList,
                    createdAt: new Date(obj.createdAt * 1000),
                    updatedAt: new Date(obj.updatedAt * 1000)
                };
                const { urisList, accessControlList, inboundCredentials, outboundCredentials, ...rest } = outObj;
                resolve(rest);
            });
        });
    }
    /**
     * Updates an existing Trunk in the Workspace.
     *
     * @param {UpdateTrunkRequest} request - The request object that contains the necessary information to update an existing Trunk
     * @param {string} request.ref - The reference of the Trunk to update
     * @param {string} request.name - The name of the Trunk
     * @param {boolean} request.sendRegister - The sendRegister of the Trunk
     * @param {string} request.accessControlListRef - The accessControlListRef of the Trunk
     * @param {string} request.inboundCredentialsRef - The inboundCredentialsRef of the Trunk
     * @param {string} request.outboundCredentialsRef - The outboundCredentialsRef of the Trunk
     * @param {TrunkUri[]} request.uris - The uris of the Trunk
     * @param {string} request.uris[].host - The host of the Trunk
     * @param {number} request.uris[].port - The port of the Trunk
     * @param {Transport} request.uris[].transport - The transport of the Trunk
     * @param {string} request.uris[].user - Optional user of the Trunk
     * @param {number} request.uris[].weight - Optional weight of the Trunk
     * @param {number} request.uris[].priority - Optional priority of the Trunk
     * @param {boolean} request.uris[].enabled - Optional enabled of the Trunk
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the updated Trunk
     * @example
     * const trunks = new SDK.Trunks(client); // Existing client object
     *
     * const request = {
     *   ref: "00000000-0000-0000-0000-000000000000",
     *   name: "My Trunk",
     *   sendRegister: true
     *   uris: [{
     *     host: "sip.company.fonoster.io",
     *     port: 5060,
     *     transport: "UDP",
     *     user: "user",
     *     weight: 0,
     *     priority: 0,
     *     enabled: true
     *   }]
     * };
     *
     * trunks
     *   .updateTrunk(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async updateTrunk(request) {
        const client = this.client.getTrunksClient();
        const updateTrunkRequest = new trunks_pb_1.UpdateTrunkRequest();
        updateTrunkRequest.setRef(request.ref);
        updateTrunkRequest.setName(request.name);
        updateTrunkRequest.setInboundUri(request.inboundUri);
        updateTrunkRequest.setSendRegister(request.sendRegister);
        updateTrunkRequest.setAccessControlListRef(request.accessControlListRef);
        updateTrunkRequest.setInboundCredentialsRef(request.inboundCredentialsRef);
        updateTrunkRequest.setOutboundCredentialsRef(request.outboundCredentialsRef);
        request.uris?.forEach((uri) => {
            const current = new trunks_pb_1.TrunkURI();
            current.setHost(uri.host);
            current.setPort(uri.port);
            current.setTransport(uri.transport);
            current.setUser(uri.user);
            current.setWeight(uri.weight);
            current.setPriority(uri.priority);
            current.setEnabled(uri.enabled);
            updateTrunkRequest.addUris(current);
        });
        return new Promise((resolve, reject) => {
            client.updateTrunk(updateTrunkRequest, this.client.getMetadata(), (err, response) => {
                if (err) {
                    reject(err);
                    return;
                }
                resolve(response.toObject());
            });
        });
    }
    /**
     * Retrieves a list of Trunks from a Workspace.
     *
     * @param {ListTrunksRequest} request - The request object that contains the necessary information to retrieve a list of Trunks
     * @param {number} request.pageSize - The trunk of Trunks to retrieve
     * @param {string} request.pageToken - The token to retrieve the next page of Trunks
     * @return {Promise<ListTrunksResponse>} - The response object that contains the list of Trunks
     * @example
     * const trunks = new SDK.Trunks(client); // Existing client object
     *
     * const request = {
     *   pageSize: 10,
     *   pageToken: "00000000-0000-0000-0000-000000000000"
     * };
     *
     * trunks
     *   .listTrunks(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async listTrunks(request) {
        const client = this.client.getTrunksClient();
        const listTrunksRequest = new trunks_pb_1.ListTrunksRequest();
        listTrunksRequest.setPageSize(request.pageSize);
        listTrunksRequest.setPageToken(request.pageToken);
        return new Promise((resolve, reject) => {
            client.listTrunks(listTrunksRequest, this.client.getMetadata(), (err, response) => {
                if (err) {
                    reject(err);
                    return;
                }
                const items = response.getItemsList().map((item) => {
                    const obj = item.toObject();
                    const outObj = {
                        ...obj,
                        uris: obj.urisList,
                        createdAt: new Date(obj.createdAt * 1000),
                        updatedAt: new Date(obj.updatedAt * 1000)
                    };
                    // eslint-disable-next-line @typescript-eslint/no-unused-vars
                    const { urisList, ...rest } = outObj;
                    return rest;
                });
                resolve({
                    items: items,
                    nextPageToken: response.getNextPageToken()
                });
            });
        });
    }
    /**
     * Deletes an existing Trunk from Fonoster.
     * Note that this operation is irreversible.
     *
     * @param {string} ref - The reference of the Trunk to delete
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the deleted Trunk
     * @example
     * const trunks = new SDK.Trunks(client); // Existing client object
     *
     * const ref = "00000000-0000-0000-0000-000000000000";
     *
     * trunks
     *    .deleteTrunk(ref)
     *    .then(console.log) // successful response
     *    .catch(console.error); // an error occurred
     */
    async deleteTrunk(ref) {
        const applicationsClient = this.client.getTrunksClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: applicationsClient.deleteTrunk.bind(applicationsClient),
            requestPBObjectConstructor: trunks_pb_1.DeleteTrunkRequest,
            metadata: this.client.getMetadata(),
            request: { ref }
        });
    }
}
exports.Trunks = Trunks;
