// @fonoster/sdk/dist/node/Users.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Users = void 0;
const makeRpcRequest_1 = require("./client/makeRpcRequest");
const identity_pb_1 = require("./generated/node/identity_pb");
/**
 * @classdesc Fonoster Users, part of the Fonoster Identity subsystem,
 * allows you to create, update, retrieve, and delete a Users in the system.
 * Note that an active Fonoster deployment is required.
 *
 * @example
 * const SDK = require("@fonoster/sdk");
 *
 * const client = SDK.Client();
 * const users = new SDK.Users(client);
 *
 * const request = {
 *   name: "John Doe",
 *   email: "john.doe@example.com",
 *   password: "password",
 *   avatar: "https://example.com/avatar.jpg"
 * };
 *
 * users.createUser(request)
 *   .then(console.log) // successful response
 *   .catch(console.error); // an error occurred
 */
class Users {
    client;
    /**
     * Constructs a new Users object.
     *
     * @param {FonosterClient} client - Client object with underlying implementations to make requests to Fonoster's API
     * @see AbstractClient
     * @see FonosterClient
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Creates a new User in the Workspace.
     *
     * @param {CreateUserRequest} request - The request object that contains the necessary information to create a new User
     * @param {string} request.name - The name of the User
     * @param {string} request.email - The email of the User
     * @param {string} request.password - The password of the User
     * @param {string} request.avatar - The avatar of the User
     * @param {string} [request.phone] - The phone number of the User (optional)
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the created User
     * @example
     * const users = new SDK.Users(client); // Existing client object
     *
     * const request = {
     *   name: "John Doe",
     *   email: "john.doe@example.com",
     *   password: "password",
     *   avatar: "https://example.com/avatar.jpg",
     *   phone: "+1234567890"
     * };
     *
     * users
     *   .createUser(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async createUser(request) {
        const client = this.client.getIdentityClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.createUser.bind(client),
            requestPBObjectConstructor: identity_pb_1.CreateUserRequest,
            metadata: this.client.getMetadata(),
            request
        });
    }
    /**
     * Create a new User using an OAuth2 code and return the id, access, and refresh tokens for the User.
     *
     * @param {CreateUserWithOauth2CodeRequest} request - The request object with the OAuth2 code
     * @param {string} request.code - The OAuth2 code of the User
     * @return {Promise<ExchangeCredentialsResponse>} - The response object that contains the id, access, and refresh tokens
     * @example
     * const users = new SDK.Users(client); // Existing client object
     *
     * const request = {
     *   code: "fd4d78beb31aa25b93de"
     * };
     *
     * users.createUserWithOauth2Code(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async createUserWithOauth2Code(request) {
        const client = this.client.getIdentityClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.createUserWithOauth2Code.bind(client),
            requestPBObjectConstructor: identity_pb_1.CreateUserWithOauth2CodeRequest,
            metadata: this.client.getMetadata(),
            request
        });
    }
    /**
     * Retrieves an existing User in the Workspace.
     *
     * @param {string} ref - The reference of the User to retrieve
     * @return {Promise<Acl>} - The response object that contains the User
     * @example
     * const users = new SDK.Users(client); // Existing client object
     *
     * const ref = "00000000-0000-0000-0000-000000000000";
     *
     * users
     *   .getUser(ref)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async getUser(ref) {
        const client = this.client.getIdentityClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.getUser.bind(client),
            requestPBObjectConstructor: identity_pb_1.GetUserRequest,
            metadata: this.client.getMetadata(),
            request: { ref }
        });
    }
    /**
     * Updates an existing User in the Workspace.
     *
     * @param {UpdateUserRequest} request - The request object that contains the necessary information to update a User
     * @param {string} request.ref - The reference of the User to update
     * @param {string} [request.name] - The name of the User
     * @param {string} [request.password] - The password of the User
     * @param {string} [request.avatar] - The avatar of the User
     * @param {string} [request.phone] - The phone number of the User (optional)
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the updated User
     * @example
     * const users = new SDK.Users(client); // Existing client object
     *
     * const request = {
     *   ref: "00000000-0000-0000-0000-000000000000",
     *   name: "John Doe",
     *   password: "password",
     *   avatar: "https://example.com/avatar.jpg",
     *   phone: "+1234567890"
     * };
     *
     * users
     *   .updateUser(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async updateUser(request) {
        const client = this.client.getIdentityClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.updateUser.bind(client),
            requestPBObjectConstructor: identity_pb_1.UpdateUserRequest,
            metadata: this.client.getMetadata(),
            request
        });
    }
    /**
     * Sends a reset password code to the User.
     *
     * @param {SendResetPasswordCodeRequest} request - The request object that contains the necessary information to send a reset password code to a User
     * @param {string} request.username - The username of the User
     * @param {string} request.resetPasswordUrl - The URL to reset the password
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the User
     * @example
     * const users = new SDK.Users(client); // Existing client object
     *
     * const request = {
     *   username: "john.doe@example.com",
     *   resetPasswordUrl: "https://example.com/reset-password"
     * };
     *
     * users
     *   .sendResetPasswordCode(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async sendResetPasswordCode(request) {
        const client = this.client.getIdentityClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.sendResetPasswordCode.bind(client),
            requestPBObjectConstructor: identity_pb_1.SendResetPasswordCodeRequest,
            metadata: this.client.getMetadata(),
            request
        });
    }
    /**
     * Resets the password of the User.
     *
     * @param {ResetPasswordRequest} request - The request object that contains the necessary information to reset the password of a User
     * @param {string} request.username - The username of the User
     * @param {string} request.password - The new password of the User
     * @param {string} request.verificationCode - The verification code of the User
     * @return {Promise<void>} - The response object that contains the reference to the User
     * @example
     * const users = new SDK.Users(client); // Existing client object
     *
     * const request = {
     *   username: "john.doe@example.com",
     *   password: "password",
     *   verificationCode: "123456"
     * };
     *
     * users
     *   .resetPassword(request)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async resetPassword(request) {
        const client = this.client.getIdentityClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.resetPassword.bind(client),
            requestPBObjectConstructor: identity_pb_1.ResetPasswordRequest,
            metadata: this.client.getMetadata(),
            request
        });
    }
    /**
     * Deletes an existing User from Fonoster.
     * Note that this operation is irreversible.
     *
     * @param {string} ref - The reference of the User to delete
     * @return {Promise<BaseApiObject>} - The response object that contains the reference to the deleted User
     * @example
     * const users = new SDK.Users(client); // Existing client object
     *
     * const ref = "00000000-0000-0000-0000-000000000000";
     *
     * users
     *   .deleteUser(ref)
     *   .then(console.log) // successful response
     *   .catch(console.error); // an error occurred
     */
    async deleteUser(ref) {
        const client = this.client.getIdentityClient();
        return await (0, makeRpcRequest_1.makeRpcRequest)({
            method: client.deleteUser.bind(client),
            requestPBObjectConstructor: identity_pb_1.DeleteUserRequest,
            metadata: this.client.getMetadata(),
            request: { ref }
        });
    }
}
exports.Users = Users;
