// @fonoster/sdk/dist/node/utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.buildStructOverride = buildStructOverride;
exports.buildStructOverrideReverse = buildStructOverrideReverse;
exports.dialStatusToString = dialStatusToString;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const struct_pb_1 = require("google-protobuf/google/protobuf/struct_pb");
const calls_pb_1 = require("./generated/web/calls_pb");
function buildStructOverride(request) {
    return {
        ...request,
        textToSpeech: request.textToSpeech
            ? {
                ...request.textToSpeech,
                config: request.textToSpeech.config
                    ? struct_pb_1.Struct.fromJavaScript(request.textToSpeech.config)
                    : undefined
            }
            : undefined,
        speechToText: request.speechToText
            ? {
                ...request.speechToText,
                config: request.speechToText.config
                    ? struct_pb_1.Struct.fromJavaScript(request.speechToText.config)
                    : undefined
            }
            : undefined,
        intelligence: request.intelligence
            ? {
                ...request.intelligence,
                credentials: request.intelligence.credentials
                    ? struct_pb_1.Struct.fromJavaScript(request.intelligence.credentials)
                    : undefined,
                config: request.intelligence.config
                    ? struct_pb_1.Struct.fromJavaScript(request.intelligence.config)
                    : undefined
            }
            : undefined
    };
}
function buildStructOverrideReverse(request) {
    const result = {
        ...request,
        textToSpeech: request.textToSpeech
            ? {
                ...request.textToSpeech,
                config: request.textToSpeech.config
                    ? request.textToSpeech.config.toJavaScript()
                    : undefined
            }
            : undefined,
        speechToText: request.speechToText
            ? {
                ...request.speechToText,
                config: request.speechToText.config
                    ? request.speechToText.config.toJavaScript()
                    : undefined
            }
            : undefined,
        intelligence: request.intelligence
            ? {
                ...request.intelligence,
                credentials: request.intelligence.credentials
                    ? request.intelligence.credentials.toJavaScript()
                    : undefined,
                config: request.intelligence.config
                    ? request.intelligence.config.toJavaScript()
                    : undefined
            }
            : undefined
    };
    Object.keys(result).forEach((key) => result[key] === undefined && delete result[key]);
    return result;
}
function dialStatusToString(status) {
    switch (status) {
        case calls_pb_1.TrackCallResponse.Status.TRYING:
            return "TRYING";
        case calls_pb_1.TrackCallResponse.Status.ANSWER:
            return "ANSWER";
        case calls_pb_1.TrackCallResponse.Status.BUSY:
            return "BUSY";
        case calls_pb_1.TrackCallResponse.Status.CANCEL:
            return "CANCEL";
        case calls_pb_1.TrackCallResponse.Status.NOANSWER:
            return "ERROR";
        case calls_pb_1.TrackCallResponse.Status.FAILED:
            return "FAILED";
        case calls_pb_1.TrackCallResponse.Status.PROGRESS:
            return "PROGRESS";
    }
}
