// @fonoster/common/dist/validators/sipnet/agents.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateAgentRequestSchema = exports.createAgentRequestSchema = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const zod_1 = require("zod");
const messages_1 = require("../../messages");
const common_1 = require("../common");
const createAgentRequestSchema = zod_1.z.object({
    name: common_1.nameSchema,
    username: common_1.usernameSchema,
    privacy: zod_1.z.enum(["PRIVATE", "NONE"]).optional(),
    enabled: zod_1.z.boolean().optional(),
    maxContacts: zod_1.z
        .number()
        .positive({ message: messages_1.POSITIVE_INTEGER_MESSAGE })
        .optional(),
    expires: zod_1.z.number().positive({ message: messages_1.POSITIVE_INTEGER_MESSAGE }).optional()
});
exports.createAgentRequestSchema = createAgentRequestSchema;
const updateAgentRequestSchema = zod_1.z.object({
    name: common_1.nameSchema.optional(),
    privacy: zod_1.z.enum(["PRIVATE", "NONE"]).optional(),
    enabled: zod_1.z.boolean().optional(),
    maxContacts: zod_1.z
        .number()
        .positive({ message: messages_1.POSITIVE_INTEGER_MESSAGE })
        .optional(),
    expires: zod_1.z.number().positive({ message: messages_1.POSITIVE_INTEGER_MESSAGE }).optional()
});
exports.updateAgentRequestSchema = updateAgentRequestSchema;
