// @fonoster/common/dist/validators/sipnet/domains.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateDomainRequestSchema = exports.createDomainRequestSchema = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const zod_1 = require("zod");
const envs_1 = require("../../envs");
const common_1 = require("../common");
const domainSchema = zod_1.z
    .string()
    .refine((domain) => {
    const domainRegex = /^[a-zA-Z0-9-]+(\.[a-zA-Z]{2,})+$/;
    return domainRegex.test(domain);
}, {
    message: `Invalid domain format. Should be like: "acme.${envs_1.ROOT_DOMAIN}"`
})
    .refine((domain) => {
    return domain.endsWith(envs_1.ROOT_DOMAIN);
}, {
    message: `Domain must end with "${envs_1.ROOT_DOMAIN}"`
});
const createDomainRequestSchema = zod_1.z.object({
    name: common_1.nameSchema,
    domainUri: domainSchema,
    egressPolicies: zod_1.z
        .array(zod_1.z.object({
        rule: zod_1.z.string()
    }))
        .optional()
});
exports.createDomainRequestSchema = createDomainRequestSchema;
const updateDomainRequestSchema = zod_1.z.object({
    name: common_1.nameSchema.optional(),
    egressPolicies: zod_1.z
        .array(zod_1.z.object({
        rule: zod_1.z.string()
    }))
        .optional()
});
exports.updateDomainRequestSchema = updateDomainRequestSchema;
