// @fonoster/common/dist/validators/sipnet/numbers.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateNumberRequestSchema = exports.createNumberRequestSchema = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const isISO31661Alpha2_1 = __importDefault(require("validator/lib/isISO31661Alpha2"));
const zod_1 = require("zod");
const common_1 = require("../common");
const sipUriRegex = /^sip:[a-zA-Z0-9_.-]+@[a-zA-Z0-9_.-]+$/;
const sipUriSchema = zod_1.z.string().regex(sipUriRegex, "Invalid SIP URI");
const countryIsoCodeSchema = zod_1.z.string().refine((val) => (0, isISO31661Alpha2_1.default)(val), {
    message: "Invalid country ISO code"
});
const createNumberRequestSchema = zod_1.z
    .object({
    name: common_1.nameSchema,
    telUrl: zod_1.z.string(),
    city: zod_1.z.string(),
    country: zod_1.z.string(),
    agentAor: sipUriSchema.optional(),
    appRef: zod_1.z.string().optional(),
    countryIsoCode: countryIsoCodeSchema
})
    .refine(({ agentAor, appRef }) => !(agentAor !== undefined && appRef !== undefined), {
    message: "You can only provide one of the following fields: 'agentAor' or 'appRef'"
});
exports.createNumberRequestSchema = createNumberRequestSchema;
const updateNumberRequestSchema = zod_1.z
    .object({
    name: common_1.nameSchema.optional(),
    agentAor: sipUriSchema.optional(),
    appRef: zod_1.z.string().optional()
})
    .refine(({ agentAor, appRef }) => !(agentAor !== undefined && appRef !== undefined), {
    message: "You can only provide one of the following fields: 'agentAor' or 'appRef'"
});
exports.updateNumberRequestSchema = updateNumberRequestSchema;
