// @fonoster/common/dist/validators/sipnet/trunks.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateTrunkRequestSchema = exports.createTrunkRequestSchema = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const types_1 = require("@fonoster/types");
const Validator = __importStar(require("validator"));
const zod_1 = require("zod");
const common_1 = require("../common");
const hostOrIPSchema = zod_1.z
    .string()
    .refine((host) => Validator.isIP(host, 4) || Validator.isFQDN(host), {
    message: "Must be a valid IP or FQDN"
});
const createTrunkRequestSchema = zod_1.z.object({
    name: common_1.nameSchema,
    sendRegister: zod_1.z.boolean(),
    inboundUri: hostOrIPSchema,
    uris: zod_1.z.array(zod_1.z.object({
        host: hostOrIPSchema,
        port: zod_1.z.number(),
        transport: zod_1.z.nativeEnum(types_1.Transport, { message: "Invalid transport" }),
        user: zod_1.z.string().optional(),
        weight: zod_1.z.number(),
        priority: zod_1.z.number(),
        enabled: zod_1.z.boolean()
    }))
});
exports.createTrunkRequestSchema = createTrunkRequestSchema;
const updateTrunkRequestSchema = zod_1.z.object({});
exports.updateTrunkRequestSchema = updateTrunkRequestSchema;
