// @fonoster/common/dist/utils/sendHttpRequest.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AllowedHttpMethod = void 0;
exports.sendHttpRequest = sendHttpRequest;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const url_1 = require("url");
const logger_1 = require("@fonoster/logger");
const zod_1 = require("zod");
const responseSchema = zod_1.z.object({
    result: zod_1.z.string()
});
const logger = (0, logger_1.getLogger)({ service: "common", filePath: __filename });
var AllowedHttpMethod;
(function (AllowedHttpMethod) {
    AllowedHttpMethod["GET"] = "get";
    AllowedHttpMethod["POST"] = "post";
})(AllowedHttpMethod || (exports.AllowedHttpMethod = AllowedHttpMethod = {}));
function sendHttpRequest(request) {
    return __awaiter(this, void 0, void 0, function* () {
        const { url, method, params, headers, waitForResponse } = request;
        const effectiveMethod = method.toLowerCase();
        let effectiveUrl = url;
        if (effectiveMethod === AllowedHttpMethod.GET && params) {
            const queryParams = new url_1.URLSearchParams();
            Object.entries(params).forEach(([key, value]) => {
                queryParams.append(key, String(value));
            });
            effectiveUrl = `${url}${url.includes("?") ? "&" : "?"}${queryParams.toString()}`;
        }
        const options = {
            method: effectiveMethod,
            headers: Object.assign(Object.assign({}, headers), { accept: "application/json", "content-type": "application/json" }),
            body: effectiveMethod === AllowedHttpMethod.POST
                ? JSON.stringify(params)
                : undefined
        };
        logger.silly(`sending request to ${effectiveUrl}`, {
            params,
            method: effectiveMethod
        });
        if (!waitForResponse && effectiveMethod === AllowedHttpMethod.POST) {
            setTimeout(() => fetch(effectiveUrl, options), 0);
            return { result: "success" };
        }
        else {
            const response = yield fetch(effectiveUrl, options);
            const data = yield response.json();
            try {
                return responseSchema.parse(data);
            }
            catch (error) {
                throw new Error(`Invalid response: expected ${JSON.stringify(responseSchema, null, 2)}, got ${JSON.stringify(data, null, 2)}`);
            }
        }
    });
}
