// @fonoster/logger/dist/envs.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.transports = exports.level = exports.format = exports.fluent = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const fluent_logger_1 = __importDefault(require("fluent-logger"));
const winston_1 = __importDefault(require("winston"));
const getEnv_1 = require("./getEnv");
const FluentTransport = fluent_logger_1.default.support.winstonTransport();
const LOGS_DRIVER_HOST = (0, getEnv_1.getEnv)("LOGS_DRIVER_HOST");
// Note: if LOGS_DRIVER_PORT is provided as a string in the environment, it will be used as-is.
const LOGS_DRIVER_PORT = (0, getEnv_1.getEnv)("LOGS_DRIVER_PORT", 24224);
const LOGS_OPT_TAG_PREFIX = (0, getEnv_1.getEnv)("LOGS_OPT_TAG_PREFIX", "fonoster-logs");
const LOGS_FORMAT = (0, getEnv_1.getEnv)("LOGS_FORMAT", "json");
const LOGS_LEVEL = (0, getEnv_1.getEnv)("LOGS_LEVEL", "info");
exports.level = LOGS_LEVEL;
const LOGS_TRANSPORT = (0, getEnv_1.getEnv)("LOGS_TRANSPORT", "console");
const fluent = new FluentTransport(`${LOGS_OPT_TAG_PREFIX}`, {
    host: LOGS_DRIVER_HOST,
    port: LOGS_DRIVER_PORT,
    timeout: 3.0,
    requireAckResponse: false
});
exports.fluent = fluent;
const format = LOGS_FORMAT === "json"
    ? winston_1.default.format.combine(winston_1.default.format.timestamp(), winston_1.default.format.json())
    : winston_1.default.format.combine(winston_1.default.format.colorize(), winston_1.default.format.simple());
exports.format = format;
const transports = LOGS_TRANSPORT === "fluent" ? [fluent] : [new winston_1.default.transports.Console()];
exports.transports = transports;
