// @fonoster/logger/dist/logger.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.unmute = exports.mute = exports.default = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const winston_1 = __importDefault(require("winston"));
const envs_1 = require("./envs");
const logger = winston_1.default.createLogger({
    levels: winston_1.default.config.npm.levels,
    format: envs_1.format,
    transports: envs_1.transports,
    level: envs_1.level
});
exports.default = logger;
logger.on("finish", () => {
    envs_1.fluent.sender.end("end", {}, () => { });
});
const mute = () => logger.transports.forEach((t) => (t.silent = true));
exports.mute = mute;
const unmute = () => logger.transports.forEach((t) => (t.silent = false));
exports.unmute = unmute;
