// @fonoster/types/dist/calls.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CallType = exports.CallStatus = exports.CallDirection = void 0;
var CallType;
(function (CallType) {
    CallType["SIP_ORIGINATED"] = "SIP_ORIGINATED";
    CallType["API_ORIGINATED"] = "API_ORIGINATED";
})(CallType || (exports.CallType = CallType = {}));
var CallStatus;
(function (CallStatus) {
    CallStatus["NORMAL_CLEARING"] = "NORMAL_CLEARING";
    CallStatus["CALL_REJECTED"] = "CALL_REJECTED";
    CallStatus["UNALLOCATED"] = "UNALLOCATED";
    CallStatus["NO_USER_RESPONSE"] = "NO_USER_RESPONSE";
    CallStatus["NO_ROUTE_DESTINATION"] = "NO_ROUTE_DESTINATION";
    CallStatus["NO_ANSWER"] = "NO_ANSWER";
    CallStatus["USER_BUSY"] = "USER_BUSY";
    CallStatus["NOT_ACCEPTABLE_HERE"] = "NOT_ACCEPTABLE_HERE";
    CallStatus["SERVICE_UNAVAILABLE"] = "SERVICE_UNAVAILABLE";
    CallStatus["INVALID_NUMBER_FORMAT"] = "INVALID_NUMBER_FORMAT";
})(CallStatus || (exports.CallStatus = CallStatus = {}));
var CallDirection;
(function (CallDirection) {
    CallDirection["FROM_PSTN"] = "FROM_PSTN";
    CallDirection["TO_PSTN"] = "TO_PSTN";
    CallDirection["INTRA_NETWORK"] = "INTRA_NETWORK";
})(CallDirection || (exports.CallDirection = CallDirection = {}));
