// @fonoster/common/dist/validators/calls.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.listCallsRequestSchema = exports.getCallRequestSchema = exports.createCallRequestSchema = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const types_1 = require("@fonoster/types");
const zod_1 = require("zod");
const messages_1 = require("../messages");
const createCallRequestSchema = zod_1.z.object({
    from: zod_1.z.string(),
    to: zod_1.z.string(),
    appRef: zod_1.z.string().uuid({ message: "Invalid call reference" }),
    timeout: zod_1.z
        .number()
        .max(120, { message: "Timeout must be less than 120s" })
        .default(30),
    metadata: zod_1.z.record(zod_1.z.string(), zod_1.z.any()).optional()
});
exports.createCallRequestSchema = createCallRequestSchema;
const getCallRequestSchema = zod_1.z.object({
    ref: zod_1.z.string().uuid({ message: "Invalid call reference" })
});
exports.getCallRequestSchema = getCallRequestSchema;
const listCallsRequestSchema = zod_1.z.object({
    after: zod_1.z.string().datetime({ offset: true, message: messages_1.VALID_DATE }).optional(),
    before: zod_1.z.string().datetime({ offset: true, message: messages_1.VALID_DATE }).optional(),
    pageSize: zod_1.z
        .number()
        .int({
        message: messages_1.POSITIVE_INTEGER_MESSAGE
    })
        .positive({
        message: messages_1.POSITIVE_INTEGER_MESSAGE
    })
        .optional(),
    type: zod_1.z
        .nativeEnum(types_1.CallType, {
        message: "Invalid call type"
    })
        .optional(),
    status: zod_1.z
        .nativeEnum(types_1.CallStatus, { message: "Invalid call status" })
        .optional(),
    pageToken: zod_1.z.string().optional()
});
exports.listCallsRequestSchema = listCallsRequestSchema;
