// @fonoster/common/dist/validators/common.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.usernameSchema = exports.nameSchema = exports.listRequestSchema = exports.emptySchema = exports.baseApiObjectSchema = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const zod_1 = require("zod");
const messages_1 = require("../messages");
const usernameSchema = zod_1.z.string().regex(/^[a-z0-9._-]+$/, {
    message: "Must be a lowercase string and with no spaces."
});
exports.usernameSchema = usernameSchema;
const baseApiObjectSchema = zod_1.z.object({
    ref: zod_1.z.string().uuid({
        message: messages_1.VALID_UUID
    })
});
exports.baseApiObjectSchema = baseApiObjectSchema;
const emptySchema = zod_1.z.object({});
exports.emptySchema = emptySchema;
const listRequestSchema = zod_1.z
    .object({
    pageSize: zod_1.z
        .number()
        .int({
        message: messages_1.POSITIVE_INTEGER_MESSAGE
    })
        .positive({
        message: messages_1.POSITIVE_INTEGER_MESSAGE
    })
        .optional(),
    pageToken: zod_1.z.string().optional()
})
    .optional();
exports.listRequestSchema = listRequestSchema;
const nameSchema = zod_1.z.string().min(1, { message: "Value required" });
exports.nameSchema = nameSchema;
