// @fonoster/common/dist/validators/identity.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createUserWithOauth2CodeRequestSchema = exports.resetPasswordRequestSchema = exports.sendResetPasswordCodeRequestSchema = exports.verifyCodeRequestSchema = exports.updateWorkspaceRequestSchema = exports.updateUserRequestSchema = exports.sendVerificationCodeRequestSchema = exports.resendWorkspaceMembershipInvitationRequestSchema = exports.removeUserFromWorkspaceRequestSchema = exports.inviteUserToWorkspaceRequestSchema = exports.exchangeRefreshTokenRequestSchema = exports.exchangeOauth2RequestSchema = exports.exchangeCredentialsRequestSchema = exports.exchangeApiKeysRequestSchema = exports.createWorkspaceRequestSchema = exports.createUserRequestSchema = exports.createApiKeyRequestSchema = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const types_1 = require("@fonoster/types");
const zod_1 = require("zod");
const messages_1 = require("../messages");
const MIN_NAME_MESSAGE = "The name is required";
const MAX_NAME_MESSAGE = "Name must contain at most 50 characters";
const EMAIL_MESSAGE = "Invalid email";
const PASSWORD_MESSAGE = "Password must contain at least 8 characters";
const USER_REF_MESSAGE = "Invalid user reference";
const WORKSPACE_REF_MESSAGE = "Invalid workspace reference";
const createWorkspaceRequestSchema = zod_1.z.object({
    name: zod_1.z
        .string()
        .min(1, { message: MIN_NAME_MESSAGE })
        .max(50, { message: MAX_NAME_MESSAGE })
});
exports.createWorkspaceRequestSchema = createWorkspaceRequestSchema;
const createApiKeyRequestSchema = zod_1.z.object({
    role: zod_1.z.enum([types_1.Role.WORKSPACE_ADMIN]),
    expiresAt: zod_1.z
        .number()
        .int({ message: messages_1.POSITIVE_INTEGER_MESSAGE })
        .positive({ message: messages_1.POSITIVE_INTEGER_MESSAGE })
        .optional()
});
exports.createApiKeyRequestSchema = createApiKeyRequestSchema;
const exchangeApiKeysRequestSchema = zod_1.z.object({
    accessKeyId: zod_1.z.string(),
    accessKeySecret: zod_1.z.string()
});
exports.exchangeApiKeysRequestSchema = exchangeApiKeysRequestSchema;
const exchangeCredentialsRequestSchema = zod_1.z.object({
    username: zod_1.z
        .string()
        .email({ message: "Invalid username. Must be an email address" }),
    password: zod_1.z.string(),
    twoFactorCode: zod_1.z.string().optional()
});
exports.exchangeCredentialsRequestSchema = exchangeCredentialsRequestSchema;
const exchangeOauth2RequestSchema = zod_1.z.object({
    provider: zod_1.z.enum(["GITHUB"]).default("GITHUB"),
    code: zod_1.z.string()
});
exports.exchangeOauth2RequestSchema = exchangeOauth2RequestSchema;
const exchangeRefreshTokenRequestSchema = zod_1.z.object({
    refreshToken: zod_1.z.string()
});
exports.exchangeRefreshTokenRequestSchema = exchangeRefreshTokenRequestSchema;
const createUserRequestSchema = zod_1.z.object({
    name: zod_1.z.string().max(50, { message: MAX_NAME_MESSAGE }),
    email: zod_1.z.string().email({ message: EMAIL_MESSAGE }),
    phone: zod_1.z.string().max(20).optional(),
    password: zod_1.z.string().min(8, { message: PASSWORD_MESSAGE }).max(100),
    avatar: zod_1.z
        .string()
        .url()
        .max(255, { message: "Invalid avatar URL" })
        .optional()
});
exports.createUserRequestSchema = createUserRequestSchema;
const createUserWithOauth2CodeRequestSchema = zod_1.z.object({
    code: zod_1.z.string()
});
exports.createUserWithOauth2CodeRequestSchema = createUserWithOauth2CodeRequestSchema;
const updateUserRequestSchema = zod_1.z.object({
    ref: zod_1.z.string().uuid({ message: USER_REF_MESSAGE }),
    name: zod_1.z
        .string()
        .max(50, { message: MAX_NAME_MESSAGE })
        .or(zod_1.z.string().optional()),
    phone: zod_1.z.string().max(20).optional(),
    password: zod_1.z
        .string()
        .min(8, { message: PASSWORD_MESSAGE })
        .or(zod_1.z.string().optional()),
    avatar: zod_1.z.string().url().or(zod_1.z.string().optional())
});
exports.updateUserRequestSchema = updateUserRequestSchema;
const inviteUserToWorkspaceRequestSchema = zod_1.z.object({
    email: zod_1.z.string().email({ message: EMAIL_MESSAGE }),
    name: zod_1.z.string().max(50, { message: MAX_NAME_MESSAGE }),
    role: zod_1.z.enum([types_1.Role.WORKSPACE_ADMIN, types_1.Role.WORKSPACE_MEMBER]),
    password: zod_1.z.string().min(8, { message: PASSWORD_MESSAGE }).or(zod_1.z.undefined())
});
exports.inviteUserToWorkspaceRequestSchema = inviteUserToWorkspaceRequestSchema;
const resendWorkspaceMembershipInvitationRequestSchema = zod_1.z.object({
    userRef: zod_1.z.string().uuid({ message: USER_REF_MESSAGE })
});
exports.resendWorkspaceMembershipInvitationRequestSchema = resendWorkspaceMembershipInvitationRequestSchema;
const updateWorkspaceRequestSchema = zod_1.z.object({
    ref: zod_1.z.string().uuid({ message: WORKSPACE_REF_MESSAGE }),
    name: zod_1.z
        .string()
        .min(1, { message: MIN_NAME_MESSAGE })
        .max(50, { message: MAX_NAME_MESSAGE })
        .or(zod_1.z.string().optional())
});
exports.updateWorkspaceRequestSchema = updateWorkspaceRequestSchema;
const removeUserFromWorkspaceRequestSchema = zod_1.z.object({
    userRef: zod_1.z.string().uuid({ message: USER_REF_MESSAGE })
});
exports.removeUserFromWorkspaceRequestSchema = removeUserFromWorkspaceRequestSchema;
const sendVerificationCodeRequestSchema = zod_1.z.object({
    contactType: zod_1.z.enum(["EMAIL", "PHONE"]).default("EMAIL"),
    value: zod_1.z.string()
});
exports.sendVerificationCodeRequestSchema = sendVerificationCodeRequestSchema;
const verifyCodeRequestSchema = zod_1.z.object({
    username: zod_1.z.string(),
    contactType: zod_1.z.enum(["EMAIL", "PHONE"]).default("EMAIL"),
    value: zod_1.z.string(),
    verificationCode: zod_1.z.string()
});
exports.verifyCodeRequestSchema = verifyCodeRequestSchema;
const sendResetPasswordCodeRequestSchema = zod_1.z.object({
    username: zod_1.z
        .string()
        .email({ message: "Invalid username. Must be an email address" }),
    resetPasswordUrl: zod_1.z.string()
});
exports.sendResetPasswordCodeRequestSchema = sendResetPasswordCodeRequestSchema;
const resetPasswordRequestSchema = zod_1.z.object({
    username: zod_1.z.string(),
    password: zod_1.z.string(),
    verificationCode: zod_1.z.string()
});
exports.resetPasswordRequestSchema = resetPasswordRequestSchema;
