// @fonoster/common/dist/tts/AzureVoices.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AzureVoiceDetails = exports.AzureVoice = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const types_1 = require("./types");
var AzureVoice;
(function (AzureVoice) {
    AzureVoice["AF_ZA_ADRI_NEURAL"] = "af-ZA-AdriNeural";
    AzureVoice["AF_ZA_WILLEM_NEURAL"] = "af-ZA-WillemNeural";
    AzureVoice["AM_ET_MEKDES_NEURAL"] = "am-ET-MekdesNeural";
    AzureVoice["AM_ET_AMEHA_NEURAL"] = "am-ET-AmehaNeural";
    AzureVoice["AR_AE_FATIMA_NEURAL"] = "ar-AE-FatimaNeural";
    AzureVoice["AR_AE_HAMDAN_NEURAL"] = "ar-AE-HamdanNeural";
    AzureVoice["AR_BH_LAILA_NEURAL"] = "ar-BH-LailaNeural";
    AzureVoice["AR_BH_ALI_NEURAL"] = "ar-BH-AliNeural";
    AzureVoice["AR_DZ_AMINA_NEURAL"] = "ar-DZ-AminaNeural";
    AzureVoice["AR_DZ_ISMAEL_NEURAL"] = "ar-DZ-IsmaelNeural";
    AzureVoice["AR_EG_SALMA_NEURAL"] = "ar-EG-SalmaNeural";
    AzureVoice["AR_EG_SHAKIR_NEURAL"] = "ar-EG-ShakirNeural";
    AzureVoice["AR_IQ_RANA_NEURAL"] = "ar-IQ-RanaNeural";
    AzureVoice["AR_IQ_BASSEL_NEURAL"] = "ar-IQ-BasselNeural";
    AzureVoice["AR_JO_SANA_NEURAL"] = "ar-JO-SanaNeural";
    AzureVoice["AR_JO_TAIM_NEURAL"] = "ar-JO-TaimNeural";
    AzureVoice["AR_KW_NOURA_NEURAL"] = "ar-KW-NouraNeural";
    AzureVoice["AR_KW_FAHED_NEURAL"] = "ar-KW-FahedNeural";
    AzureVoice["AR_LB_LAYLA_NEURAL"] = "ar-LB-LaylaNeural";
    AzureVoice["AR_LB_RAMI_NEURAL"] = "ar-LB-RamiNeural";
    AzureVoice["AR_LY_IMAN_NEURAL"] = "ar-LY-ImanNeural";
    AzureVoice["AR_LY_OMAR_NEURAL"] = "ar-LY-OmarNeural";
    AzureVoice["AR_MA_MOUNA_NEURAL"] = "ar-MA-MounaNeural";
    AzureVoice["AR_MA_JAMAL_NEURAL"] = "ar-MA-JamalNeural";
    AzureVoice["AR_OM_AYSHA_NEURAL"] = "ar-OM-AyshaNeural";
    AzureVoice["AR_OM_ABDULLAH_NEURAL"] = "ar-OM-AbdullahNeural";
    AzureVoice["AR_QA_AMAL_NEURAL"] = "ar-QA-AmalNeural";
    AzureVoice["AR_QA_MOAZ_NEURAL"] = "ar-QA-MoazNeural";
    AzureVoice["AR_SA_ZARIYAH_NEURAL"] = "ar-SA-ZariyahNeural";
    AzureVoice["AR_SA_HAMED_NEURAL"] = "ar-SA-HamedNeural";
    AzureVoice["AR_SY_AMANY_NEURAL"] = "ar-SY-AmanyNeural";
    AzureVoice["AR_SY_LAITH_NEURAL"] = "ar-SY-LaithNeural";
    AzureVoice["AR_TN_REEM_NEURAL"] = "ar-TN-ReemNeural";
    AzureVoice["AR_TN_HEDI_NEURAL"] = "ar-TN-HediNeural";
    AzureVoice["AR_YE_MARYAM_NEURAL"] = "ar-YE-MaryamNeural";
    AzureVoice["AR_YE_SALEH_NEURAL"] = "ar-YE-SalehNeural";
    AzureVoice["AS_IN_PRIYOM_NEURAL"] = "as-IN-PriyomNeural";
    AzureVoice["AS_IN_YASHICA_NEURAL"] = "as-IN-YashicaNeural";
    AzureVoice["AZ_AZ_BANU_NEURAL"] = "az-AZ-BanuNeural";
    AzureVoice["AZ_AZ_BABEK_NEURAL"] = "az-AZ-BabekNeural";
    AzureVoice["BG_BG_KALINA_NEURAL"] = "bg-BG-KalinaNeural";
    AzureVoice["BG_BG_BORISLAV_NEURAL"] = "bg-BG-BorislavNeural";
    AzureVoice["BN_BD_NABANITA_NEURAL"] = "bn-BD-NabanitaNeural";
    AzureVoice["BN_BD_PRADEEP_NEURAL"] = "bn-BD-PradeepNeural";
    AzureVoice["BN_IN_TANISHAA_NEURAL"] = "bn-IN-TanishaaNeural";
    AzureVoice["BN_IN_BASHKAR_NEURAL"] = "bn-IN-BashkarNeural";
    AzureVoice["BS_BA_VESNA_NEURAL"] = "bs-BA-VesnaNeural";
    AzureVoice["BS_BA_GORAN_NEURAL"] = "bs-BA-GoranNeural";
    AzureVoice["CA_ES_JOANA_NEURAL"] = "ca-ES-JoanaNeural";
    AzureVoice["CA_ES_ENRIC_NEURAL"] = "ca-ES-EnricNeural";
    AzureVoice["CA_ES_ALBA_NEURAL"] = "ca-ES-AlbaNeural";
    AzureVoice["CS_CZ_VLASTA_NEURAL"] = "cs-CZ-VlastaNeural";
    AzureVoice["CS_CZ_ANTONIN_NEURAL"] = "cs-CZ-AntoninNeural";
    AzureVoice["CY_GB_NIA_NEURAL"] = "cy-GB-NiaNeural";
    AzureVoice["CY_GB_ALED_NEURAL"] = "cy-GB-AledNeural";
    AzureVoice["DA_DK_CHRISTEL_NEURAL"] = "da-DK-ChristelNeural";
    AzureVoice["DA_DK_JEPPE_NEURAL"] = "da-DK-JeppeNeural";
    AzureVoice["DE_AT_INGRID_NEURAL"] = "de-AT-IngridNeural";
    AzureVoice["DE_AT_JONAS_NEURAL"] = "de-AT-JonasNeural";
    AzureVoice["DE_CH_LENI_NEURAL"] = "de-CH-LeniNeural";
    AzureVoice["DE_CH_JAN_NEURAL"] = "de-CH-JanNeural";
    AzureVoice["DE_DE_KATJA_NEURAL"] = "de-DE-KatjaNeural";
    AzureVoice["DE_DE_CONRAD_NEURAL"] = "de-DE-ConradNeural";
    AzureVoice["DE_DE_AMALA_NEURAL"] = "de-DE-AmalaNeural";
    AzureVoice["DE_DE_BERND_NEURAL"] = "de-DE-BerndNeural";
    AzureVoice["DE_DE_CHRISTOPH_NEURAL"] = "de-DE-ChristophNeural";
    AzureVoice["DE_DE_ELKE_NEURAL"] = "de-DE-ElkeNeural";
    AzureVoice["DE_DE_FLORIANMULTILINGUAL_NEURAL"] = "de-DE-FlorianMultilingualNeural";
    AzureVoice["DE_DE_GISELA_NEURAL"] = "de-DE-GiselaNeural";
    AzureVoice["DE_DE_KASPER_NEURAL"] = "de-DE-KasperNeural";
    AzureVoice["DE_DE_KILLIAN_NEURAL"] = "de-DE-KillianNeural";
    AzureVoice["DE_DE_KLARISSA_NEURAL"] = "de-DE-KlarissaNeural";
    AzureVoice["DE_DE_KLAUS_NEURAL"] = "de-DE-KlausNeural";
    AzureVoice["DE_DE_LOUISA_NEURAL"] = "de-DE-LouisaNeural";
    AzureVoice["DE_DE_MAJA_NEURAL"] = "de-DE-MajaNeural";
    AzureVoice["DE_DE_RALF_NEURAL"] = "de-DE-RalfNeural";
    AzureVoice["DE_DE_SERAPHINAMULTILINGUAL_NEURAL"] = "de-DE-SeraphinaMultilingualNeural";
    AzureVoice["DE_DE_TANJA_NEURAL"] = "de-DE-TanjaNeural";
    AzureVoice["EL_GR_ATHINA_NEURAL"] = "el-GR-AthinaNeural";
    AzureVoice["EL_GR_NESTORAS_NEURAL"] = "el-GR-NestorasNeural";
    AzureVoice["EN_AU_NATASHA_NEURAL"] = "en-AU-NatashaNeural";
    AzureVoice["EN_AU_WILLIAM_NEURAL"] = "en-AU-WilliamNeural";
    AzureVoice["EN_AU_ANNETTE_NEURAL"] = "en-AU-AnnetteNeural";
    AzureVoice["EN_AU_CARLY_NEURAL"] = "en-AU-CarlyNeural";
    AzureVoice["EN_AU_DARREN_NEURAL"] = "en-AU-DarrenNeural";
    AzureVoice["EN_AU_DUNCAN_NEURAL"] = "en-AU-DuncanNeural";
    AzureVoice["EN_AU_ELSIE_NEURAL"] = "en-AU-ElsieNeural";
    AzureVoice["EN_AU_FREYA_NEURAL"] = "en-AU-FreyaNeural";
    AzureVoice["EN_AU_JOANNE_NEURAL"] = "en-AU-JoanneNeural";
    AzureVoice["EN_AU_KEN_NEURAL"] = "en-AU-KenNeural";
    AzureVoice["EN_AU_KIM_NEURAL"] = "en-AU-KimNeural";
    AzureVoice["EN_AU_NEIL_NEURAL"] = "en-AU-NeilNeural";
    AzureVoice["EN_AU_TIM_NEURAL"] = "en-AU-TimNeural";
    AzureVoice["EN_AU_TINA_NEURAL"] = "en-AU-TinaNeural";
    AzureVoice["EN_CA_CLARA_NEURAL"] = "en-CA-ClaraNeural";
    AzureVoice["EN_CA_LIAM_NEURAL"] = "en-CA-LiamNeural";
    AzureVoice["EN_GB_SONIA_NEURAL"] = "en-GB-SoniaNeural";
    AzureVoice["EN_GB_RYAN_NEURAL"] = "en-GB-RyanNeural";
    AzureVoice["EN_GB_LIBBY_NEURAL"] = "en-GB-LibbyNeural";
    AzureVoice["EN_GB_ABBI_NEURAL"] = "en-GB-AbbiNeural";
    AzureVoice["EN_GB_ADAMULTILINGUAL_NEURAL"] = "en-GB-AdaMultilingualNeural";
    AzureVoice["EN_GB_ALFIE_NEURAL"] = "en-GB-AlfieNeural";
    AzureVoice["EN_GB_BELLA_NEURAL"] = "en-GB-BellaNeural";
    AzureVoice["EN_GB_ELLIOT_NEURAL"] = "en-GB-ElliotNeural";
    AzureVoice["EN_GB_ETHAN_NEURAL"] = "en-GB-EthanNeural";
    AzureVoice["EN_GB_HOLLIE_NEURAL"] = "en-GB-HollieNeural";
    AzureVoice["EN_GB_MAISIE_NEURAL"] = "en-GB-MaisieNeural";
    AzureVoice["EN_GB_NOAH_NEURAL"] = "en-GB-NoahNeural";
    AzureVoice["EN_GB_OLIVER_NEURAL"] = "en-GB-OliverNeural";
    AzureVoice["EN_GB_OLIVIA_NEURAL"] = "en-GB-OliviaNeural";
    AzureVoice["EN_GB_THOMAS_NEURAL"] = "en-GB-ThomasNeural";
    AzureVoice["EN_GB_OLLIEMULTILINGUAL_NEURAL"] = "en-GB-OllieMultilingualNeural";
    AzureVoice["EN_GB_MIA_NEURAL"] = "en-GB-MiaNeural";
    AzureVoice["EN_HK_YAN_NEURAL"] = "en-HK-YanNeural";
    AzureVoice["EN_HK_SAM_NEURAL"] = "en-HK-SamNeural";
    AzureVoice["EN_IE_EMILY_NEURAL"] = "en-IE-EmilyNeural";
    AzureVoice["EN_IE_CONNOR_NEURAL"] = "en-IE-ConnorNeural";
    AzureVoice["EN_IN_NEERJA_NEURAL"] = "en-IN-NeerjaNeural";
    AzureVoice["EN_IN_PRABHAT_NEURAL"] = "en-IN-PrabhatNeural";
    AzureVoice["EN_IN_AARAV_NEURAL"] = "en-IN-AaravNeural";
    AzureVoice["EN_IN_AASHI_NEURAL"] = "en-IN-AashiNeural";
    AzureVoice["EN_IN_ANANYA_NEURAL"] = "en-IN-AnanyaNeural";
    AzureVoice["EN_IN_KAVYA_NEURAL"] = "en-IN-KavyaNeural";
    AzureVoice["EN_IN_KUNAL_NEURAL"] = "en-IN-KunalNeural";
    AzureVoice["EN_IN_REHAAN_NEURAL"] = "en-IN-RehaanNeural";
    AzureVoice["EN_KE_ASILIA_NEURAL"] = "en-KE-AsiliaNeural";
    AzureVoice["EN_KE_CHILEMBA_NEURAL"] = "en-KE-ChilembaNeural";
    AzureVoice["EN_NG_EZINNE_NEURAL"] = "en-NG-EzinneNeural";
    AzureVoice["EN_NG_ABEO_NEURAL"] = "en-NG-AbeoNeural";
    AzureVoice["EN_NZ_MOLLY_NEURAL"] = "en-NZ-MollyNeural";
    AzureVoice["EN_NZ_MITCHELL_NEURAL"] = "en-NZ-MitchellNeural";
    AzureVoice["EN_PH_ROSA_NEURAL"] = "en-PH-RosaNeural";
    AzureVoice["EN_PH_JAMES_NEURAL"] = "en-PH-JamesNeural";
    AzureVoice["EN_SG_LUNA_NEURAL"] = "en-SG-LunaNeural";
    AzureVoice["EN_SG_WAYNE_NEURAL"] = "en-SG-WayneNeural";
    AzureVoice["EN_TZ_IMANI_NEURAL"] = "en-TZ-ImaniNeural";
    AzureVoice["EN_TZ_ELIMU_NEURAL"] = "en-TZ-ElimuNeural";
    AzureVoice["EN_US_AVAMULTILINGUAL_NEURAL"] = "en-US-AvaMultilingualNeural";
    AzureVoice["EN_US_ANDREWMULTILINGUAL_NEURAL"] = "en-US-AndrewMultilingualNeural";
    AzureVoice["EN_US_EMMAMULTILINGUAL_NEURAL"] = "en-US-EmmaMultilingualNeural";
    AzureVoice["EN_US_BRIANMULTILINGUAL_NEURAL"] = "en-US-BrianMultilingualNeural";
    AzureVoice["EN_US_AVA_NEURAL"] = "en-US-AvaNeural";
    AzureVoice["EN_US_ANDREW_NEURAL"] = "en-US-AndrewNeural";
    AzureVoice["EN_US_EMMA_NEURAL"] = "en-US-EmmaNeural";
    AzureVoice["EN_US_BRIAN_NEURAL"] = "en-US-BrianNeural";
    AzureVoice["EN_US_JENNY_NEURAL"] = "en-US-JennyNeural";
    AzureVoice["EN_US_GUY_NEURAL"] = "en-US-GuyNeural";
    AzureVoice["EN_US_ARIA_NEURAL"] = "en-US-AriaNeural";
    AzureVoice["EN_US_DAVIS_NEURAL"] = "en-US-DavisNeural";
    AzureVoice["EN_US_JANE_NEURAL"] = "en-US-JaneNeural";
    AzureVoice["EN_US_JASON_NEURAL"] = "en-US-JasonNeural";
    AzureVoice["EN_US_SARA_NEURAL"] = "en-US-SaraNeural";
    AzureVoice["EN_US_TONY_NEURAL"] = "en-US-TonyNeural";
    AzureVoice["EN_US_NANCY_NEURAL"] = "en-US-NancyNeural";
    AzureVoice["EN_US_AMBER_NEURAL"] = "en-US-AmberNeural";
    AzureVoice["EN_US_ANA_NEURAL"] = "en-US-AnaNeural";
    AzureVoice["EN_US_ASHLEY_NEURAL"] = "en-US-AshleyNeural";
    AzureVoice["EN_US_BRANDON_NEURAL"] = "en-US-BrandonNeural";
    AzureVoice["EN_US_CHRISTOPHER_NEURAL"] = "en-US-ChristopherNeural";
    AzureVoice["EN_US_CORA_NEURAL"] = "en-US-CoraNeural";
    AzureVoice["EN_US_ELIZABETH_NEURAL"] = "en-US-ElizabethNeural";
    AzureVoice["EN_US_ERIC_NEURAL"] = "en-US-EricNeural";
    AzureVoice["EN_US_JACOB_NEURAL"] = "en-US-JacobNeural";
    AzureVoice["EN_US_JENNYMULTILINGUAL_NEURAL"] = "en-US-JennyMultilingualNeural";
    AzureVoice["EN_US_MICHELLE_NEURAL"] = "en-US-MichelleNeural";
    AzureVoice["EN_US_MONICA_NEURAL"] = "en-US-MonicaNeural";
    AzureVoice["EN_US_ROGER_NEURAL"] = "en-US-RogerNeural";
    AzureVoice["EN_US_RYANMULTILINGUAL_NEURAL"] = "en-US-RyanMultilingualNeural";
    AzureVoice["EN_US_STEFFAN_NEURAL"] = "en-US-SteffanNeural";
    AzureVoice["EN_US_AIGENERATE1_NEURAL"] = "en-US-AIGenerate1Neural";
    AzureVoice["EN_US_AIGENERATE2_NEURAL"] = "en-US-AIGenerate2Neural";
    AzureVoice["EN_US_BLUE_NEURAL"] = "en-US-BlueNeural";
    AzureVoice["EN_US_KAI_NEURAL"] = "en-US-KaiNeural";
    AzureVoice["EN_US_LUNA_NEURAL"] = "en-US-LunaNeural";
    AzureVoice["EN_ZA_LEAH_NEURAL"] = "en-ZA-LeahNeural";
    AzureVoice["EN_ZA_LUKE_NEURAL"] = "en-ZA-LukeNeural";
    AzureVoice["ES_AR_ELENA_NEURAL"] = "es-AR-ElenaNeural";
    AzureVoice["ES_AR_TOMAS_NEURAL"] = "es-AR-TomasNeural";
    AzureVoice["ES_BO_SOFIA_NEURAL"] = "es-BO-SofiaNeural";
    AzureVoice["ES_BO_MARCELO_NEURAL"] = "es-BO-MarceloNeural";
    AzureVoice["ES_CL_CATALINA_NEURAL"] = "es-CL-CatalinaNeural";
    AzureVoice["ES_CL_LORENZO_NEURAL"] = "es-CL-LorenzoNeural";
    AzureVoice["ES_CO_SALOME_NEURAL"] = "es-CO-SalomeNeural";
    AzureVoice["ES_CO_GONZALO_NEURAL"] = "es-CO-GonzaloNeural";
    AzureVoice["ES_CR_MARIA_NEURAL"] = "es-CR-MariaNeural";
    AzureVoice["ES_CR_JUAN_NEURAL"] = "es-CR-JuanNeural";
    AzureVoice["ES_CU_BELKYS_NEURAL"] = "es-CU-BelkysNeural";
    AzureVoice["ES_CU_MANUEL_NEURAL"] = "es-CU-ManuelNeural";
    AzureVoice["ES_DO_RAMONA_NEURAL"] = "es-DO-RamonaNeural";
    AzureVoice["ES_DO_EMILIO_NEURAL"] = "es-DO-EmilioNeural";
    AzureVoice["ES_EC_ANDREA_NEURAL"] = "es-EC-AndreaNeural";
    AzureVoice["ES_EC_LUIS_NEURAL"] = "es-EC-LuisNeural";
    AzureVoice["ES_ES_ELVIRA_NEURAL"] = "es-ES-ElviraNeural";
    AzureVoice["ES_ES_ALVARO_NEURAL"] = "es-ES-AlvaroNeural";
    AzureVoice["ES_ES_ABRIL_NEURAL"] = "es-ES-AbrilNeural";
    AzureVoice["ES_ES_ARNAU_NEURAL"] = "es-ES-ArnauNeural";
    AzureVoice["ES_ES_DARIO_NEURAL"] = "es-ES-DarioNeural";
    AzureVoice["ES_ES_ELIAS_NEURAL"] = "es-ES-EliasNeural";
    AzureVoice["ES_ES_ESTRELLA_NEURAL"] = "es-ES-EstrellaNeural";
    AzureVoice["ES_ES_IRENE_NEURAL"] = "es-ES-IreneNeural";
    AzureVoice["ES_ES_LAIA_NEURAL"] = "es-ES-LaiaNeural";
    AzureVoice["ES_ES_LIA_NEURAL"] = "es-ES-LiaNeural";
    AzureVoice["ES_ES_NIL_NEURAL"] = "es-ES-NilNeural";
    AzureVoice["ES_ES_SAUL_NEURAL"] = "es-ES-SaulNeural";
    AzureVoice["ES_ES_TEO_NEURAL"] = "es-ES-TeoNeural";
    AzureVoice["ES_ES_TRIANA_NEURAL"] = "es-ES-TrianaNeural";
    AzureVoice["ES_ES_VERA_NEURAL"] = "es-ES-VeraNeural";
    AzureVoice["ES_ES_XIMENA_NEURAL"] = "es-ES-XimenaNeural";
    AzureVoice["ES_ES_ARABELLAMULTILINGUAL_NEURAL"] = "es-ES-ArabellaMultilingualNeural";
    AzureVoice["ES_ES_ISIDORAMULTILINGUAL_NEURAL"] = "es-ES-IsidoraMultilingualNeural";
    AzureVoice["ES_GQ_TERESA_NEURAL"] = "es-GQ-TeresaNeural";
    AzureVoice["ES_GQ_JAVIER_NEURAL"] = "es-GQ-JavierNeural";
    AzureVoice["ES_GT_MARTA_NEURAL"] = "es-GT-MartaNeural";
    AzureVoice["ES_GT_ANDRES_NEURAL"] = "es-GT-AndresNeural";
    AzureVoice["ES_HN_KARLA_NEURAL"] = "es-HN-KarlaNeural";
    AzureVoice["ES_HN_CARLOS_NEURAL"] = "es-HN-CarlosNeural";
    AzureVoice["ES_MX_DALIA_NEURAL"] = "es-MX-DaliaNeural";
    AzureVoice["ES_MX_JORGE_NEURAL"] = "es-MX-JorgeNeural";
    AzureVoice["ES_MX_BEATRIZ_NEURAL"] = "es-MX-BeatrizNeural";
    AzureVoice["ES_MX_CANDELA_NEURAL"] = "es-MX-CandelaNeural";
    AzureVoice["ES_MX_CARLOTA_NEURAL"] = "es-MX-CarlotaNeural";
    AzureVoice["ES_MX_CECILIO_NEURAL"] = "es-MX-CecilioNeural";
    AzureVoice["ES_MX_GERARDO_NEURAL"] = "es-MX-GerardoNeural";
    AzureVoice["ES_MX_LARISSA_NEURAL"] = "es-MX-LarissaNeural";
    AzureVoice["ES_MX_LIBERTO_NEURAL"] = "es-MX-LibertoNeural";
    AzureVoice["ES_MX_LUCIANO_NEURAL"] = "es-MX-LucianoNeural";
    AzureVoice["ES_MX_MARINA_NEURAL"] = "es-MX-MarinaNeural";
    AzureVoice["ES_MX_NURIA_NEURAL"] = "es-MX-NuriaNeural";
    AzureVoice["ES_MX_PELAYO_NEURAL"] = "es-MX-PelayoNeural";
    AzureVoice["ES_MX_RENATA_NEURAL"] = "es-MX-RenataNeural";
    AzureVoice["ES_MX_YAGO_NEURAL"] = "es-MX-YagoNeural";
    AzureVoice["ES_NI_YOLANDA_NEURAL"] = "es-NI-YolandaNeural";
    AzureVoice["ES_NI_FEDERICO_NEURAL"] = "es-NI-FedericoNeural";
    AzureVoice["ES_PA_MARGARITA_NEURAL"] = "es-PA-MargaritaNeural";
    AzureVoice["ES_PA_ROBERTO_NEURAL"] = "es-PA-RobertoNeural";
    AzureVoice["ES_PE_CAMILA_NEURAL"] = "es-PE-CamilaNeural";
    AzureVoice["ES_PE_ALEX_NEURAL"] = "es-PE-AlexNeural";
    AzureVoice["ES_PR_KARINA_NEURAL"] = "es-PR-KarinaNeural";
    AzureVoice["ES_PR_VICTOR_NEURAL"] = "es-PR-VictorNeural";
    AzureVoice["ES_PY_TANIA_NEURAL"] = "es-PY-TaniaNeural";
    AzureVoice["ES_PY_MARIO_NEURAL"] = "es-PY-MarioNeural";
    AzureVoice["ES_SV_LORENA_NEURAL"] = "es-SV-LorenaNeural";
    AzureVoice["ES_SV_RODRIGO_NEURAL"] = "es-SV-RodrigoNeural";
    AzureVoice["ES_US_PALOMA_NEURAL"] = "es-US-PalomaNeural";
    AzureVoice["ES_US_ALONSO_NEURAL"] = "es-US-AlonsoNeural";
    AzureVoice["ES_UY_VALENTINA_NEURAL"] = "es-UY-ValentinaNeural";
    AzureVoice["ES_UY_MATEO_NEURAL"] = "es-UY-MateoNeural";
    AzureVoice["ES_VE_PAOLA_NEURAL"] = "es-VE-PaolaNeural";
    AzureVoice["ES_VE_SEBASTIAN_NEURAL"] = "es-VE-SebastianNeural";
    AzureVoice["ET_EE_ANU_NEURAL"] = "et-EE-AnuNeural";
    AzureVoice["ET_EE_KERT_NEURAL"] = "et-EE-KertNeural";
    AzureVoice["EU_ES_AINHOA_NEURAL"] = "eu-ES-AinhoaNeural";
    AzureVoice["EU_ES_ANDER_NEURAL"] = "eu-ES-AnderNeural";
    AzureVoice["FA_IR_DILARA_NEURAL"] = "fa-IR-DilaraNeural";
    AzureVoice["FA_IR_FARID_NEURAL"] = "fa-IR-FaridNeural";
    AzureVoice["FI_FI_SELMA_NEURAL"] = "fi-FI-SelmaNeural";
    AzureVoice["FI_FI_HARRI_NEURAL"] = "fi-FI-HarriNeural";
    AzureVoice["FI_FI_NOORA_NEURAL"] = "fi-FI-NooraNeural";
    AzureVoice["FIL_PH_BLESSICA_NEURAL"] = "fil-PH-BlessicaNeural";
    AzureVoice["FIL_PH_ANGELO_NEURAL"] = "fil-PH-AngeloNeural";
    AzureVoice["FR_BE_CHARLINE_NEURAL"] = "fr-BE-CharlineNeural";
    AzureVoice["FR_BE_GERARD_NEURAL"] = "fr-BE-GerardNeural";
    AzureVoice["FR_CA_SYLVIE_NEURAL"] = "fr-CA-SylvieNeural";
    AzureVoice["FR_CA_JEAN_NEURAL"] = "fr-CA-JeanNeural";
    AzureVoice["FR_CA_ANTOINE_NEURAL"] = "fr-CA-AntoineNeural";
    AzureVoice["FR_CA_THIERRY_NEURAL"] = "fr-CA-ThierryNeural";
    AzureVoice["FR_CH_ARIANE_NEURAL"] = "fr-CH-ArianeNeural";
    AzureVoice["FR_CH_FABRICE_NEURAL"] = "fr-CH-FabriceNeural";
    AzureVoice["FR_FR_DENISE_NEURAL"] = "fr-FR-DeniseNeural";
    AzureVoice["FR_FR_HENRI_NEURAL"] = "fr-FR-HenriNeural";
    AzureVoice["FR_FR_ALAIN_NEURAL"] = "fr-FR-AlainNeural";
    AzureVoice["FR_FR_BRIGITTE_NEURAL"] = "fr-FR-BrigitteNeural";
    AzureVoice["FR_FR_CELESTE_NEURAL"] = "fr-FR-CelesteNeural";
    AzureVoice["FR_FR_CLAUDE_NEURAL"] = "fr-FR-ClaudeNeural";
    AzureVoice["FR_FR_CORALIE_NEURAL"] = "fr-FR-CoralieNeural";
    AzureVoice["FR_FR_ELOISE_NEURAL"] = "fr-FR-EloiseNeural";
    AzureVoice["FR_FR_JACQUELINE_NEURAL"] = "fr-FR-JacquelineNeural";
    AzureVoice["FR_FR_JEROME_NEURAL"] = "fr-FR-JeromeNeural";
    AzureVoice["FR_FR_JOSEPHINE_NEURAL"] = "fr-FR-JosephineNeural";
    AzureVoice["FR_FR_MAURICE_NEURAL"] = "fr-FR-MauriceNeural";
    AzureVoice["FR_FR_REMYMULTILINGUAL_NEURAL"] = "fr-FR-RemyMultilingualNeural";
    AzureVoice["FR_FR_VIVIENNEMULTILINGUAL_NEURAL"] = "fr-FR-VivienneMultilingualNeural";
    AzureVoice["FR_FR_YVES_NEURAL"] = "fr-FR-YvesNeural";
    AzureVoice["FR_FR_YVETTE_NEURAL"] = "fr-FR-YvetteNeural";
    AzureVoice["GA_IE_ORLA_NEURAL"] = "ga-IE-OrlaNeural";
    AzureVoice["GA_IE_COLM_NEURAL"] = "ga-IE-ColmNeural";
    AzureVoice["GL_ES_SABELA_NEURAL"] = "gl-ES-SabelaNeural";
    AzureVoice["GL_ES_ROI_NEURAL"] = "gl-ES-RoiNeural";
    AzureVoice["GU_IN_DHWANI_NEURAL"] = "gu-IN-DhwaniNeural";
    AzureVoice["GU_IN_NIRANJAN_NEURAL"] = "gu-IN-NiranjanNeural";
    AzureVoice["HE_IL_HILA_NEURAL"] = "he-IL-HilaNeural";
    AzureVoice["HE_IL_AVRI_NEURAL"] = "he-IL-AvriNeural";
    AzureVoice["HI_IN_SWARA_NEURAL"] = "hi-IN-SwaraNeural";
    AzureVoice["HI_IN_MADHUR_NEURAL"] = "hi-IN-MadhurNeural";
    AzureVoice["HI_IN_AARAV_NEURAL"] = "hi-IN-AaravNeural";
    AzureVoice["HI_IN_ANANYA_NEURAL"] = "hi-IN-AnanyaNeural";
    AzureVoice["HI_IN_KAVYA_NEURAL"] = "hi-IN-KavyaNeural";
    AzureVoice["HI_IN_KUNAL_NEURAL"] = "hi-IN-KunalNeural";
    AzureVoice["HI_IN_REHAAN_NEURAL"] = "hi-IN-RehaanNeural";
    AzureVoice["HR_HR_GABRIJELA_NEURAL"] = "hr-HR-GabrijelaNeural";
    AzureVoice["HR_HR_SRECKO_NEURAL"] = "hr-HR-SreckoNeural";
    AzureVoice["HU_HU_NOEMI_NEURAL"] = "hu-HU-NoemiNeural";
    AzureVoice["HU_HU_TAMAS_NEURAL"] = "hu-HU-TamasNeural";
    AzureVoice["HY_AM_ANAHIT_NEURAL"] = "hy-AM-AnahitNeural";
    AzureVoice["HY_AM_HAYK_NEURAL"] = "hy-AM-HaykNeural";
    AzureVoice["ID_ID_GADIS_NEURAL"] = "id-ID-GadisNeural";
    AzureVoice["ID_ID_ARDI_NEURAL"] = "id-ID-ArdiNeural";
    AzureVoice["IS_IS_GUDRUN_NEURAL"] = "is-IS-GudrunNeural";
    AzureVoice["IS_IS_GUNNAR_NEURAL"] = "is-IS-GunnarNeural";
    AzureVoice["IT_IT_ELSA_NEURAL"] = "it-IT-ElsaNeural";
    AzureVoice["IT_IT_ISABELLA_NEURAL"] = "it-IT-IsabellaNeural";
    AzureVoice["IT_IT_DIEGO_NEURAL"] = "it-IT-DiegoNeural";
    AzureVoice["IT_IT_BENIGNO_NEURAL"] = "it-IT-BenignoNeural";
    AzureVoice["IT_IT_CALIMERO_NEURAL"] = "it-IT-CalimeroNeural";
    AzureVoice["IT_IT_CATALDO_NEURAL"] = "it-IT-CataldoNeural";
    AzureVoice["IT_IT_FABIOLA_NEURAL"] = "it-IT-FabiolaNeural";
    AzureVoice["IT_IT_FIAMMA_NEURAL"] = "it-IT-FiammaNeural";
    AzureVoice["IT_IT_GIANNI_NEURAL"] = "it-IT-GianniNeural";
    AzureVoice["IT_IT_GIUSEPPE_NEURAL"] = "it-IT-GiuseppeNeural";
    AzureVoice["IT_IT_IMELDA_NEURAL"] = "it-IT-ImeldaNeural";
    AzureVoice["IT_IT_IRMA_NEURAL"] = "it-IT-IrmaNeural";
    AzureVoice["IT_IT_LISANDRO_NEURAL"] = "it-IT-LisandroNeural";
    AzureVoice["IT_IT_PALMIRA_NEURAL"] = "it-IT-PalmiraNeural";
    AzureVoice["IT_IT_PIERINA_NEURAL"] = "it-IT-PierinaNeural";
    AzureVoice["IT_IT_RINALDO_NEURAL"] = "it-IT-RinaldoNeural";
    AzureVoice["IT_IT_ALESSIOMULTILINGUAL_NEURAL"] = "it-IT-AlessioMultilingualNeural";
    AzureVoice["IT_IT_ISABELLAMULTILINGUAL_NEURAL"] = "it-IT-IsabellaMultilingualNeural";
    AzureVoice["IT_IT_MARCELLOMULTILINGUAL_NEURAL"] = "it-IT-MarcelloMultilingualNeural";
    AzureVoice["JA_JP_NANAMI_NEURAL"] = "ja-JP-NanamiNeural";
    AzureVoice["JA_JP_KEITA_NEURAL"] = "ja-JP-KeitaNeural";
    AzureVoice["JA_JP_AOI_NEURAL"] = "ja-JP-AoiNeural";
    AzureVoice["JA_JP_DAICHI_NEURAL"] = "ja-JP-DaichiNeural";
    AzureVoice["JA_JP_MAYU_NEURAL"] = "ja-JP-MayuNeural";
    AzureVoice["JA_JP_NAOKI_NEURAL"] = "ja-JP-NaokiNeural";
    AzureVoice["JA_JP_SHIORI_NEURAL"] = "ja-JP-ShioriNeural";
    AzureVoice["JA_JP_MASARUMULTILINGUAL_NEURAL"] = "ja-JP-MasaruMultilingualNeural";
    AzureVoice["JV_ID_SITI_NEURAL"] = "jv-ID-SitiNeural";
    AzureVoice["JV_ID_DIMAS_NEURAL"] = "jv-ID-DimasNeural";
    AzureVoice["KA_GE_EKA_NEURAL"] = "ka-GE-EkaNeural";
    AzureVoice["KA_GE_GIORGI_NEURAL"] = "ka-GE-GiorgiNeural";
    AzureVoice["KK_KZ_AIGUL_NEURAL"] = "kk-KZ-AigulNeural";
    AzureVoice["KK_KZ_DAULET_NEURAL"] = "kk-KZ-DauletNeural";
    AzureVoice["KM_KH_SREYMOM_NEURAL"] = "km-KH-SreymomNeural";
    AzureVoice["KM_KH_PISETH_NEURAL"] = "km-KH-PisethNeural";
    AzureVoice["KN_IN_SAPNA_NEURAL"] = "kn-IN-SapnaNeural";
    AzureVoice["KN_IN_GAGAN_NEURAL"] = "kn-IN-GaganNeural";
    AzureVoice["KO_KR_SUNHI_NEURAL"] = "ko-KR-SunHiNeural";
    AzureVoice["KO_KR_INJOON_NEURAL"] = "ko-KR-InJoonNeural";
    AzureVoice["KO_KR_BONGJIN_NEURAL"] = "ko-KR-BongJinNeural";
    AzureVoice["KO_KR_GOOKMIN_NEURAL"] = "ko-KR-GookMinNeural";
    AzureVoice["KO_KR_HYUNSU_NEURAL"] = "ko-KR-HyunsuNeural";
    AzureVoice["KO_KR_JIMIN_NEURAL"] = "ko-KR-JiMinNeural";
    AzureVoice["KO_KR_SEOHYEON_NEURAL"] = "ko-KR-SeoHyeonNeural";
    AzureVoice["KO_KR_SOONBOK_NEURAL"] = "ko-KR-SoonBokNeural";
    AzureVoice["KO_KR_YUJIN_NEURAL"] = "ko-KR-YuJinNeural";
    AzureVoice["LO_LA_KEOMANY_NEURAL"] = "lo-LA-KeomanyNeural";
    AzureVoice["LO_LA_CHANTHAVONG_NEURAL"] = "lo-LA-ChanthavongNeural";
    AzureVoice["LT_LT_ONA_NEURAL"] = "lt-LT-OnaNeural";
    AzureVoice["LT_LT_LEONAS_NEURAL"] = "lt-LT-LeonasNeural";
    AzureVoice["LV_LV_EVERITA_NEURAL"] = "lv-LV-EveritaNeural";
    AzureVoice["LV_LV_NILS_NEURAL"] = "lv-LV-NilsNeural";
    AzureVoice["MK_MK_MARIJA_NEURAL"] = "mk-MK-MarijaNeural";
    AzureVoice["MK_MK_ALEKSANDAR_NEURAL"] = "mk-MK-AleksandarNeural";
    AzureVoice["ML_IN_SOBHANA_NEURAL"] = "ml-IN-SobhanaNeural";
    AzureVoice["ML_IN_MIDHUN_NEURAL"] = "ml-IN-MidhunNeural";
    AzureVoice["MN_MN_YESUI_NEURAL"] = "mn-MN-YesuiNeural";
    AzureVoice["MN_MN_BATAA_NEURAL"] = "mn-MN-BataaNeural";
    AzureVoice["MR_IN_AAROHI_NEURAL"] = "mr-IN-AarohiNeural";
    AzureVoice["MR_IN_MANOHAR_NEURAL"] = "mr-IN-ManoharNeural";
    AzureVoice["MS_MY_YASMIN_NEURAL"] = "ms-MY-YasminNeural";
    AzureVoice["MS_MY_OSMAN_NEURAL"] = "ms-MY-OsmanNeural";
    AzureVoice["MT_MT_GRACE_NEURAL"] = "mt-MT-GraceNeural";
    AzureVoice["MT_MT_JOSEPH_NEURAL"] = "mt-MT-JosephNeural";
    AzureVoice["MY_MM_NILAR_NEURAL"] = "my-MM-NilarNeural";
    AzureVoice["MY_MM_THIHA_NEURAL"] = "my-MM-ThihaNeural";
    AzureVoice["NB_NO_PERNILLE_NEURAL"] = "nb-NO-PernilleNeural";
    AzureVoice["NB_NO_FINN_NEURAL"] = "nb-NO-FinnNeural";
    AzureVoice["NB_NO_ISELIN_NEURAL"] = "nb-NO-IselinNeural";
    AzureVoice["NE_NP_HEMKALA_NEURAL"] = "ne-NP-HemkalaNeural";
    AzureVoice["NE_NP_SAGAR_NEURAL"] = "ne-NP-SagarNeural";
    AzureVoice["NL_BE_DENA_NEURAL"] = "nl-BE-DenaNeural";
    AzureVoice["NL_BE_ARNAUD_NEURAL"] = "nl-BE-ArnaudNeural";
    AzureVoice["NL_NL_FENNA_NEURAL"] = "nl-NL-FennaNeural";
    AzureVoice["NL_NL_MAARTEN_NEURAL"] = "nl-NL-MaartenNeural";
    AzureVoice["NL_NL_COLETTE_NEURAL"] = "nl-NL-ColetteNeural";
    AzureVoice["OR_IN_SUBHASINI_NEURAL"] = "or-IN-SubhasiniNeural";
    AzureVoice["OR_IN_SUKANT_NEURAL"] = "or-IN-SukantNeural";
    AzureVoice["PA_IN_OJAS_NEURAL"] = "pa-IN-OjasNeural";
    AzureVoice["PA_IN_VAANI_NEURAL"] = "pa-IN-VaaniNeural";
    AzureVoice["PL_PL_AGNIESZKA_NEURAL"] = "pl-PL-AgnieszkaNeural";
    AzureVoice["PL_PL_MAREK_NEURAL"] = "pl-PL-MarekNeural";
    AzureVoice["PL_PL_ZOFIA_NEURAL"] = "pl-PL-ZofiaNeural";
    AzureVoice["PS_AF_LATIFA_NEURAL"] = "ps-AF-LatifaNeural";
    AzureVoice["PS_AF_GULNAWAZ_NEURAL"] = "ps-AF-GulNawazNeural";
    AzureVoice["PT_BR_FRANCISCA_NEURAL"] = "pt-BR-FranciscaNeural";
    AzureVoice["PT_BR_ANTONIO_NEURAL"] = "pt-BR-AntonioNeural";
    AzureVoice["PT_BR_BRENDA_NEURAL"] = "pt-BR-BrendaNeural";
    AzureVoice["PT_BR_DONATO_NEURAL"] = "pt-BR-DonatoNeural";
    AzureVoice["PT_BR_ELZA_NEURAL"] = "pt-BR-ElzaNeural";
    AzureVoice["PT_BR_FABIO_NEURAL"] = "pt-BR-FabioNeural";
    AzureVoice["PT_BR_GIOVANNA_NEURAL"] = "pt-BR-GiovannaNeural";
    AzureVoice["PT_BR_HUMBERTO_NEURAL"] = "pt-BR-HumbertoNeural";
    AzureVoice["PT_BR_JULIO_NEURAL"] = "pt-BR-JulioNeural";
    AzureVoice["PT_BR_LEILA_NEURAL"] = "pt-BR-LeilaNeural";
    AzureVoice["PT_BR_LETICIA_NEURAL"] = "pt-BR-LeticiaNeural";
    AzureVoice["PT_BR_MANUELA_NEURAL"] = "pt-BR-ManuelaNeural";
    AzureVoice["PT_BR_NICOLAU_NEURAL"] = "pt-BR-NicolauNeural";
    AzureVoice["PT_BR_THALITAMULTILINGUAL_NEURAL"] = "pt-BR-ThalitaMultilingualNeural";
    AzureVoice["PT_BR_THALITA_NEURAL"] = "pt-BR-ThalitaNeural";
    AzureVoice["PT_BR_VALERIO_NEURAL"] = "pt-BR-ValerioNeural";
    AzureVoice["PT_BR_YARA_NEURAL"] = "pt-BR-YaraNeural";
    AzureVoice["PT_PT_RAQUEL_NEURAL"] = "pt-PT-RaquelNeural";
    AzureVoice["PT_PT_DUARTE_NEURAL"] = "pt-PT-DuarteNeural";
    AzureVoice["PT_PT_FERNANDA_NEURAL"] = "pt-PT-FernandaNeural";
    AzureVoice["RO_RO_ALINA_NEURAL"] = "ro-RO-AlinaNeural";
    AzureVoice["RO_RO_EMIL_NEURAL"] = "ro-RO-EmilNeural";
    AzureVoice["RU_RU_SVETLANA_NEURAL"] = "ru-RU-SvetlanaNeural";
    AzureVoice["RU_RU_DMITRY_NEURAL"] = "ru-RU-DmitryNeural";
    AzureVoice["RU_RU_DARIYA_NEURAL"] = "ru-RU-DariyaNeural";
    AzureVoice["SI_LK_THILINI_NEURAL"] = "si-LK-ThiliniNeural";
    AzureVoice["SI_LK_SAMEERA_NEURAL"] = "si-LK-SameeraNeural";
    AzureVoice["SK_SK_VIKTORIA_NEURAL"] = "sk-SK-ViktoriaNeural";
    AzureVoice["SK_SK_LUKAS_NEURAL"] = "sk-SK-LukasNeural";
    AzureVoice["SL_SI_PETRA_NEURAL"] = "sl-SI-PetraNeural";
    AzureVoice["SL_SI_ROK_NEURAL"] = "sl-SI-RokNeural";
    AzureVoice["SO_SO_UBAX_NEURAL"] = "so-SO-UbaxNeural";
    AzureVoice["SO_SO_MUUSE_NEURAL"] = "so-SO-MuuseNeural";
    AzureVoice["SQ_AL_ANILA_NEURAL"] = "sq-AL-AnilaNeural";
    AzureVoice["SQ_AL_ILIR_NEURAL"] = "sq-AL-IlirNeural";
    AzureVoice["SR_LATN_RS_NICHOLAS_NEURAL"] = "sr-Latn-RS-NicholasNeural";
    AzureVoice["SR_LATN_RS_SOPHIE_NEURAL"] = "sr-Latn-RS-SophieNeural";
    AzureVoice["SR_RS_SOPHIE_NEURAL"] = "sr-RS-SophieNeural";
    AzureVoice["SR_RS_NICHOLAS_NEURAL"] = "sr-RS-NicholasNeural";
    AzureVoice["SU_ID_TUTI_NEURAL"] = "su-ID-TutiNeural";
    AzureVoice["SU_ID_JAJANG_NEURAL"] = "su-ID-JajangNeural";
    AzureVoice["SV_SE_SOFIE_NEURAL"] = "sv-SE-SofieNeural";
    AzureVoice["SV_SE_MATTIAS_NEURAL"] = "sv-SE-MattiasNeural";
    AzureVoice["SV_SE_HILLEVI_NEURAL"] = "sv-SE-HilleviNeural";
    AzureVoice["SW_KE_ZURI_NEURAL"] = "sw-KE-ZuriNeural";
    AzureVoice["SW_KE_RAFIKI_NEURAL"] = "sw-KE-RafikiNeural";
    AzureVoice["SW_TZ_REHEMA_NEURAL"] = "sw-TZ-RehemaNeural";
    AzureVoice["SW_TZ_DAUDI_NEURAL"] = "sw-TZ-DaudiNeural";
    AzureVoice["TA_IN_PALLAVI_NEURAL"] = "ta-IN-PallaviNeural";
    AzureVoice["TA_IN_VALLUVAR_NEURAL"] = "ta-IN-ValluvarNeural";
    AzureVoice["TA_LK_SARANYA_NEURAL"] = "ta-LK-SaranyaNeural";
    AzureVoice["TA_LK_KUMAR_NEURAL"] = "ta-LK-KumarNeural";
    AzureVoice["TA_MY_KANI_NEURAL"] = "ta-MY-KaniNeural";
    AzureVoice["TA_MY_SURYA_NEURAL"] = "ta-MY-SuryaNeural";
    AzureVoice["TA_SG_VENBA_NEURAL"] = "ta-SG-VenbaNeural";
    AzureVoice["TA_SG_ANBU_NEURAL"] = "ta-SG-AnbuNeural";
    AzureVoice["TE_IN_SHRUTI_NEURAL"] = "te-IN-ShrutiNeural";
    AzureVoice["TE_IN_MOHAN_NEURAL"] = "te-IN-MohanNeural";
    AzureVoice["TH_TH_PREMWADEE_NEURAL"] = "th-TH-PremwadeeNeural";
    AzureVoice["TH_TH_NIWAT_NEURAL"] = "th-TH-NiwatNeural";
    AzureVoice["TH_TH_ACHARA_NEURAL"] = "th-TH-AcharaNeural";
    AzureVoice["TR_TR_EMEL_NEURAL"] = "tr-TR-EmelNeural";
    AzureVoice["TR_TR_AHMET_NEURAL"] = "tr-TR-AhmetNeural";
    AzureVoice["UK_UA_POLINA_NEURAL"] = "uk-UA-PolinaNeural";
    AzureVoice["UK_UA_OSTAP_NEURAL"] = "uk-UA-OstapNeural";
    AzureVoice["UR_IN_GUL_NEURAL"] = "ur-IN-GulNeural";
    AzureVoice["UR_IN_SALMAN_NEURAL"] = "ur-IN-SalmanNeural";
    AzureVoice["UR_PK_UZMA_NEURAL"] = "ur-PK-UzmaNeural";
    AzureVoice["UR_PK_ASAD_NEURAL"] = "ur-PK-AsadNeural";
    AzureVoice["UZ_UZ_MADINA_NEURAL"] = "uz-UZ-MadinaNeural";
    AzureVoice["UZ_UZ_SARDOR_NEURAL"] = "uz-UZ-SardorNeural";
    AzureVoice["VI_VN_HOAIMY_NEURAL"] = "vi-VN-HoaiMyNeural";
    AzureVoice["VI_VN_NAMMINH_NEURAL"] = "vi-VN-NamMinhNeural";
    AzureVoice["WUU_CN_XIAOTONG_NEURAL"] = "wuu-CN-XiaotongNeural";
    AzureVoice["WUU_CN_YUNZHE_NEURAL"] = "wuu-CN-YunzheNeural";
    AzureVoice["YUE_CN_XIAOMIN_NEURAL"] = "yue-CN-XiaoMinNeural";
    AzureVoice["YUE_CN_YUNSONG_NEURAL"] = "yue-CN-YunSongNeural";
    AzureVoice["ZH_CN_XIAOXIAO_NEURAL"] = "zh-CN-XiaoxiaoNeural";
    AzureVoice["ZH_CN_YUNXI_NEURAL"] = "zh-CN-YunxiNeural";
    AzureVoice["ZH_CN_YUNJIAN_NEURAL"] = "zh-CN-YunjianNeural";
    AzureVoice["ZH_CN_XIAOYI_NEURAL"] = "zh-CN-XiaoyiNeural";
    AzureVoice["ZH_CN_YUNYANG_NEURAL"] = "zh-CN-YunyangNeural";
    AzureVoice["ZH_CN_XIAOCHEN_NEURAL"] = "zh-CN-XiaochenNeural";
    AzureVoice["ZH_CN_XIAOCHENMULTILINGUAL_NEURAL"] = "zh-CN-XiaochenMultilingualNeural";
    AzureVoice["ZH_CN_XIAOHAN_NEURAL"] = "zh-CN-XiaohanNeural";
    AzureVoice["ZH_CN_XIAOMENG_NEURAL"] = "zh-CN-XiaomengNeural";
    AzureVoice["ZH_CN_XIAOMO_NEURAL"] = "zh-CN-XiaomoNeural";
    AzureVoice["ZH_CN_XIAOQIU_NEURAL"] = "zh-CN-XiaoqiuNeural";
    AzureVoice["ZH_CN_XIAOROU_NEURAL"] = "zh-CN-XiaorouNeural";
    AzureVoice["ZH_CN_XIAORUI_NEURAL"] = "zh-CN-XiaoruiNeural";
    AzureVoice["ZH_CN_XIAOSHUANG_NEURAL"] = "zh-CN-XiaoshuangNeural";
    AzureVoice["ZH_CN_XIAOXIAODIALECTS_NEURAL"] = "zh-CN-XiaoxiaoDialectsNeural";
    AzureVoice["ZH_CN_XIAOXIAOMULTILINGUAL_NEURAL"] = "zh-CN-XiaoxiaoMultilingualNeural";
    AzureVoice["ZH_CN_XIAOYAN_NEURAL"] = "zh-CN-XiaoyanNeural";
    AzureVoice["ZH_CN_XIAOYOU_NEURAL"] = "zh-CN-XiaoyouNeural";
    AzureVoice["ZH_CN_XIAOYUMULTILINGUAL_NEURAL"] = "zh-CN-XiaoyuMultilingualNeural";
    AzureVoice["ZH_CN_XIAOZHEN_NEURAL"] = "zh-CN-XiaozhenNeural";
    AzureVoice["ZH_CN_YUNFENG_NEURAL"] = "zh-CN-YunfengNeural";
    AzureVoice["ZH_CN_YUNHAO_NEURAL"] = "zh-CN-YunhaoNeural";
    AzureVoice["ZH_CN_YUNJIE_NEURAL"] = "zh-CN-YunjieNeural";
    AzureVoice["ZH_CN_YUNXIA_NEURAL"] = "zh-CN-YunxiaNeural";
    AzureVoice["ZH_CN_YUNYE_NEURAL"] = "zh-CN-YunyeNeural";
    AzureVoice["ZH_CN_YUNYIMULTILINGUAL_NEURAL"] = "zh-CN-YunyiMultilingualNeural";
    AzureVoice["ZH_CN_YUNZE_NEURAL"] = "zh-CN-YunzeNeural";
    AzureVoice["ZH_CN_GUANGXI_YUNQI_NEURAL"] = "zh-CN-guangxi-YunqiNeural";
    AzureVoice["ZH_CN_HENAN_YUNDENG_NEURAL"] = "zh-CN-henan-YundengNeural";
    AzureVoice["ZH_CN_LIAONING_XIAOBEI_NEURAL"] = "zh-CN-liaoning-XiaobeiNeural";
    AzureVoice["ZH_CN_LIAONING_YUNBIAO_NEURAL"] = "zh-CN-liaoning-YunbiaoNeural";
    AzureVoice["ZH_CN_SHAANXI_XIAONI_NEURAL"] = "zh-CN-shaanxi-XiaoniNeural";
    AzureVoice["ZH_CN_SHANDONG_YUNXIANG_NEURAL"] = "zh-CN-shandong-YunxiangNeural";
    AzureVoice["ZH_CN_SICHUAN_YUNXI_NEURAL"] = "zh-CN-sichuan-YunxiNeural";
    AzureVoice["ZH_HK_HIUMAAN_NEURAL"] = "zh-HK-HiuMaanNeural";
    AzureVoice["ZH_HK_WANLUNG_NEURAL"] = "zh-HK-WanLungNeural";
    AzureVoice["ZH_HK_HIUGAAI_NEURAL"] = "zh-HK-HiuGaaiNeural";
    AzureVoice["ZH_TW_HSIAOCHEN_NEURAL"] = "zh-TW-HsiaoChenNeural";
    AzureVoice["ZH_TW_YUNJHE_NEURAL"] = "zh-TW-YunJheNeural";
    AzureVoice["ZH_TW_HSIAOYU_NEURAL"] = "zh-TW-HsiaoYuNeural";
    AzureVoice["ZU_ZA_THANDO_NEURAL"] = "zu-ZA-ThandoNeural";
    AzureVoice["ZU_ZA_THEMBA_NEURAL"] = "zu-ZA-ThembaNeural";
})(AzureVoice || (exports.AzureVoice = AzureVoice = {}));
const AzureVoiceDetails = [
    [
        {
            name: AzureVoice.AF_ZA_ADRI_NEURAL,
            displayName: "Adri",
            languageCode: types_1.VoiceLanguage.AF_ZA,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AF_ZA_WILLEM_NEURAL,
            displayName: "Willem",
            languageCode: types_1.VoiceLanguage.AF_ZA,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AM_ET_MEKDES_NEURAL,
            displayName: "Mekdes",
            languageCode: types_1.VoiceLanguage.AM_ET,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AM_ET_AMEHA_NEURAL,
            displayName: "Ameha",
            languageCode: types_1.VoiceLanguage.AM_ET,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AR_AE_FATIMA_NEURAL,
            displayName: "Fatima",
            languageCode: types_1.VoiceLanguage.AR_AE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AR_AE_HAMDAN_NEURAL,
            displayName: "Hamdan",
            languageCode: types_1.VoiceLanguage.AR_AE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AR_BH_LAILA_NEURAL,
            displayName: "Laila",
            languageCode: types_1.VoiceLanguage.AR_BH,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AR_BH_ALI_NEURAL,
            displayName: "Ali",
            languageCode: types_1.VoiceLanguage.AR_BH,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AR_DZ_AMINA_NEURAL,
            displayName: "Amina",
            languageCode: types_1.VoiceLanguage.AR_DZ,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AR_DZ_ISMAEL_NEURAL,
            displayName: "Ismael",
            languageCode: types_1.VoiceLanguage.AR_DZ,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AR_EG_SALMA_NEURAL,
            displayName: "Salma",
            languageCode: types_1.VoiceLanguage.AR_EG,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AR_EG_SHAKIR_NEURAL,
            displayName: "Shakir",
            languageCode: types_1.VoiceLanguage.AR_EG,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AR_IQ_RANA_NEURAL,
            displayName: "Rana",
            languageCode: types_1.VoiceLanguage.AR_IQ,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AR_IQ_BASSEL_NEURAL,
            displayName: "Bassel",
            languageCode: types_1.VoiceLanguage.AR_IQ,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AR_JO_SANA_NEURAL,
            displayName: "Sana",
            languageCode: types_1.VoiceLanguage.AR_JO,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AR_JO_TAIM_NEURAL,
            displayName: "Taim",
            languageCode: types_1.VoiceLanguage.AR_JO,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AR_KW_NOURA_NEURAL,
            displayName: "Noura",
            languageCode: types_1.VoiceLanguage.AR_KW,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AR_KW_FAHED_NEURAL,
            displayName: "Fahed",
            languageCode: types_1.VoiceLanguage.AR_KW,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AR_LB_LAYLA_NEURAL,
            displayName: "Layla",
            languageCode: types_1.VoiceLanguage.AR_LB,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AR_LB_RAMI_NEURAL,
            displayName: "Rami",
            languageCode: types_1.VoiceLanguage.AR_LB,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AR_LY_IMAN_NEURAL,
            displayName: "Iman",
            languageCode: types_1.VoiceLanguage.AR_LY,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AR_LY_OMAR_NEURAL,
            displayName: "Omar",
            languageCode: types_1.VoiceLanguage.AR_LY,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AR_MA_MOUNA_NEURAL,
            displayName: "Mouna",
            languageCode: types_1.VoiceLanguage.AR_MA,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AR_MA_JAMAL_NEURAL,
            displayName: "Jamal",
            languageCode: types_1.VoiceLanguage.AR_MA,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AR_OM_AYSHA_NEURAL,
            displayName: "Aysha",
            languageCode: types_1.VoiceLanguage.AR_OM,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AR_OM_ABDULLAH_NEURAL,
            displayName: "Abdullah",
            languageCode: types_1.VoiceLanguage.AR_OM,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AR_QA_AMAL_NEURAL,
            displayName: "Amal",
            languageCode: types_1.VoiceLanguage.AR_QA,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AR_QA_MOAZ_NEURAL,
            displayName: "Moaz",
            languageCode: types_1.VoiceLanguage.AR_QA,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AR_SA_ZARIYAH_NEURAL,
            displayName: "Zariyah",
            languageCode: types_1.VoiceLanguage.AR_SA,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AR_SA_HAMED_NEURAL,
            displayName: "Hamed",
            languageCode: types_1.VoiceLanguage.AR_SA,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AR_SY_AMANY_NEURAL,
            displayName: "Amany",
            languageCode: types_1.VoiceLanguage.AR_SY,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AR_SY_LAITH_NEURAL,
            displayName: "Laith",
            languageCode: types_1.VoiceLanguage.AR_SY,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AR_TN_REEM_NEURAL,
            displayName: "Reem",
            languageCode: types_1.VoiceLanguage.AR_TN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AR_TN_HEDI_NEURAL,
            displayName: "Hedi",
            languageCode: types_1.VoiceLanguage.AR_TN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AR_YE_MARYAM_NEURAL,
            displayName: "Maryam",
            languageCode: types_1.VoiceLanguage.AR_YE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AR_YE_SALEH_NEURAL,
            displayName: "Saleh",
            languageCode: types_1.VoiceLanguage.AR_YE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AS_IN_PRIYOM_NEURAL,
            displayName: "Priyom",
            languageCode: types_1.VoiceLanguage.AS_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.AS_IN_YASHICA_NEURAL,
            displayName: "Yashica",
            languageCode: types_1.VoiceLanguage.AS_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AZ_AZ_BANU_NEURAL,
            displayName: "Banu",
            languageCode: types_1.VoiceLanguage.AZ_AZ,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.AZ_AZ_BABEK_NEURAL,
            displayName: "Babek",
            languageCode: types_1.VoiceLanguage.AZ_AZ,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.BG_BG_KALINA_NEURAL,
            displayName: "Kalina",
            languageCode: types_1.VoiceLanguage.BG_BG,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.BG_BG_BORISLAV_NEURAL,
            displayName: "Borislav",
            languageCode: types_1.VoiceLanguage.BG_BG,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.BN_BD_NABANITA_NEURAL,
            displayName: "Nabanita",
            languageCode: types_1.VoiceLanguage.BN_BD,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.BN_BD_PRADEEP_NEURAL,
            displayName: "Pradeep",
            languageCode: types_1.VoiceLanguage.BN_BD,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.BN_IN_TANISHAA_NEURAL,
            displayName: "Tanishaa",
            languageCode: types_1.VoiceLanguage.BN_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.BN_IN_BASHKAR_NEURAL,
            displayName: "Bashkar",
            languageCode: types_1.VoiceLanguage.BN_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.BS_BA_VESNA_NEURAL,
            displayName: "Vesna",
            languageCode: types_1.VoiceLanguage.BS_BA,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.BS_BA_GORAN_NEURAL,
            displayName: "Goran",
            languageCode: types_1.VoiceLanguage.BS_BA,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.CA_ES_JOANA_NEURAL,
            displayName: "Joana",
            languageCode: types_1.VoiceLanguage.CA_ES,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.CA_ES_ENRIC_NEURAL,
            displayName: "Enric",
            languageCode: types_1.VoiceLanguage.CA_ES,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.CA_ES_ALBA_NEURAL,
            displayName: "Alba",
            languageCode: types_1.VoiceLanguage.CA_ES,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.CS_CZ_VLASTA_NEURAL,
            displayName: "Vlasta",
            languageCode: types_1.VoiceLanguage.CS_CZ,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.CS_CZ_ANTONIN_NEURAL,
            displayName: "Antonin",
            languageCode: types_1.VoiceLanguage.CS_CZ,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.CY_GB_NIA_NEURAL,
            displayName: "Nia",
            languageCode: types_1.VoiceLanguage.CY_GB,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.CY_GB_ALED_NEURAL,
            displayName: "Aled",
            languageCode: types_1.VoiceLanguage.CY_GB,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.DA_DK_CHRISTEL_NEURAL,
            displayName: "Christel",
            languageCode: types_1.VoiceLanguage.DA_DK,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.DA_DK_JEPPE_NEURAL,
            displayName: "Jeppe",
            languageCode: types_1.VoiceLanguage.DA_DK,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.DE_AT_INGRID_NEURAL,
            displayName: "Ingrid",
            languageCode: types_1.VoiceLanguage.DE_AT,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.DE_AT_JONAS_NEURAL,
            displayName: "Jonas",
            languageCode: types_1.VoiceLanguage.DE_AT,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.DE_CH_LENI_NEURAL,
            displayName: "Leni",
            languageCode: types_1.VoiceLanguage.DE_CH,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.DE_CH_JAN_NEURAL,
            displayName: "Jan",
            languageCode: types_1.VoiceLanguage.DE_CH,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.DE_DE_KATJA_NEURAL,
            displayName: "Katja",
            languageCode: types_1.VoiceLanguage.DE_DE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.DE_DE_CONRAD_NEURAL,
            displayName: "Conrad",
            languageCode: types_1.VoiceLanguage.DE_DE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.DE_DE_AMALA_NEURAL,
            displayName: "Amala",
            languageCode: types_1.VoiceLanguage.DE_DE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.DE_DE_BERND_NEURAL,
            displayName: "Bernd",
            languageCode: types_1.VoiceLanguage.DE_DE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.DE_DE_CHRISTOPH_NEURAL,
            displayName: "Christoph",
            languageCode: types_1.VoiceLanguage.DE_DE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.DE_DE_ELKE_NEURAL,
            displayName: "Elke",
            languageCode: types_1.VoiceLanguage.DE_DE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.DE_DE_FLORIANMULTILINGUAL_NEURAL,
            displayName: "Florian Multilingual",
            languageCode: types_1.VoiceLanguage.DE_DE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.DE_DE_GISELA_NEURAL,
            displayName: "Gisela",
            languageCode: types_1.VoiceLanguage.DE_DE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.DE_DE_KASPER_NEURAL,
            displayName: "Kasper",
            languageCode: types_1.VoiceLanguage.DE_DE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.DE_DE_KILLIAN_NEURAL,
            displayName: "Killian",
            languageCode: types_1.VoiceLanguage.DE_DE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.DE_DE_KLARISSA_NEURAL,
            displayName: "Klarissa",
            languageCode: types_1.VoiceLanguage.DE_DE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.DE_DE_KLAUS_NEURAL,
            displayName: "Klaus",
            languageCode: types_1.VoiceLanguage.DE_DE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.DE_DE_LOUISA_NEURAL,
            displayName: "Louisa",
            languageCode: types_1.VoiceLanguage.DE_DE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.DE_DE_MAJA_NEURAL,
            displayName: "Maja",
            languageCode: types_1.VoiceLanguage.DE_DE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.DE_DE_RALF_NEURAL,
            displayName: "Ralf",
            languageCode: types_1.VoiceLanguage.DE_DE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.DE_DE_SERAPHINAMULTILINGUAL_NEURAL,
            displayName: "Seraphina Multilingual",
            languageCode: types_1.VoiceLanguage.DE_DE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.DE_DE_TANJA_NEURAL,
            displayName: "Tanja",
            languageCode: types_1.VoiceLanguage.DE_DE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EL_GR_ATHINA_NEURAL,
            displayName: "Athina",
            languageCode: types_1.VoiceLanguage.EL_GR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EL_GR_NESTORAS_NEURAL,
            displayName: "Nestoras",
            languageCode: types_1.VoiceLanguage.EL_GR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_AU_NATASHA_NEURAL,
            displayName: "Natasha",
            languageCode: types_1.VoiceLanguage.EN_AU,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_AU_WILLIAM_NEURAL,
            displayName: "William",
            languageCode: types_1.VoiceLanguage.EN_AU,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_AU_ANNETTE_NEURAL,
            displayName: "Annette",
            languageCode: types_1.VoiceLanguage.EN_AU,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_AU_CARLY_NEURAL,
            displayName: "Carly",
            languageCode: types_1.VoiceLanguage.EN_AU,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_AU_DARREN_NEURAL,
            displayName: "Darren",
            languageCode: types_1.VoiceLanguage.EN_AU,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_AU_DUNCAN_NEURAL,
            displayName: "Duncan",
            languageCode: types_1.VoiceLanguage.EN_AU,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_AU_ELSIE_NEURAL,
            displayName: "Elsie",
            languageCode: types_1.VoiceLanguage.EN_AU,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_AU_FREYA_NEURAL,
            displayName: "Freya",
            languageCode: types_1.VoiceLanguage.EN_AU,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_AU_JOANNE_NEURAL,
            displayName: "Joanne",
            languageCode: types_1.VoiceLanguage.EN_AU,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_AU_KEN_NEURAL,
            displayName: "Ken",
            languageCode: types_1.VoiceLanguage.EN_AU,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_AU_KIM_NEURAL,
            displayName: "Kim",
            languageCode: types_1.VoiceLanguage.EN_AU,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_AU_NEIL_NEURAL,
            displayName: "Neil",
            languageCode: types_1.VoiceLanguage.EN_AU,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_AU_TIM_NEURAL,
            displayName: "Tim",
            languageCode: types_1.VoiceLanguage.EN_AU,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_AU_TINA_NEURAL,
            displayName: "Tina",
            languageCode: types_1.VoiceLanguage.EN_AU,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_CA_CLARA_NEURAL,
            displayName: "Clara",
            languageCode: types_1.VoiceLanguage.EN_CA,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_CA_LIAM_NEURAL,
            displayName: "Liam",
            languageCode: types_1.VoiceLanguage.EN_CA,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_GB_SONIA_NEURAL,
            displayName: "Sonia",
            languageCode: types_1.VoiceLanguage.EN_GB,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_GB_RYAN_NEURAL,
            displayName: "Ryan",
            languageCode: types_1.VoiceLanguage.EN_GB,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_GB_LIBBY_NEURAL,
            displayName: "Libby",
            languageCode: types_1.VoiceLanguage.EN_GB,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_GB_ABBI_NEURAL,
            displayName: "Abbi",
            languageCode: types_1.VoiceLanguage.EN_GB,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_GB_ADAMULTILINGUAL_NEURAL,
            displayName: "Ada Multilingual",
            languageCode: types_1.VoiceLanguage.EN_GB,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_GB_ALFIE_NEURAL,
            displayName: "Alfie",
            languageCode: types_1.VoiceLanguage.EN_GB,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_GB_BELLA_NEURAL,
            displayName: "Bella",
            languageCode: types_1.VoiceLanguage.EN_GB,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_GB_ELLIOT_NEURAL,
            displayName: "Elliot",
            languageCode: types_1.VoiceLanguage.EN_GB,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_GB_ETHAN_NEURAL,
            displayName: "Ethan",
            languageCode: types_1.VoiceLanguage.EN_GB,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_GB_HOLLIE_NEURAL,
            displayName: "Hollie",
            languageCode: types_1.VoiceLanguage.EN_GB,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_GB_MAISIE_NEURAL,
            displayName: "Maisie",
            languageCode: types_1.VoiceLanguage.EN_GB,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_GB_NOAH_NEURAL,
            displayName: "Noah",
            languageCode: types_1.VoiceLanguage.EN_GB,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_GB_OLIVER_NEURAL,
            displayName: "Oliver",
            languageCode: types_1.VoiceLanguage.EN_GB,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_GB_OLIVIA_NEURAL,
            displayName: "Olivia",
            languageCode: types_1.VoiceLanguage.EN_GB,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_GB_THOMAS_NEURAL,
            displayName: "Thomas",
            languageCode: types_1.VoiceLanguage.EN_GB,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_GB_OLLIEMULTILINGUAL_NEURAL,
            displayName: "Ollie Multilingual",
            languageCode: types_1.VoiceLanguage.EN_GB,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_GB_MIA_NEURAL,
            displayName: "Mia",
            languageCode: types_1.VoiceLanguage.EN_GB,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_HK_YAN_NEURAL,
            displayName: "Yan",
            languageCode: types_1.VoiceLanguage.EN_HK,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_HK_SAM_NEURAL,
            displayName: "Sam",
            languageCode: types_1.VoiceLanguage.EN_HK,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_IE_EMILY_NEURAL,
            displayName: "Emily",
            languageCode: types_1.VoiceLanguage.EN_IE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_IE_CONNOR_NEURAL,
            displayName: "Connor",
            languageCode: types_1.VoiceLanguage.EN_IE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_IN_NEERJA_NEURAL,
            displayName: "Neerja",
            languageCode: types_1.VoiceLanguage.EN_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_IN_PRABHAT_NEURAL,
            displayName: "Prabhat",
            languageCode: types_1.VoiceLanguage.EN_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_IN_AARAV_NEURAL,
            displayName: "Aarav",
            languageCode: types_1.VoiceLanguage.EN_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_IN_AASHI_NEURAL,
            displayName: "Aashi",
            languageCode: types_1.VoiceLanguage.EN_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_IN_ANANYA_NEURAL,
            displayName: "Ananya",
            languageCode: types_1.VoiceLanguage.EN_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_IN_KAVYA_NEURAL,
            displayName: "Kavya",
            languageCode: types_1.VoiceLanguage.EN_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_IN_KUNAL_NEURAL,
            displayName: "Kunal",
            languageCode: types_1.VoiceLanguage.EN_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_IN_REHAAN_NEURAL,
            displayName: "Rehaan",
            languageCode: types_1.VoiceLanguage.EN_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_KE_ASILIA_NEURAL,
            displayName: "Asilia",
            languageCode: types_1.VoiceLanguage.EN_KE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_KE_CHILEMBA_NEURAL,
            displayName: "Chilemba",
            languageCode: types_1.VoiceLanguage.EN_KE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_NG_EZINNE_NEURAL,
            displayName: "Ezinne",
            languageCode: types_1.VoiceLanguage.EN_NG,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_NG_ABEO_NEURAL,
            displayName: "Abeo",
            languageCode: types_1.VoiceLanguage.EN_NG,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_NZ_MOLLY_NEURAL,
            displayName: "Molly",
            languageCode: types_1.VoiceLanguage.EN_NZ,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_NZ_MITCHELL_NEURAL,
            displayName: "Mitchell",
            languageCode: types_1.VoiceLanguage.EN_NZ,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_PH_ROSA_NEURAL,
            displayName: "Rosa",
            languageCode: types_1.VoiceLanguage.EN_PH,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_PH_JAMES_NEURAL,
            displayName: "James",
            languageCode: types_1.VoiceLanguage.EN_PH,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_SG_LUNA_NEURAL,
            displayName: "Luna",
            languageCode: types_1.VoiceLanguage.EN_SG,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_SG_WAYNE_NEURAL,
            displayName: "Wayne",
            languageCode: types_1.VoiceLanguage.EN_SG,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_TZ_IMANI_NEURAL,
            displayName: "Imani",
            languageCode: types_1.VoiceLanguage.EN_TZ,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_TZ_ELIMU_NEURAL,
            displayName: "Elimu",
            languageCode: types_1.VoiceLanguage.EN_TZ,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_AVAMULTILINGUAL_NEURAL,
            displayName: "Ava Multilingual",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_ANDREWMULTILINGUAL_NEURAL,
            displayName: "Andrew Multilingual",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_EMMAMULTILINGUAL_NEURAL,
            displayName: "Emma Multilingual",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_BRIANMULTILINGUAL_NEURAL,
            displayName: "Brian Multilingual",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_AVA_NEURAL,
            displayName: "Ava",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_ANDREW_NEURAL,
            displayName: "Andrew",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_EMMA_NEURAL,
            displayName: "Emma",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_BRIAN_NEURAL,
            displayName: "Brian",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_JENNY_NEURAL,
            displayName: "Jenny",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_GUY_NEURAL,
            displayName: "Guy",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_ARIA_NEURAL,
            displayName: "Aria",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_DAVIS_NEURAL,
            displayName: "Davis",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_JANE_NEURAL,
            displayName: "Jane",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_JASON_NEURAL,
            displayName: "Jason",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_SARA_NEURAL,
            displayName: "Sara",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_TONY_NEURAL,
            displayName: "Tony",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_NANCY_NEURAL,
            displayName: "Nancy",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_AMBER_NEURAL,
            displayName: "Amber",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_ANA_NEURAL,
            displayName: "Ana",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_ASHLEY_NEURAL,
            displayName: "Ashley",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_BRANDON_NEURAL,
            displayName: "Brandon",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_CHRISTOPHER_NEURAL,
            displayName: "Christopher",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_CORA_NEURAL,
            displayName: "Cora",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_ELIZABETH_NEURAL,
            displayName: "Elizabeth",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_ERIC_NEURAL,
            displayName: "Eric",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_JACOB_NEURAL,
            displayName: "Jacob",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_JENNYMULTILINGUAL_NEURAL,
            displayName: "Jenny Multilingual",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_MICHELLE_NEURAL,
            displayName: "Michelle",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_MONICA_NEURAL,
            displayName: "Monica",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_ROGER_NEURAL,
            displayName: "Roger",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_RYANMULTILINGUAL_NEURAL,
            displayName: "Ryan Multilingual",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_STEFFAN_NEURAL,
            displayName: "Steffan",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_AIGENERATE1_NEURAL,
            displayName: "AIGenerate1",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_AIGENERATE2_NEURAL,
            displayName: "AIGenerate2",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_US_BLUE_NEURAL,
            displayName: "Blue",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.NEUTRAL
        },
        {
            name: AzureVoice.EN_US_KAI_NEURAL,
            displayName: "Kai",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EN_US_LUNA_NEURAL,
            displayName: "Luna",
            languageCode: types_1.VoiceLanguage.EN_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_ZA_LEAH_NEURAL,
            displayName: "Leah",
            languageCode: types_1.VoiceLanguage.EN_ZA,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EN_ZA_LUKE_NEURAL,
            displayName: "Luke",
            languageCode: types_1.VoiceLanguage.EN_ZA,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_AR_ELENA_NEURAL,
            displayName: "Elena",
            languageCode: types_1.VoiceLanguage.ES_AR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_AR_TOMAS_NEURAL,
            displayName: "Tomas",
            languageCode: types_1.VoiceLanguage.ES_AR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_BO_SOFIA_NEURAL,
            displayName: "Sofia",
            languageCode: types_1.VoiceLanguage.ES_BO,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_BO_MARCELO_NEURAL,
            displayName: "Marcelo",
            languageCode: types_1.VoiceLanguage.ES_BO,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_CL_CATALINA_NEURAL,
            displayName: "Catalina",
            languageCode: types_1.VoiceLanguage.ES_CL,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_CL_LORENZO_NEURAL,
            displayName: "Lorenzo",
            languageCode: types_1.VoiceLanguage.ES_CL,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_CO_SALOME_NEURAL,
            displayName: "Salome",
            languageCode: types_1.VoiceLanguage.ES_CO,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_CO_GONZALO_NEURAL,
            displayName: "Gonzalo",
            languageCode: types_1.VoiceLanguage.ES_CO,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_CR_MARIA_NEURAL,
            displayName: "Maria",
            languageCode: types_1.VoiceLanguage.ES_CR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_CR_JUAN_NEURAL,
            displayName: "Juan",
            languageCode: types_1.VoiceLanguage.ES_CR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_CU_BELKYS_NEURAL,
            displayName: "Belkys",
            languageCode: types_1.VoiceLanguage.ES_CU,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_CU_MANUEL_NEURAL,
            displayName: "Manuel",
            languageCode: types_1.VoiceLanguage.ES_CU,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_DO_RAMONA_NEURAL,
            displayName: "Ramona",
            languageCode: types_1.VoiceLanguage.ES_DO,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_DO_EMILIO_NEURAL,
            displayName: "Emilio",
            languageCode: types_1.VoiceLanguage.ES_DO,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_EC_ANDREA_NEURAL,
            displayName: "Andrea",
            languageCode: types_1.VoiceLanguage.ES_EC,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_EC_LUIS_NEURAL,
            displayName: "Luis",
            languageCode: types_1.VoiceLanguage.ES_EC,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_ES_ELVIRA_NEURAL,
            displayName: "Elvira",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_ES_ALVARO_NEURAL,
            displayName: "Alvaro",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_ES_ABRIL_NEURAL,
            displayName: "Abril",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_ES_ARNAU_NEURAL,
            displayName: "Arnau",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_ES_DARIO_NEURAL,
            displayName: "Dario",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_ES_ELIAS_NEURAL,
            displayName: "Elias",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_ES_ESTRELLA_NEURAL,
            displayName: "Estrella",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_ES_IRENE_NEURAL,
            displayName: "Irene",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_ES_LAIA_NEURAL,
            displayName: "Laia",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_ES_LIA_NEURAL,
            displayName: "Lia",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_ES_NIL_NEURAL,
            displayName: "Nil",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_ES_SAUL_NEURAL,
            displayName: "Saul",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_ES_TEO_NEURAL,
            displayName: "Teo",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_ES_TRIANA_NEURAL,
            displayName: "Triana",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_ES_VERA_NEURAL,
            displayName: "Vera",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_ES_XIMENA_NEURAL,
            displayName: "Ximena",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_ES_ARABELLAMULTILINGUAL_NEURAL,
            displayName: "Arabella Multilingual",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_ES_ISIDORAMULTILINGUAL_NEURAL,
            displayName: "Isidora Multilingual",
            languageCode: types_1.VoiceLanguage.ES_ES,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_GQ_TERESA_NEURAL,
            displayName: "Teresa",
            languageCode: types_1.VoiceLanguage.ES_GQ,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_GQ_JAVIER_NEURAL,
            displayName: "Javier",
            languageCode: types_1.VoiceLanguage.ES_GQ,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_GT_MARTA_NEURAL,
            displayName: "Marta",
            languageCode: types_1.VoiceLanguage.ES_GT,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_GT_ANDRES_NEURAL,
            displayName: "Andres",
            languageCode: types_1.VoiceLanguage.ES_GT,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_HN_KARLA_NEURAL,
            displayName: "Karla",
            languageCode: types_1.VoiceLanguage.ES_HN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_HN_CARLOS_NEURAL,
            displayName: "Carlos",
            languageCode: types_1.VoiceLanguage.ES_HN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_MX_DALIA_NEURAL,
            displayName: "Dalia",
            languageCode: types_1.VoiceLanguage.ES_MX,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_MX_JORGE_NEURAL,
            displayName: "Jorge",
            languageCode: types_1.VoiceLanguage.ES_MX,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_MX_BEATRIZ_NEURAL,
            displayName: "Beatriz",
            languageCode: types_1.VoiceLanguage.ES_MX,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_MX_CANDELA_NEURAL,
            displayName: "Candela",
            languageCode: types_1.VoiceLanguage.ES_MX,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_MX_CARLOTA_NEURAL,
            displayName: "Carlota",
            languageCode: types_1.VoiceLanguage.ES_MX,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_MX_CECILIO_NEURAL,
            displayName: "Cecilio",
            languageCode: types_1.VoiceLanguage.ES_MX,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_MX_GERARDO_NEURAL,
            displayName: "Gerardo",
            languageCode: types_1.VoiceLanguage.ES_MX,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_MX_LARISSA_NEURAL,
            displayName: "Larissa",
            languageCode: types_1.VoiceLanguage.ES_MX,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_MX_LIBERTO_NEURAL,
            displayName: "Liberto",
            languageCode: types_1.VoiceLanguage.ES_MX,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_MX_LUCIANO_NEURAL,
            displayName: "Luciano",
            languageCode: types_1.VoiceLanguage.ES_MX,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_MX_MARINA_NEURAL,
            displayName: "Marina",
            languageCode: types_1.VoiceLanguage.ES_MX,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_MX_NURIA_NEURAL,
            displayName: "Nuria",
            languageCode: types_1.VoiceLanguage.ES_MX,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_MX_PELAYO_NEURAL,
            displayName: "Pelayo",
            languageCode: types_1.VoiceLanguage.ES_MX,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_MX_RENATA_NEURAL,
            displayName: "Renata",
            languageCode: types_1.VoiceLanguage.ES_MX,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_MX_YAGO_NEURAL,
            displayName: "Yago",
            languageCode: types_1.VoiceLanguage.ES_MX,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_NI_YOLANDA_NEURAL,
            displayName: "Yolanda",
            languageCode: types_1.VoiceLanguage.ES_NI,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_NI_FEDERICO_NEURAL,
            displayName: "Federico",
            languageCode: types_1.VoiceLanguage.ES_NI,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_PA_MARGARITA_NEURAL,
            displayName: "Margarita",
            languageCode: types_1.VoiceLanguage.ES_PA,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_PA_ROBERTO_NEURAL,
            displayName: "Roberto",
            languageCode: types_1.VoiceLanguage.ES_PA,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_PE_CAMILA_NEURAL,
            displayName: "Camila",
            languageCode: types_1.VoiceLanguage.ES_PE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_PE_ALEX_NEURAL,
            displayName: "Alex",
            languageCode: types_1.VoiceLanguage.ES_PE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_PR_KARINA_NEURAL,
            displayName: "Karina",
            languageCode: types_1.VoiceLanguage.ES_PR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_PR_VICTOR_NEURAL,
            displayName: "Victor",
            languageCode: types_1.VoiceLanguage.ES_PR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_PY_TANIA_NEURAL,
            displayName: "Tania",
            languageCode: types_1.VoiceLanguage.ES_PY,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_PY_MARIO_NEURAL,
            displayName: "Mario",
            languageCode: types_1.VoiceLanguage.ES_PY,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_SV_LORENA_NEURAL,
            displayName: "Lorena",
            languageCode: types_1.VoiceLanguage.ES_SV,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_SV_RODRIGO_NEURAL,
            displayName: "Rodrigo",
            languageCode: types_1.VoiceLanguage.ES_SV,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_US_PALOMA_NEURAL,
            displayName: "Paloma",
            languageCode: types_1.VoiceLanguage.ES_US,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_US_ALONSO_NEURAL,
            displayName: "Alonso",
            languageCode: types_1.VoiceLanguage.ES_US,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_UY_VALENTINA_NEURAL,
            displayName: "Valentina",
            languageCode: types_1.VoiceLanguage.ES_UY,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_UY_MATEO_NEURAL,
            displayName: "Mateo",
            languageCode: types_1.VoiceLanguage.ES_UY,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ES_VE_PAOLA_NEURAL,
            displayName: "Paola",
            languageCode: types_1.VoiceLanguage.ES_VE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ES_VE_SEBASTIAN_NEURAL,
            displayName: "Sebastian",
            languageCode: types_1.VoiceLanguage.ES_VE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ET_EE_ANU_NEURAL,
            displayName: "Anu",
            languageCode: types_1.VoiceLanguage.ET_EE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ET_EE_KERT_NEURAL,
            displayName: "Kert",
            languageCode: types_1.VoiceLanguage.ET_EE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.EU_ES_AINHOA_NEURAL,
            displayName: "Ainhoa",
            languageCode: types_1.VoiceLanguage.EU_ES,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.EU_ES_ANDER_NEURAL,
            displayName: "Ander",
            languageCode: types_1.VoiceLanguage.EU_ES,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.FA_IR_DILARA_NEURAL,
            displayName: "Dilara",
            languageCode: types_1.VoiceLanguage.FA_IR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.FA_IR_FARID_NEURAL,
            displayName: "Farid",
            languageCode: types_1.VoiceLanguage.FA_IR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.FI_FI_SELMA_NEURAL,
            displayName: "Selma",
            languageCode: types_1.VoiceLanguage.FI_FI,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.FI_FI_HARRI_NEURAL,
            displayName: "Harri",
            languageCode: types_1.VoiceLanguage.FI_FI,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.FI_FI_NOORA_NEURAL,
            displayName: "Noora",
            languageCode: types_1.VoiceLanguage.FI_FI,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.FIL_PH_BLESSICA_NEURAL,
            displayName: "Blessica",
            languageCode: types_1.VoiceLanguage.FIL_PH,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.FIL_PH_ANGELO_NEURAL,
            displayName: "Angelo",
            languageCode: types_1.VoiceLanguage.FIL_PH,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.FR_BE_CHARLINE_NEURAL,
            displayName: "Charline",
            languageCode: types_1.VoiceLanguage.FR_BE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.FR_BE_GERARD_NEURAL,
            displayName: "Gerard",
            languageCode: types_1.VoiceLanguage.FR_BE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.FR_CA_SYLVIE_NEURAL,
            displayName: "Sylvie",
            languageCode: types_1.VoiceLanguage.FR_CA,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.FR_CA_JEAN_NEURAL,
            displayName: "Jean",
            languageCode: types_1.VoiceLanguage.FR_CA,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.FR_CA_ANTOINE_NEURAL,
            displayName: "Antoine",
            languageCode: types_1.VoiceLanguage.FR_CA,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.FR_CA_THIERRY_NEURAL,
            displayName: "Thierry",
            languageCode: types_1.VoiceLanguage.FR_CA,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.FR_CH_ARIANE_NEURAL,
            displayName: "Ariane",
            languageCode: types_1.VoiceLanguage.FR_CH,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.FR_CH_FABRICE_NEURAL,
            displayName: "Fabrice",
            languageCode: types_1.VoiceLanguage.FR_CH,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.FR_FR_DENISE_NEURAL,
            displayName: "Denise",
            languageCode: types_1.VoiceLanguage.FR_FR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.FR_FR_HENRI_NEURAL,
            displayName: "Henri",
            languageCode: types_1.VoiceLanguage.FR_FR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.FR_FR_ALAIN_NEURAL,
            displayName: "Alain",
            languageCode: types_1.VoiceLanguage.FR_FR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.FR_FR_BRIGITTE_NEURAL,
            displayName: "Brigitte",
            languageCode: types_1.VoiceLanguage.FR_FR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.FR_FR_CELESTE_NEURAL,
            displayName: "Celeste",
            languageCode: types_1.VoiceLanguage.FR_FR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.FR_FR_CLAUDE_NEURAL,
            displayName: "Claude",
            languageCode: types_1.VoiceLanguage.FR_FR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.FR_FR_CORALIE_NEURAL,
            displayName: "Coralie",
            languageCode: types_1.VoiceLanguage.FR_FR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.FR_FR_ELOISE_NEURAL,
            displayName: "Eloise",
            languageCode: types_1.VoiceLanguage.FR_FR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.FR_FR_JACQUELINE_NEURAL,
            displayName: "Jacqueline",
            languageCode: types_1.VoiceLanguage.FR_FR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.FR_FR_JEROME_NEURAL,
            displayName: "Jerome",
            languageCode: types_1.VoiceLanguage.FR_FR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.FR_FR_JOSEPHINE_NEURAL,
            displayName: "Josephine",
            languageCode: types_1.VoiceLanguage.FR_FR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.FR_FR_MAURICE_NEURAL,
            displayName: "Maurice",
            languageCode: types_1.VoiceLanguage.FR_FR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.FR_FR_REMYMULTILINGUAL_NEURAL,
            displayName: "Remy Multilingual",
            languageCode: types_1.VoiceLanguage.FR_FR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.FR_FR_VIVIENNEMULTILINGUAL_NEURAL,
            displayName: "Vivienne Multilingual",
            languageCode: types_1.VoiceLanguage.FR_FR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.FR_FR_YVES_NEURAL,
            displayName: "Yves",
            languageCode: types_1.VoiceLanguage.FR_FR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.FR_FR_YVETTE_NEURAL,
            displayName: "Yvette",
            languageCode: types_1.VoiceLanguage.FR_FR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.GA_IE_ORLA_NEURAL,
            displayName: "Orla",
            languageCode: types_1.VoiceLanguage.GA_IE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.GA_IE_COLM_NEURAL,
            displayName: "Colm",
            languageCode: types_1.VoiceLanguage.GA_IE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.GL_ES_SABELA_NEURAL,
            displayName: "Sabela",
            languageCode: types_1.VoiceLanguage.GL_ES,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.GL_ES_ROI_NEURAL,
            displayName: "Roi",
            languageCode: types_1.VoiceLanguage.GL_ES,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.GU_IN_DHWANI_NEURAL,
            displayName: "Dhwani",
            languageCode: types_1.VoiceLanguage.GU_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.GU_IN_NIRANJAN_NEURAL,
            displayName: "Niranjan",
            languageCode: types_1.VoiceLanguage.GU_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.HE_IL_HILA_NEURAL,
            displayName: "Hila",
            languageCode: types_1.VoiceLanguage.HE_IL,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.HE_IL_AVRI_NEURAL,
            displayName: "Avri",
            languageCode: types_1.VoiceLanguage.HE_IL,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.HI_IN_SWARA_NEURAL,
            displayName: "Swara",
            languageCode: types_1.VoiceLanguage.HI_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.HI_IN_MADHUR_NEURAL,
            displayName: "Madhur",
            languageCode: types_1.VoiceLanguage.HI_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.HI_IN_AARAV_NEURAL,
            displayName: "Aarav",
            languageCode: types_1.VoiceLanguage.HI_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.HI_IN_ANANYA_NEURAL,
            displayName: "Ananya",
            languageCode: types_1.VoiceLanguage.HI_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.HI_IN_KAVYA_NEURAL,
            displayName: "Kavya",
            languageCode: types_1.VoiceLanguage.HI_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.HI_IN_KUNAL_NEURAL,
            displayName: "Kunal",
            languageCode: types_1.VoiceLanguage.HI_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.HI_IN_REHAAN_NEURAL,
            displayName: "Rehaan",
            languageCode: types_1.VoiceLanguage.HI_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.HR_HR_GABRIJELA_NEURAL,
            displayName: "Gabrijela",
            languageCode: types_1.VoiceLanguage.HR_HR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.HR_HR_SRECKO_NEURAL,
            displayName: "Srecko",
            languageCode: types_1.VoiceLanguage.HR_HR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.HU_HU_NOEMI_NEURAL,
            displayName: "Noemi",
            languageCode: types_1.VoiceLanguage.HU_HU,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.HU_HU_TAMAS_NEURAL,
            displayName: "Tamas",
            languageCode: types_1.VoiceLanguage.HU_HU,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.HY_AM_ANAHIT_NEURAL,
            displayName: "Anahit",
            languageCode: types_1.VoiceLanguage.HY_AM,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.HY_AM_HAYK_NEURAL,
            displayName: "Hayk",
            languageCode: types_1.VoiceLanguage.HY_AM,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ID_ID_GADIS_NEURAL,
            displayName: "Gadis",
            languageCode: types_1.VoiceLanguage.ID_ID,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ID_ID_ARDI_NEURAL,
            displayName: "Ardi",
            languageCode: types_1.VoiceLanguage.ID_ID,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.IS_IS_GUDRUN_NEURAL,
            displayName: "Gudrun",
            languageCode: types_1.VoiceLanguage.IS_IS,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.IS_IS_GUNNAR_NEURAL,
            displayName: "Gunnar",
            languageCode: types_1.VoiceLanguage.IS_IS,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.IT_IT_ELSA_NEURAL,
            displayName: "Elsa",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.IT_IT_ISABELLA_NEURAL,
            displayName: "Isabella",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.IT_IT_DIEGO_NEURAL,
            displayName: "Diego",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.IT_IT_BENIGNO_NEURAL,
            displayName: "Benigno",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.IT_IT_CALIMERO_NEURAL,
            displayName: "Calimero",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.IT_IT_CATALDO_NEURAL,
            displayName: "Cataldo",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.IT_IT_FABIOLA_NEURAL,
            displayName: "Fabiola",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.IT_IT_FIAMMA_NEURAL,
            displayName: "Fiamma",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.IT_IT_GIANNI_NEURAL,
            displayName: "Gianni",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.IT_IT_GIUSEPPE_NEURAL,
            displayName: "Giuseppe",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.IT_IT_IMELDA_NEURAL,
            displayName: "Imelda",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.IT_IT_IRMA_NEURAL,
            displayName: "Irma",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.IT_IT_LISANDRO_NEURAL,
            displayName: "Lisandro",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.IT_IT_PALMIRA_NEURAL,
            displayName: "Palmira",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.IT_IT_PIERINA_NEURAL,
            displayName: "Pierina",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.IT_IT_RINALDO_NEURAL,
            displayName: "Rinaldo",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.IT_IT_ALESSIOMULTILINGUAL_NEURAL,
            displayName: "Alessio Multilingual",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.IT_IT_ISABELLAMULTILINGUAL_NEURAL,
            displayName: "Isabella Multilingual",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.IT_IT_MARCELLOMULTILINGUAL_NEURAL,
            displayName: "Marcello Multilingual",
            languageCode: types_1.VoiceLanguage.IT_IT,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.JA_JP_NANAMI_NEURAL,
            displayName: "Nanami",
            languageCode: types_1.VoiceLanguage.JA_JP,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.JA_JP_KEITA_NEURAL,
            displayName: "Keita",
            languageCode: types_1.VoiceLanguage.JA_JP,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.JA_JP_AOI_NEURAL,
            displayName: "Aoi",
            languageCode: types_1.VoiceLanguage.JA_JP,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.JA_JP_DAICHI_NEURAL,
            displayName: "Daichi",
            languageCode: types_1.VoiceLanguage.JA_JP,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.JA_JP_MAYU_NEURAL,
            displayName: "Mayu",
            languageCode: types_1.VoiceLanguage.JA_JP,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.JA_JP_NAOKI_NEURAL,
            displayName: "Naoki",
            languageCode: types_1.VoiceLanguage.JA_JP,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.JA_JP_SHIORI_NEURAL,
            displayName: "Shiori",
            languageCode: types_1.VoiceLanguage.JA_JP,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.JA_JP_MASARUMULTILINGUAL_NEURAL,
            displayName: "Masaru Multilingual",
            languageCode: types_1.VoiceLanguage.JA_JP,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.JV_ID_SITI_NEURAL,
            displayName: "Siti",
            languageCode: types_1.VoiceLanguage.JV_ID,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.JV_ID_DIMAS_NEURAL,
            displayName: "Dimas",
            languageCode: types_1.VoiceLanguage.JV_ID,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.KA_GE_EKA_NEURAL,
            displayName: "Eka",
            languageCode: types_1.VoiceLanguage.KA_GE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.KA_GE_GIORGI_NEURAL,
            displayName: "Giorgi",
            languageCode: types_1.VoiceLanguage.KA_GE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.KK_KZ_AIGUL_NEURAL,
            displayName: "Aigul",
            languageCode: types_1.VoiceLanguage.KK_KZ,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.KK_KZ_DAULET_NEURAL,
            displayName: "Daulet",
            languageCode: types_1.VoiceLanguage.KK_KZ,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.KM_KH_SREYMOM_NEURAL,
            displayName: "Sreymom",
            languageCode: types_1.VoiceLanguage.KM_KH,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.KM_KH_PISETH_NEURAL,
            displayName: "Piseth",
            languageCode: types_1.VoiceLanguage.KM_KH,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.KN_IN_SAPNA_NEURAL,
            displayName: "Sapna",
            languageCode: types_1.VoiceLanguage.KN_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.KN_IN_GAGAN_NEURAL,
            displayName: "Gagan",
            languageCode: types_1.VoiceLanguage.KN_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.KO_KR_SUNHI_NEURAL,
            displayName: "Sun-Hi",
            languageCode: types_1.VoiceLanguage.KO_KR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.KO_KR_INJOON_NEURAL,
            displayName: "InJoon",
            languageCode: types_1.VoiceLanguage.KO_KR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.KO_KR_BONGJIN_NEURAL,
            displayName: "BongJin",
            languageCode: types_1.VoiceLanguage.KO_KR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.KO_KR_GOOKMIN_NEURAL,
            displayName: "GookMin",
            languageCode: types_1.VoiceLanguage.KO_KR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.KO_KR_HYUNSU_NEURAL,
            displayName: "Hyunsu",
            languageCode: types_1.VoiceLanguage.KO_KR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.KO_KR_JIMIN_NEURAL,
            displayName: "JiMin",
            languageCode: types_1.VoiceLanguage.KO_KR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.KO_KR_SEOHYEON_NEURAL,
            displayName: "SeoHyeon",
            languageCode: types_1.VoiceLanguage.KO_KR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.KO_KR_SOONBOK_NEURAL,
            displayName: "SoonBok",
            languageCode: types_1.VoiceLanguage.KO_KR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.KO_KR_YUJIN_NEURAL,
            displayName: "YuJin",
            languageCode: types_1.VoiceLanguage.KO_KR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.LO_LA_KEOMANY_NEURAL,
            displayName: "Keomany",
            languageCode: types_1.VoiceLanguage.LO_LA,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.LO_LA_CHANTHAVONG_NEURAL,
            displayName: "Chanthavong",
            languageCode: types_1.VoiceLanguage.LO_LA,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.LT_LT_ONA_NEURAL,
            displayName: "Ona",
            languageCode: types_1.VoiceLanguage.LT_LT,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.LT_LT_LEONAS_NEURAL,
            displayName: "Leonas",
            languageCode: types_1.VoiceLanguage.LT_LT,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.LV_LV_EVERITA_NEURAL,
            displayName: "Everita",
            languageCode: types_1.VoiceLanguage.LV_LV,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.LV_LV_NILS_NEURAL,
            displayName: "Nils",
            languageCode: types_1.VoiceLanguage.LV_LV,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.MK_MK_MARIJA_NEURAL,
            displayName: "Marija",
            languageCode: types_1.VoiceLanguage.MK_MK,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.MK_MK_ALEKSANDAR_NEURAL,
            displayName: "Aleksandar",
            languageCode: types_1.VoiceLanguage.MK_MK,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ML_IN_SOBHANA_NEURAL,
            displayName: "Sobhana",
            languageCode: types_1.VoiceLanguage.ML_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ML_IN_MIDHUN_NEURAL,
            displayName: "Midhun",
            languageCode: types_1.VoiceLanguage.ML_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.MN_MN_YESUI_NEURAL,
            displayName: "Yesui",
            languageCode: types_1.VoiceLanguage.MN_MN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.MN_MN_BATAA_NEURAL,
            displayName: "Bataa",
            languageCode: types_1.VoiceLanguage.MN_MN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.MR_IN_AAROHI_NEURAL,
            displayName: "Aarohi",
            languageCode: types_1.VoiceLanguage.MR_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.MR_IN_MANOHAR_NEURAL,
            displayName: "Manohar",
            languageCode: types_1.VoiceLanguage.MR_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.MS_MY_YASMIN_NEURAL,
            displayName: "Yasmin",
            languageCode: types_1.VoiceLanguage.MS_MY,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.MS_MY_OSMAN_NEURAL,
            displayName: "Osman",
            languageCode: types_1.VoiceLanguage.MS_MY,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.MT_MT_GRACE_NEURAL,
            displayName: "Grace",
            languageCode: types_1.VoiceLanguage.MT_MT,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.MT_MT_JOSEPH_NEURAL,
            displayName: "Joseph",
            languageCode: types_1.VoiceLanguage.MT_MT,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.MY_MM_NILAR_NEURAL,
            displayName: "Nilar",
            languageCode: types_1.VoiceLanguage.MY_MM,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.MY_MM_THIHA_NEURAL,
            displayName: "Thiha",
            languageCode: types_1.VoiceLanguage.MY_MM,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.NB_NO_PERNILLE_NEURAL,
            displayName: "Pernille",
            languageCode: types_1.VoiceLanguage.NB_NO,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.NB_NO_FINN_NEURAL,
            displayName: "Finn",
            languageCode: types_1.VoiceLanguage.NB_NO,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.NB_NO_ISELIN_NEURAL,
            displayName: "Iselin",
            languageCode: types_1.VoiceLanguage.NB_NO,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.NE_NP_HEMKALA_NEURAL,
            displayName: "Hemkala",
            languageCode: types_1.VoiceLanguage.NE_NP,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.NE_NP_SAGAR_NEURAL,
            displayName: "Sagar",
            languageCode: types_1.VoiceLanguage.NE_NP,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.NL_BE_DENA_NEURAL,
            displayName: "Dena",
            languageCode: types_1.VoiceLanguage.NL_BE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.NL_BE_ARNAUD_NEURAL,
            displayName: "Arnaud",
            languageCode: types_1.VoiceLanguage.NL_BE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.NL_NL_FENNA_NEURAL,
            displayName: "Fenna",
            languageCode: types_1.VoiceLanguage.NL_NL,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.NL_NL_MAARTEN_NEURAL,
            displayName: "Maarten",
            languageCode: types_1.VoiceLanguage.NL_NL,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.NL_NL_COLETTE_NEURAL,
            displayName: "Colette",
            languageCode: types_1.VoiceLanguage.NL_NL,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.OR_IN_SUBHASINI_NEURAL,
            displayName: "Subhasini",
            languageCode: types_1.VoiceLanguage.OR_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.OR_IN_SUKANT_NEURAL,
            displayName: "Sukant",
            languageCode: types_1.VoiceLanguage.OR_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.PA_IN_OJAS_NEURAL,
            displayName: "Ojas",
            languageCode: types_1.VoiceLanguage.PA_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.PA_IN_VAANI_NEURAL,
            displayName: "Vaani",
            languageCode: types_1.VoiceLanguage.PA_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.PL_PL_AGNIESZKA_NEURAL,
            displayName: "Agnieszka",
            languageCode: types_1.VoiceLanguage.PL_PL,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.PL_PL_MAREK_NEURAL,
            displayName: "Marek",
            languageCode: types_1.VoiceLanguage.PL_PL,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.PL_PL_ZOFIA_NEURAL,
            displayName: "Zofia",
            languageCode: types_1.VoiceLanguage.PL_PL,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.PS_AF_LATIFA_NEURAL,
            displayName: "Latifa",
            languageCode: types_1.VoiceLanguage.PS_AF,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.PS_AF_GULNAWAZ_NEURAL,
            displayName: "Gul Nawaz",
            languageCode: types_1.VoiceLanguage.PS_AF,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.PT_BR_FRANCISCA_NEURAL,
            displayName: "Francisca",
            languageCode: types_1.VoiceLanguage.PT_BR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.PT_BR_ANTONIO_NEURAL,
            displayName: "Antonio",
            languageCode: types_1.VoiceLanguage.PT_BR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.PT_BR_BRENDA_NEURAL,
            displayName: "Brenda",
            languageCode: types_1.VoiceLanguage.PT_BR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.PT_BR_DONATO_NEURAL,
            displayName: "Donato",
            languageCode: types_1.VoiceLanguage.PT_BR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.PT_BR_ELZA_NEURAL,
            displayName: "Elza",
            languageCode: types_1.VoiceLanguage.PT_BR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.PT_BR_FABIO_NEURAL,
            displayName: "Fabio",
            languageCode: types_1.VoiceLanguage.PT_BR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.PT_BR_GIOVANNA_NEURAL,
            displayName: "Giovanna",
            languageCode: types_1.VoiceLanguage.PT_BR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.PT_BR_HUMBERTO_NEURAL,
            displayName: "Humberto",
            languageCode: types_1.VoiceLanguage.PT_BR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.PT_BR_JULIO_NEURAL,
            displayName: "Julio",
            languageCode: types_1.VoiceLanguage.PT_BR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.PT_BR_LEILA_NEURAL,
            displayName: "Leila",
            languageCode: types_1.VoiceLanguage.PT_BR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.PT_BR_LETICIA_NEURAL,
            displayName: "Leticia",
            languageCode: types_1.VoiceLanguage.PT_BR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.PT_BR_MANUELA_NEURAL,
            displayName: "Manuela",
            languageCode: types_1.VoiceLanguage.PT_BR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.PT_BR_NICOLAU_NEURAL,
            displayName: "Nicolau",
            languageCode: types_1.VoiceLanguage.PT_BR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.PT_BR_THALITAMULTILINGUAL_NEURAL,
            displayName: "Thalita Multilingual",
            languageCode: types_1.VoiceLanguage.PT_BR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.PT_BR_THALITA_NEURAL,
            displayName: "Thalita",
            languageCode: types_1.VoiceLanguage.PT_BR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.PT_BR_VALERIO_NEURAL,
            displayName: "Valerio",
            languageCode: types_1.VoiceLanguage.PT_BR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.PT_BR_YARA_NEURAL,
            displayName: "Yara",
            languageCode: types_1.VoiceLanguage.PT_BR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.PT_PT_RAQUEL_NEURAL,
            displayName: "Raquel",
            languageCode: types_1.VoiceLanguage.PT_PT,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.PT_PT_DUARTE_NEURAL,
            displayName: "Duarte",
            languageCode: types_1.VoiceLanguage.PT_PT,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.PT_PT_FERNANDA_NEURAL,
            displayName: "Fernanda",
            languageCode: types_1.VoiceLanguage.PT_PT,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.RO_RO_ALINA_NEURAL,
            displayName: "Alina",
            languageCode: types_1.VoiceLanguage.RO_RO,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.RO_RO_EMIL_NEURAL,
            displayName: "Emil",
            languageCode: types_1.VoiceLanguage.RO_RO,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.RU_RU_SVETLANA_NEURAL,
            displayName: "Svetlana",
            languageCode: types_1.VoiceLanguage.RU_RU,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.RU_RU_DMITRY_NEURAL,
            displayName: "Dmitry",
            languageCode: types_1.VoiceLanguage.RU_RU,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.RU_RU_DARIYA_NEURAL,
            displayName: "Dariya",
            languageCode: types_1.VoiceLanguage.RU_RU,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.SI_LK_THILINI_NEURAL,
            displayName: "Thilini",
            languageCode: types_1.VoiceLanguage.SI_LK,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.SI_LK_SAMEERA_NEURAL,
            displayName: "Sameera",
            languageCode: types_1.VoiceLanguage.SI_LK,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.SK_SK_VIKTORIA_NEURAL,
            displayName: "Viktoria",
            languageCode: types_1.VoiceLanguage.SK_SK,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.SK_SK_LUKAS_NEURAL,
            displayName: "Lukas",
            languageCode: types_1.VoiceLanguage.SK_SK,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.SL_SI_PETRA_NEURAL,
            displayName: "Petra",
            languageCode: types_1.VoiceLanguage.SL_SI,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.SL_SI_ROK_NEURAL,
            displayName: "Rok",
            languageCode: types_1.VoiceLanguage.SL_SI,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.SO_SO_UBAX_NEURAL,
            displayName: "Ubax",
            languageCode: types_1.VoiceLanguage.SO_SO,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.SO_SO_MUUSE_NEURAL,
            displayName: "Muuse",
            languageCode: types_1.VoiceLanguage.SO_SO,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.SQ_AL_ANILA_NEURAL,
            displayName: "Anila",
            languageCode: types_1.VoiceLanguage.SQ_AL,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.SQ_AL_ILIR_NEURAL,
            displayName: "Ilir",
            languageCode: types_1.VoiceLanguage.SQ_AL,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.SR_LATN_RS_NICHOLAS_NEURAL,
            displayName: "Nicholas",
            languageCode: types_1.VoiceLanguage.SR_LATN_RS,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.SR_LATN_RS_SOPHIE_NEURAL,
            displayName: "Sophie",
            languageCode: types_1.VoiceLanguage.SR_LATN_RS,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.SR_RS_SOPHIE_NEURAL,
            displayName: "Sophie",
            languageCode: types_1.VoiceLanguage.SR_RS,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.SR_RS_NICHOLAS_NEURAL,
            displayName: "Nicholas",
            languageCode: types_1.VoiceLanguage.SR_RS,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.SU_ID_TUTI_NEURAL,
            displayName: "Tuti",
            languageCode: types_1.VoiceLanguage.SU_ID,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.SU_ID_JAJANG_NEURAL,
            displayName: "Jajang",
            languageCode: types_1.VoiceLanguage.SU_ID,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.SV_SE_SOFIE_NEURAL,
            displayName: "Sofie",
            languageCode: types_1.VoiceLanguage.SV_SE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.SV_SE_MATTIAS_NEURAL,
            displayName: "Mattias",
            languageCode: types_1.VoiceLanguage.SV_SE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.SV_SE_HILLEVI_NEURAL,
            displayName: "Hillevi",
            languageCode: types_1.VoiceLanguage.SV_SE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.SW_KE_ZURI_NEURAL,
            displayName: "Zuri",
            languageCode: types_1.VoiceLanguage.SW_KE,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.SW_KE_RAFIKI_NEURAL,
            displayName: "Rafiki",
            languageCode: types_1.VoiceLanguage.SW_KE,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.SW_TZ_REHEMA_NEURAL,
            displayName: "Rehema",
            languageCode: types_1.VoiceLanguage.SW_TZ,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.SW_TZ_DAUDI_NEURAL,
            displayName: "Daudi",
            languageCode: types_1.VoiceLanguage.SW_TZ,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.TA_IN_PALLAVI_NEURAL,
            displayName: "Pallavi",
            languageCode: types_1.VoiceLanguage.TA_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.TA_IN_VALLUVAR_NEURAL,
            displayName: "Valluvar",
            languageCode: types_1.VoiceLanguage.TA_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.TA_LK_SARANYA_NEURAL,
            displayName: "Saranya",
            languageCode: types_1.VoiceLanguage.TA_LK,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.TA_LK_KUMAR_NEURAL,
            displayName: "Kumar",
            languageCode: types_1.VoiceLanguage.TA_LK,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.TA_MY_KANI_NEURAL,
            displayName: "Kani",
            languageCode: types_1.VoiceLanguage.TA_MY,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.TA_MY_SURYA_NEURAL,
            displayName: "Surya",
            languageCode: types_1.VoiceLanguage.TA_MY,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.TA_SG_VENBA_NEURAL,
            displayName: "Venba",
            languageCode: types_1.VoiceLanguage.TA_SG,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.TA_SG_ANBU_NEURAL,
            displayName: "Anbu",
            languageCode: types_1.VoiceLanguage.TA_SG,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.TE_IN_SHRUTI_NEURAL,
            displayName: "Shruti",
            languageCode: types_1.VoiceLanguage.TE_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.TE_IN_MOHAN_NEURAL,
            displayName: "Mohan",
            languageCode: types_1.VoiceLanguage.TE_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.TH_TH_PREMWADEE_NEURAL,
            displayName: "Premwadee",
            languageCode: types_1.VoiceLanguage.TH_TH,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.TH_TH_NIWAT_NEURAL,
            displayName: "Niwat",
            languageCode: types_1.VoiceLanguage.TH_TH,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.TH_TH_ACHARA_NEURAL,
            displayName: "Achara",
            languageCode: types_1.VoiceLanguage.TH_TH,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.TR_TR_EMEL_NEURAL,
            displayName: "Emel",
            languageCode: types_1.VoiceLanguage.TR_TR,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.TR_TR_AHMET_NEURAL,
            displayName: "Ahmet",
            languageCode: types_1.VoiceLanguage.TR_TR,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.UK_UA_POLINA_NEURAL,
            displayName: "Polina",
            languageCode: types_1.VoiceLanguage.UK_UA,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.UK_UA_OSTAP_NEURAL,
            displayName: "Ostap",
            languageCode: types_1.VoiceLanguage.UK_UA,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.UR_IN_GUL_NEURAL,
            displayName: "Gul",
            languageCode: types_1.VoiceLanguage.UR_IN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.UR_IN_SALMAN_NEURAL,
            displayName: "Salman",
            languageCode: types_1.VoiceLanguage.UR_IN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.UR_PK_UZMA_NEURAL,
            displayName: "Uzma",
            languageCode: types_1.VoiceLanguage.UR_PK,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.UR_PK_ASAD_NEURAL,
            displayName: "Asad",
            languageCode: types_1.VoiceLanguage.UR_PK,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.UZ_UZ_MADINA_NEURAL,
            displayName: "Madina",
            languageCode: types_1.VoiceLanguage.UZ_UZ,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.UZ_UZ_SARDOR_NEURAL,
            displayName: "Sardor",
            languageCode: types_1.VoiceLanguage.UZ_UZ,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.VI_VN_HOAIMY_NEURAL,
            displayName: "HoaiMy",
            languageCode: types_1.VoiceLanguage.VI_VN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.VI_VN_NAMMINH_NEURAL,
            displayName: "NamMinh",
            languageCode: types_1.VoiceLanguage.VI_VN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.WUU_CN_XIAOTONG_NEURAL,
            displayName: "Xiaotong",
            languageCode: types_1.VoiceLanguage.WUU_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.WUU_CN_YUNZHE_NEURAL,
            displayName: "Yunzhe",
            languageCode: types_1.VoiceLanguage.WUU_CN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.YUE_CN_XIAOMIN_NEURAL,
            displayName: "XiaoMin",
            languageCode: types_1.VoiceLanguage.YUE_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.YUE_CN_YUNSONG_NEURAL,
            displayName: "YunSong",
            languageCode: types_1.VoiceLanguage.YUE_CN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_CN_XIAOXIAO_NEURAL,
            displayName: "Xiaoxiao",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_YUNXI_NEURAL,
            displayName: "Yunxi",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_CN_YUNJIAN_NEURAL,
            displayName: "Yunjian",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_CN_XIAOYI_NEURAL,
            displayName: "Xiaoyi",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_YUNYANG_NEURAL,
            displayName: "Yunyang",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_CN_XIAOCHEN_NEURAL,
            displayName: "Xiaochen",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_XIAOCHENMULTILINGUAL_NEURAL,
            displayName: "Xiaochen Multilingual",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_XIAOHAN_NEURAL,
            displayName: "Xiaohan",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_XIAOMENG_NEURAL,
            displayName: "Xiaomeng",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_XIAOMO_NEURAL,
            displayName: "Xiaomo",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_XIAOQIU_NEURAL,
            displayName: "Xiaoqiu",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_XIAOROU_NEURAL,
            displayName: "Xiaorou",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_XIAORUI_NEURAL,
            displayName: "Xiaorui",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_XIAOSHUANG_NEURAL,
            displayName: "Xiaoshuang",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_XIAOXIAODIALECTS_NEURAL,
            displayName: "Xiaoxiao Dialects",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_XIAOXIAOMULTILINGUAL_NEURAL,
            displayName: "Xiaoxiao Multilingual",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_XIAOYAN_NEURAL,
            displayName: "Xiaoyan",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_XIAOYOU_NEURAL,
            displayName: "Xiaoyou",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_XIAOYUMULTILINGUAL_NEURAL,
            displayName: "Xiaoyu Multilingual",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_XIAOZHEN_NEURAL,
            displayName: "Xiaozhen",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_YUNFENG_NEURAL,
            displayName: "Yunfeng",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_CN_YUNHAO_NEURAL,
            displayName: "Yunhao",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_CN_YUNJIE_NEURAL,
            displayName: "Yunjie",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_CN_YUNXIA_NEURAL,
            displayName: "Yunxia",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_CN_YUNYE_NEURAL,
            displayName: "Yunye",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_CN_YUNYIMULTILINGUAL_NEURAL,
            displayName: "Yunyi Multilingual",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_CN_YUNZE_NEURAL,
            displayName: "Yunze",
            languageCode: types_1.VoiceLanguage.ZH_CN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_CN_GUANGXI_YUNQI_NEURAL,
            displayName: "Yunqi",
            languageCode: types_1.VoiceLanguage.ZH_CN_GUANGXI,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_CN_HENAN_YUNDENG_NEURAL,
            displayName: "Yundeng",
            languageCode: types_1.VoiceLanguage.ZH_CN_HENAN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_CN_LIAONING_XIAOBEI_NEURAL,
            displayName: "Xiaobei",
            languageCode: types_1.VoiceLanguage.ZH_CN_LIAONING,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_LIAONING_YUNBIAO_NEURAL,
            displayName: "Yunbiao",
            languageCode: types_1.VoiceLanguage.ZH_CN_LIAONING,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_CN_SHAANXI_XIAONI_NEURAL,
            displayName: "Xiaoni",
            languageCode: types_1.VoiceLanguage.ZH_CN_SHAANXI,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_CN_SHANDONG_YUNXIANG_NEURAL,
            displayName: "Yunxiang",
            languageCode: types_1.VoiceLanguage.ZH_CN_SHANDONG,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_CN_SICHUAN_YUNXI_NEURAL,
            displayName: "Yunxi",
            languageCode: types_1.VoiceLanguage.ZH_CN_SICHUAN,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_HK_HIUMAAN_NEURAL,
            displayName: "HiuMaan",
            languageCode: types_1.VoiceLanguage.ZH_HK,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_HK_WANLUNG_NEURAL,
            displayName: "WanLung",
            languageCode: types_1.VoiceLanguage.ZH_HK,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_HK_HIUGAAI_NEURAL,
            displayName: "HiuGaai",
            languageCode: types_1.VoiceLanguage.ZH_HK,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_TW_HSIAOCHEN_NEURAL,
            displayName: "HsiaoChen",
            languageCode: types_1.VoiceLanguage.ZH_TW,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZH_TW_YUNJHE_NEURAL,
            displayName: "YunJhe",
            languageCode: types_1.VoiceLanguage.ZH_TW,
            gender: types_1.VoiceGender.MALE
        },
        {
            name: AzureVoice.ZH_TW_HSIAOYU_NEURAL,
            displayName: "HsiaoYu",
            languageCode: types_1.VoiceLanguage.ZH_TW,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZU_ZA_THANDO_NEURAL,
            displayName: "Thando",
            languageCode: types_1.VoiceLanguage.ZU_ZA,
            gender: types_1.VoiceGender.FEMALE
        },
        {
            name: AzureVoice.ZU_ZA_THEMBA_NEURAL,
            displayName: "Themba",
            languageCode: types_1.VoiceLanguage.ZU_ZA,
            gender: types_1.VoiceGender.MALE
        }
    ]
];
exports.AzureVoiceDetails = AzureVoiceDetails;
