// @fonoster/common/dist/tts/DeepgramVoices.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeepgramVoiceDetails = exports.DeepgramVoice = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const types_1 = require("./types");
var DeepgramVoice;
(function (DeepgramVoice) {
    // Aura 2 English Voices - Featured
    DeepgramVoice["AURA_2_THALIA_EN"] = "aura-2-thalia-en";
    DeepgramVoice["AURA_2_ANDROMEDA_EN"] = "aura-2-andromeda-en";
    DeepgramVoice["AURA_2_HELENA_EN"] = "aura-2-helena-en";
    DeepgramVoice["AURA_2_APOLLO_EN"] = "aura-2-apollo-en";
    DeepgramVoice["AURA_2_ARCAS_EN"] = "aura-2-arcas-en";
    DeepgramVoice["AURA_2_ARIES_EN"] = "aura-2-aries-en";
    // Aura 2 English Voices - All Available
    DeepgramVoice["AURA_2_AMALTHEA_EN"] = "aura-2-amalthea-en";
    DeepgramVoice["AURA_2_ASTERIA_EN"] = "aura-2-asteria-en";
    DeepgramVoice["AURA_2_ATHENA_EN"] = "aura-2-athena-en";
    DeepgramVoice["AURA_2_ATLAS_EN"] = "aura-2-atlas-en";
    DeepgramVoice["AURA_2_AURORA_EN"] = "aura-2-aurora-en";
    DeepgramVoice["AURA_2_CALLISTA_EN"] = "aura-2-callista-en";
    DeepgramVoice["AURA_2_CORA_EN"] = "aura-2-cora-en";
    DeepgramVoice["AURA_2_CORDELIA_EN"] = "aura-2-cordelia-en";
    DeepgramVoice["AURA_2_DELIA_EN"] = "aura-2-delia-en";
    DeepgramVoice["AURA_2_DRACO_EN"] = "aura-2-draco-en";
    DeepgramVoice["AURA_2_ELECTRA_EN"] = "aura-2-electra-en";
    DeepgramVoice["AURA_2_HARMONIA_EN"] = "aura-2-harmonia-en";
    DeepgramVoice["AURA_2_HERA_EN"] = "aura-2-hera-en";
    DeepgramVoice["AURA_2_IRIS_EN"] = "aura-2-iris-en";
    DeepgramVoice["AURA_2_JUNO_EN"] = "aura-2-juno-en";
    DeepgramVoice["AURA_2_LUNA_EN"] = "aura-2-luna-en";
    DeepgramVoice["AURA_2_MAIA_EN"] = "aura-2-maia-en";
    DeepgramVoice["AURA_2_NOVA_EN"] = "aura-2-nova-en";
    DeepgramVoice["AURA_2_ORION_EN"] = "aura-2-orion-en";
    DeepgramVoice["AURA_2_PANDORA_EN"] = "aura-2-pandora-en";
    DeepgramVoice["AURA_2_PHOEBE_EN"] = "aura-2-phoebe-en";
    DeepgramVoice["AURA_2_PLUTO_EN"] = "aura-2-pluto-en";
    DeepgramVoice["AURA_2_SATURN_EN"] = "aura-2-saturn-en";
    DeepgramVoice["AURA_2_SELENE_EN"] = "aura-2-selene-en";
    DeepgramVoice["AURA_2_THEIA_EN"] = "aura-2-theia-en";
    DeepgramVoice["AURA_2_VESTA_EN"] = "aura-2-vesta-en";
    DeepgramVoice["AURA_2_ZEUS_EN"] = "aura-2-zeus-en";
    // Aura 2 Spanish Voices - Featured
    DeepgramVoice["AURA_2_CELESTE_ES"] = "aura-2-celeste-es";
    DeepgramVoice["AURA_2_ESTRELLA_ES"] = "aura-2-estrella-es";
    DeepgramVoice["AURA_2_NESTOR_ES"] = "aura-2-nestor-es";
    // Aura 2 Spanish Voices - All Available
    DeepgramVoice["AURA_2_SIRIO_ES"] = "aura-2-sirio-es";
    DeepgramVoice["AURA_2_CARINA_ES"] = "aura-2-carina-es";
    DeepgramVoice["AURA_2_ALVARO_ES"] = "aura-2-alvaro-es";
    DeepgramVoice["AURA_2_DIANA_ES"] = "aura-2-diana-es";
    DeepgramVoice["AURA_2_AQUILA_ES"] = "aura-2-aquila-es";
    DeepgramVoice["AURA_2_SELENA_ES"] = "aura-2-selena-es";
    DeepgramVoice["AURA_2_JAVIER_ES"] = "aura-2-javier-es";
    // Aura 1 English Voices
    DeepgramVoice["AURA_ASTERIA_EN"] = "aura-asteria-en";
    DeepgramVoice["AURA_LUNA_EN"] = "aura-luna-en";
    DeepgramVoice["AURA_STELLA_EN"] = "aura-stella-en";
    DeepgramVoice["AURA_ATHENA_EN"] = "aura-athena-en";
    DeepgramVoice["AURA_HERA_EN"] = "aura-hera-en";
    DeepgramVoice["AURA_ORION_EN"] = "aura-orion-en";
    DeepgramVoice["AURA_ARCAS_EN"] = "aura-arcas-en";
    DeepgramVoice["AURA_PERSEUS_EN"] = "aura-perseus-en";
    DeepgramVoice["AURA_ANGUS_EN"] = "aura-angus-en";
    DeepgramVoice["AURA_ORPHEUS_EN"] = "aura-orpheus-en";
    DeepgramVoice["AURA_HELIOS_EN"] = "aura-helios-en";
    DeepgramVoice["AURA_ZEUS_EN"] = "aura-zeus-en";
})(DeepgramVoice || (exports.DeepgramVoice = DeepgramVoice = {}));
const DeepgramVoiceDetails = [
    // Aura 2 English Voices - Featured
    {
        name: DeepgramVoice.AURA_2_THALIA_EN,
        displayName: "Thalia",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_ANDROMEDA_EN,
        displayName: "Andromeda",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_HELENA_EN,
        displayName: "Helena",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_APOLLO_EN,
        displayName: "Apollo",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: DeepgramVoice.AURA_2_ARCAS_EN,
        displayName: "Arcas",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: DeepgramVoice.AURA_2_ARIES_EN,
        displayName: "Aries",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    // Aura 2 English Voices - All Available
    {
        name: DeepgramVoice.AURA_2_AMALTHEA_EN,
        displayName: "Amalthea",
        languageCode: types_1.VoiceLanguage.EN_PH,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_ASTERIA_EN,
        displayName: "Asteria",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_ATHENA_EN,
        displayName: "Athena",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_ATLAS_EN,
        displayName: "Atlas",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: DeepgramVoice.AURA_2_AURORA_EN,
        displayName: "Aurora",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_CALLISTA_EN,
        displayName: "Callista",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_CORA_EN,
        displayName: "Cora",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_CORDELIA_EN,
        displayName: "Cordelia",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_DELIA_EN,
        displayName: "Delia",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_DRACO_EN,
        displayName: "Draco",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: DeepgramVoice.AURA_2_ELECTRA_EN,
        displayName: "Electra",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_HARMONIA_EN,
        displayName: "Harmonia",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_HERA_EN,
        displayName: "Hera",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_IRIS_EN,
        displayName: "Iris",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_JUNO_EN,
        displayName: "Juno",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_LUNA_EN,
        displayName: "Luna",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_MAIA_EN,
        displayName: "Maia",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_NOVA_EN,
        displayName: "Nova",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_ORION_EN,
        displayName: "Orion",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: DeepgramVoice.AURA_2_PANDORA_EN,
        displayName: "Pandora",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_PHOEBE_EN,
        displayName: "Phoebe",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_PLUTO_EN,
        displayName: "Pluto",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: DeepgramVoice.AURA_2_SATURN_EN,
        displayName: "Saturn",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: DeepgramVoice.AURA_2_SELENE_EN,
        displayName: "Selene",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_THEIA_EN,
        displayName: "Theia",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_VESTA_EN,
        displayName: "Vesta",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_ZEUS_EN,
        displayName: "Zeus",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    // Aura 2 Spanish Voices - Featured
    {
        name: DeepgramVoice.AURA_2_CELESTE_ES,
        displayName: "Celeste",
        languageCode: types_1.VoiceLanguage.ES_CO,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_ESTRELLA_ES,
        displayName: "Estrella",
        languageCode: types_1.VoiceLanguage.ES_MX,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_NESTOR_ES,
        displayName: "Nestor",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.MALE
    },
    // Aura 2 Spanish Voices - All Available
    {
        name: DeepgramVoice.AURA_2_SIRIO_ES,
        displayName: "Sirio",
        languageCode: types_1.VoiceLanguage.ES_MX,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: DeepgramVoice.AURA_2_CARINA_ES,
        displayName: "Carina",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_ALVARO_ES,
        displayName: "Alvaro",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: DeepgramVoice.AURA_2_DIANA_ES,
        displayName: "Diana",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_AQUILA_ES,
        displayName: "Aquila",
        languageCode: types_1.VoiceLanguage.ES_419,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: DeepgramVoice.AURA_2_SELENA_ES,
        displayName: "Selena",
        languageCode: types_1.VoiceLanguage.ES_419,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_2_JAVIER_ES,
        displayName: "Javier",
        languageCode: types_1.VoiceLanguage.ES_MX,
        gender: types_1.VoiceGender.MALE
    },
    // Aura 1 English Voices
    {
        name: DeepgramVoice.AURA_ASTERIA_EN,
        displayName: "Asteria",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_LUNA_EN,
        displayName: "Luna",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_STELLA_EN,
        displayName: "Stella",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_ATHENA_EN,
        displayName: "Athena",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_HERA_EN,
        displayName: "Hera",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: DeepgramVoice.AURA_ORION_EN,
        displayName: "Orion",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: DeepgramVoice.AURA_ARCAS_EN,
        displayName: "Arcas",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: DeepgramVoice.AURA_PERSEUS_EN,
        displayName: "Perseus",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: DeepgramVoice.AURA_ANGUS_EN,
        displayName: "Angus",
        languageCode: types_1.VoiceLanguage.EN_IE,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: DeepgramVoice.AURA_ORPHEUS_EN,
        displayName: "Orpheus",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: DeepgramVoice.AURA_HELIOS_EN,
        displayName: "Helios",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: DeepgramVoice.AURA_ZEUS_EN,
        displayName: "Zeus",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    }
];
exports.DeepgramVoiceDetails = DeepgramVoiceDetails;
