// @fonoster/common/dist/tts/ElevenLabs.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ElevenLabsVoiceDetails = exports.ElevenLabsVoice = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const types_1 = require("./types");
var ElevenLabsVoice;
(function (ElevenLabsVoice) {
    ElevenLabsVoice["SARAH"] = "EXAVITQu4vr4xnSDxMaL";
    ElevenLabsVoice["LAURA"] = "FGY2WhTYpPnrIDTdsKH5";
    ElevenLabsVoice["CHARLIE"] = "IKne3meq5aSn9XLyUdCD";
    ElevenLabsVoice["GEORGE"] = "JBFqnCBsd6RMkjVDRZzb";
    ElevenLabsVoice["CALLUM"] = "N2lVS1w4EtoT3dr4eOWO";
    ElevenLabsVoice["LIAM"] = "TX3LPaxmHKxFdv7VOQHJ";
    ElevenLabsVoice["ALICE"] = "Xb7hH8MSUJpSbSDYk0k2";
    ElevenLabsVoice["MATILDA"] = "XrExE9yKIg1WjnnlVkGX";
    ElevenLabsVoice["WILL"] = "bIHbv24MWmeRgasZH58o";
    ElevenLabsVoice["JESSICA"] = "cgSgspJ2msm6clMCkdW9";
    ElevenLabsVoice["ERIC"] = "cjVigY5qzO86Huf0OWal";
    ElevenLabsVoice["CHRIS"] = "iP95p4xoKVk53GoZ742B";
    ElevenLabsVoice["BRIAN"] = "nPczCjzI2devNBz1zQrb";
    ElevenLabsVoice["DANIEL"] = "onwK4e9ZLuTAKqWW03F9";
    ElevenLabsVoice["LILY"] = "pFZP5JQG7iQjIQuC4Bku";
    ElevenLabsVoice["BILL"] = "pqHfZKP75CvOlQylNhV4";
})(ElevenLabsVoice || (exports.ElevenLabsVoice = ElevenLabsVoice = {}));
const ElevenLabsVoiceDetails = [
    {
        name: ElevenLabsVoice.SARAH,
        displayName: "Sarah",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: ElevenLabsVoice.LAURA,
        displayName: "Laura",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: ElevenLabsVoice.CHARLIE,
        displayName: "Charlie",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: ElevenLabsVoice.GEORGE,
        displayName: "George",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: ElevenLabsVoice.CALLUM,
        displayName: "Callum",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: ElevenLabsVoice.LIAM,
        displayName: "Liam",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: ElevenLabsVoice.ALICE,
        displayName: "Alice",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: ElevenLabsVoice.MATILDA,
        displayName: "Matilda",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: ElevenLabsVoice.WILL,
        displayName: "Will",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: ElevenLabsVoice.JESSICA,
        displayName: "Jessica",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: ElevenLabsVoice.ERIC,
        displayName: "Eric",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: ElevenLabsVoice.CHRIS,
        displayName: "Chris",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: ElevenLabsVoice.BRIAN,
        displayName: "Brian",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: ElevenLabsVoice.DANIEL,
        displayName: "Daniel",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: ElevenLabsVoice.LILY,
        displayName: "Lily",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: ElevenLabsVoice.BILL,
        displayName: "Bill",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    }
];
exports.ElevenLabsVoiceDetails = ElevenLabsVoiceDetails;
