// @fonoster/common/dist/tts/GoogleVoices.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GoogleVoiceDetails = exports.GoogleVoice = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const types_1 = require("./types");
var GoogleVoice;
(function (GoogleVoice) {
    GoogleVoice["AF_ZA_STANDARD_A"] = "af-ZA-Standard-A";
    GoogleVoice["AM_ET_STANDARD_A"] = "am-ET-Standard-A";
    GoogleVoice["AM_ET_STANDARD_B"] = "am-ET-Standard-B";
    GoogleVoice["AM_ET_WAVENET_A"] = "am-ET-Wavenet-A";
    GoogleVoice["AM_ET_WAVENET_B"] = "am-ET-Wavenet-B";
    GoogleVoice["AR_XA_STANDARD_A"] = "ar-XA-Standard-A";
    GoogleVoice["AR_XA_STANDARD_B"] = "ar-XA-Standard-B";
    GoogleVoice["AR_XA_STANDARD_C"] = "ar-XA-Standard-C";
    GoogleVoice["AR_XA_STANDARD_D"] = "ar-XA-Standard-D";
    GoogleVoice["AR_XA_WAVENET_A"] = "ar-XA-Wavenet-A";
    GoogleVoice["AR_XA_WAVENET_B"] = "ar-XA-Wavenet-B";
    GoogleVoice["AR_XA_WAVENET_C"] = "ar-XA-Wavenet-C";
    GoogleVoice["AR_XA_WAVENET_D"] = "ar-XA-Wavenet-D";
    GoogleVoice["BG_BG_STANDARD_A"] = "bg-BG-Standard-A";
    GoogleVoice["BN_IN_STANDARD_A"] = "bn-IN-Standard-A";
    GoogleVoice["BN_IN_STANDARD_B"] = "bn-IN-Standard-B";
    GoogleVoice["BN_IN_STANDARD_C"] = "bn-IN-Standard-C";
    GoogleVoice["BN_IN_STANDARD_D"] = "bn-IN-Standard-D";
    GoogleVoice["BN_IN_WAVENET_A"] = "bn-IN-Wavenet-A";
    GoogleVoice["BN_IN_WAVENET_B"] = "bn-IN-Wavenet-B";
    GoogleVoice["BN_IN_WAVENET_C"] = "bn-IN-Wavenet-C";
    GoogleVoice["BN_IN_WAVENET_D"] = "bn-IN-Wavenet-D";
    GoogleVoice["CA_ES_STANDARD_A"] = "ca-ES-Standard-A";
    GoogleVoice["CMN_CN_STANDARD_A"] = "cmn-CN-Standard-A";
    GoogleVoice["CMN_CN_STANDARD_B"] = "cmn-CN-Standard-B";
    GoogleVoice["CMN_CN_STANDARD_C"] = "cmn-CN-Standard-C";
    GoogleVoice["CMN_CN_STANDARD_D"] = "cmn-CN-Standard-D";
    GoogleVoice["CMN_CN_WAVENET_A"] = "cmn-CN-Wavenet-A";
    GoogleVoice["CMN_CN_WAVENET_B"] = "cmn-CN-Wavenet-B";
    GoogleVoice["CMN_CN_WAVENET_C"] = "cmn-CN-Wavenet-C";
    GoogleVoice["CMN_CN_WAVENET_D"] = "cmn-CN-Wavenet-D";
    GoogleVoice["CMN_TW_STANDARD_A"] = "cmn-TW-Standard-A";
    GoogleVoice["CMN_TW_STANDARD_B"] = "cmn-TW-Standard-B";
    GoogleVoice["CMN_TW_STANDARD_C"] = "cmn-TW-Standard-C";
    GoogleVoice["CMN_TW_WAVENET_A"] = "cmn-TW-Wavenet-A";
    GoogleVoice["CMN_TW_WAVENET_B"] = "cmn-TW-Wavenet-B";
    GoogleVoice["CMN_TW_WAVENET_C"] = "cmn-TW-Wavenet-C";
    GoogleVoice["CS_CZ_STANDARD_A"] = "cs-CZ-Standard-A";
    GoogleVoice["CS_CZ_WAVENET_A"] = "cs-CZ-Wavenet-A";
    GoogleVoice["DA_DK_NEURAL2_D"] = "da-DK-Neural2-D";
    GoogleVoice["DA_DK_STANDARD_A"] = "da-DK-Standard-A";
    GoogleVoice["DA_DK_STANDARD_C"] = "da-DK-Standard-C";
    GoogleVoice["DA_DK_STANDARD_D"] = "da-DK-Standard-D";
    GoogleVoice["DA_DK_STANDARD_E"] = "da-DK-Standard-E";
    GoogleVoice["DA_DK_WAVENET_A"] = "da-DK-Wavenet-A";
    GoogleVoice["DA_DK_WAVENET_C"] = "da-DK-Wavenet-C";
    GoogleVoice["DA_DK_WAVENET_D"] = "da-DK-Wavenet-D";
    GoogleVoice["DA_DK_WAVENET_E"] = "da-DK-Wavenet-E";
    GoogleVoice["DE_DE_NEURAL2_A"] = "de-DE-Neural2-A";
    GoogleVoice["DE_DE_NEURAL2_B"] = "de-DE-Neural2-B";
    GoogleVoice["DE_DE_NEURAL2_C"] = "de-DE-Neural2-C";
    GoogleVoice["DE_DE_NEURAL2_D"] = "de-DE-Neural2-D";
    GoogleVoice["DE_DE_NEURAL2_F"] = "de-DE-Neural2-F";
    GoogleVoice["DE_DE_POLYGLOT_1"] = "de-DE-Polyglot-1";
    GoogleVoice["DE_DE_STANDARD_A"] = "de-DE-Standard-A";
    GoogleVoice["DE_DE_STANDARD_B"] = "de-DE-Standard-B";
    GoogleVoice["DE_DE_STANDARD_C"] = "de-DE-Standard-C";
    GoogleVoice["DE_DE_STANDARD_D"] = "de-DE-Standard-D";
    GoogleVoice["DE_DE_STANDARD_E"] = "de-DE-Standard-E";
    GoogleVoice["DE_DE_STANDARD_F"] = "de-DE-Standard-F";
    GoogleVoice["DE_DE_STUDIO_B"] = "de-DE-Studio-B";
    GoogleVoice["DE_DE_STUDIO_C"] = "de-DE-Studio-C";
    GoogleVoice["DE_DE_WAVENET_A"] = "de-DE-Wavenet-A";
    GoogleVoice["DE_DE_WAVENET_B"] = "de-DE-Wavenet-B";
    GoogleVoice["DE_DE_WAVENET_C"] = "de-DE-Wavenet-C";
    GoogleVoice["DE_DE_WAVENET_D"] = "de-DE-Wavenet-D";
    GoogleVoice["DE_DE_WAVENET_E"] = "de-DE-Wavenet-E";
    GoogleVoice["DE_DE_WAVENET_F"] = "de-DE-Wavenet-F";
    GoogleVoice["EL_GR_STANDARD_A"] = "el-GR-Standard-A";
    GoogleVoice["EL_GR_WAVENET_A"] = "el-GR-Wavenet-A";
    GoogleVoice["EN_AU_NEURAL2_A"] = "en-AU-Neural2-A";
    GoogleVoice["EN_AU_NEURAL2_B"] = "en-AU-Neural2-B";
    GoogleVoice["EN_AU_NEURAL2_C"] = "en-AU-Neural2-C";
    GoogleVoice["EN_AU_NEURAL2_D"] = "en-AU-Neural2-D";
    GoogleVoice["EN_AU_NEWS_E"] = "en-AU-News-E";
    GoogleVoice["EN_AU_NEWS_F"] = "en-AU-News-F";
    GoogleVoice["EN_AU_NEWS_G"] = "en-AU-News-G";
    GoogleVoice["EN_AU_POLYGLOT_1"] = "en-AU-Polyglot-1";
    GoogleVoice["EN_AU_STANDARD_A"] = "en-AU-Standard-A";
    GoogleVoice["EN_AU_STANDARD_B"] = "en-AU-Standard-B";
    GoogleVoice["EN_AU_STANDARD_C"] = "en-AU-Standard-C";
    GoogleVoice["EN_AU_STANDARD_D"] = "en-AU-Standard-D";
    GoogleVoice["EN_AU_WAVENET_A"] = "en-AU-Wavenet-A";
    GoogleVoice["EN_AU_WAVENET_B"] = "en-AU-Wavenet-B";
    GoogleVoice["EN_AU_WAVENET_C"] = "en-AU-Wavenet-C";
    GoogleVoice["EN_AU_WAVENET_D"] = "en-AU-Wavenet-D";
    GoogleVoice["EN_GB_NEURAL2_A"] = "en-GB-Neural2-A";
    GoogleVoice["EN_GB_NEURAL2_B"] = "en-GB-Neural2-B";
    GoogleVoice["EN_GB_NEURAL2_C"] = "en-GB-Neural2-C";
    GoogleVoice["EN_GB_NEURAL2_D"] = "en-GB-Neural2-D";
    GoogleVoice["EN_GB_NEURAL2_F"] = "en-GB-Neural2-F";
    GoogleVoice["EN_GB_NEWS_G"] = "en-GB-News-G";
    GoogleVoice["EN_GB_NEWS_H"] = "en-GB-News-H";
    GoogleVoice["EN_GB_NEWS_I"] = "en-GB-News-I";
    GoogleVoice["EN_GB_NEWS_J"] = "en-GB-News-J";
    GoogleVoice["EN_GB_NEWS_K"] = "en-GB-News-K";
    GoogleVoice["EN_GB_NEWS_L"] = "en-GB-News-L";
    GoogleVoice["EN_GB_NEWS_M"] = "en-GB-News-M";
    GoogleVoice["EN_GB_STANDARD_A"] = "en-GB-Standard-A";
    GoogleVoice["EN_GB_STANDARD_B"] = "en-GB-Standard-B";
    GoogleVoice["EN_GB_STANDARD_C"] = "en-GB-Standard-C";
    GoogleVoice["EN_GB_STANDARD_D"] = "en-GB-Standard-D";
    GoogleVoice["EN_GB_STANDARD_F"] = "en-GB-Standard-F";
    GoogleVoice["EN_GB_STUDIO_B"] = "en-GB-Studio-B";
    GoogleVoice["EN_GB_STUDIO_C"] = "en-GB-Studio-C";
    GoogleVoice["EN_GB_WAVENET_A"] = "en-GB-Wavenet-A";
    GoogleVoice["EN_GB_WAVENET_B"] = "en-GB-Wavenet-B";
    GoogleVoice["EN_GB_WAVENET_C"] = "en-GB-Wavenet-C";
    GoogleVoice["EN_GB_WAVENET_D"] = "en-GB-Wavenet-D";
    GoogleVoice["EN_GB_WAVENET_F"] = "en-GB-Wavenet-F";
    GoogleVoice["EN_IN_NEURAL2_A"] = "en-IN-Neural2-A";
    GoogleVoice["EN_IN_NEURAL2_B"] = "en-IN-Neural2-B";
    GoogleVoice["EN_IN_NEURAL2_C"] = "en-IN-Neural2-C";
    GoogleVoice["EN_IN_NEURAL2_D"] = "en-IN-Neural2-D";
    GoogleVoice["EN_IN_STANDARD_A"] = "en-IN-Standard-A";
    GoogleVoice["EN_IN_STANDARD_B"] = "en-IN-Standard-B";
    GoogleVoice["EN_IN_STANDARD_C"] = "en-IN-Standard-C";
    GoogleVoice["EN_IN_STANDARD_D"] = "en-IN-Standard-D";
    GoogleVoice["EN_IN_WAVENET_A"] = "en-IN-Wavenet-A";
    GoogleVoice["EN_IN_WAVENET_B"] = "en-IN-Wavenet-B";
    GoogleVoice["EN_IN_WAVENET_C"] = "en-IN-Wavenet-C";
    GoogleVoice["EN_IN_WAVENET_D"] = "en-IN-Wavenet-D";
    GoogleVoice["EN_US_CASUAL_K"] = "en-US-Casual-K";
    GoogleVoice["EN_US_JOURNEY_D"] = "en-US-Journey-D";
    GoogleVoice["EN_US_JOURNEY_F"] = "en-US-Journey-F";
    GoogleVoice["EN_US_JOURNEY_O"] = "en-US-Journey-O";
    GoogleVoice["EN_US_NEURAL2_A"] = "en-US-Neural2-A";
    GoogleVoice["EN_US_NEURAL2_C"] = "en-US-Neural2-C";
    GoogleVoice["EN_US_NEURAL2_D"] = "en-US-Neural2-D";
    GoogleVoice["EN_US_NEURAL2_E"] = "en-US-Neural2-E";
    GoogleVoice["EN_US_NEURAL2_F"] = "en-US-Neural2-F";
    GoogleVoice["EN_US_NEURAL2_G"] = "en-US-Neural2-G";
    GoogleVoice["EN_US_NEURAL2_H"] = "en-US-Neural2-H";
    GoogleVoice["EN_US_NEURAL2_I"] = "en-US-Neural2-I";
    GoogleVoice["EN_US_NEURAL2_J"] = "en-US-Neural2-J";
    GoogleVoice["EN_US_NEWS_K"] = "en-US-News-K";
    GoogleVoice["EN_US_NEWS_L"] = "en-US-News-L";
    GoogleVoice["EN_US_NEWS_N"] = "en-US-News-N";
    GoogleVoice["EN_US_POLYGLOT_1"] = "en-US-Polyglot-1";
    GoogleVoice["EN_US_STANDARD_A"] = "en-US-Standard-A";
    GoogleVoice["EN_US_STANDARD_B"] = "en-US-Standard-B";
    GoogleVoice["EN_US_STANDARD_C"] = "en-US-Standard-C";
    GoogleVoice["EN_US_STANDARD_D"] = "en-US-Standard-D";
    GoogleVoice["EN_US_STANDARD_E"] = "en-US-Standard-E";
    GoogleVoice["EN_US_STANDARD_F"] = "en-US-Standard-F";
    GoogleVoice["EN_US_STANDARD_G"] = "en-US-Standard-G";
    GoogleVoice["EN_US_STANDARD_H"] = "en-US-Standard-H";
    GoogleVoice["EN_US_STANDARD_I"] = "en-US-Standard-I";
    GoogleVoice["EN_US_STANDARD_J"] = "en-US-Standard-J";
    GoogleVoice["EN_US_STUDIO_O"] = "en-US-Studio-O";
    GoogleVoice["EN_US_STUDIO_Q"] = "en-US-Studio-Q";
    GoogleVoice["EN_US_WAVENET_A"] = "en-US-Wavenet-A";
    GoogleVoice["EN_US_WAVENET_B"] = "en-US-Wavenet-B";
    GoogleVoice["EN_US_WAVENET_C"] = "en-US-Wavenet-C";
    GoogleVoice["EN_US_WAVENET_D"] = "en-US-Wavenet-D";
    GoogleVoice["EN_US_WAVENET_E"] = "en-US-Wavenet-E";
    GoogleVoice["EN_US_WAVENET_F"] = "en-US-Wavenet-F";
    GoogleVoice["EN_US_WAVENET_G"] = "en-US-Wavenet-G";
    GoogleVoice["EN_US_WAVENET_H"] = "en-US-Wavenet-H";
    GoogleVoice["EN_US_WAVENET_I"] = "en-US-Wavenet-I";
    GoogleVoice["EN_US_WAVENET_J"] = "en-US-Wavenet-J";
    GoogleVoice["ES_ES_NEURAL2_A"] = "es-ES-Neural2-A";
    GoogleVoice["ES_ES_NEURAL2_B"] = "es-ES-Neural2-B";
    GoogleVoice["ES_ES_NEURAL2_C"] = "es-ES-Neural2-C";
    GoogleVoice["ES_ES_NEURAL2_D"] = "es-ES-Neural2-D";
    GoogleVoice["ES_ES_NEURAL2_E"] = "es-ES-Neural2-E";
    GoogleVoice["ES_ES_NEURAL2_F"] = "es-ES-Neural2-F";
    GoogleVoice["ES_ES_POLYGLOT_1"] = "es-ES-Polyglot-1";
    GoogleVoice["ES_ES_STANDARD_A"] = "es-ES-Standard-A";
    GoogleVoice["ES_ES_STANDARD_B"] = "es-ES-Standard-B";
    GoogleVoice["ES_ES_STANDARD_C"] = "es-ES-Standard-C";
    GoogleVoice["ES_ES_STANDARD_D"] = "es-ES-Standard-D";
    GoogleVoice["ES_ES_STUDIO_C"] = "es-ES-Studio-C";
    GoogleVoice["ES_ES_STUDIO_F"] = "es-ES-Studio-F";
    GoogleVoice["ES_ES_WAVENET_B"] = "es-ES-Wavenet-B";
    GoogleVoice["ES_ES_WAVENET_C"] = "es-ES-Wavenet-C";
    GoogleVoice["ES_ES_WAVENET_D"] = "es-ES-Wavenet-D";
    GoogleVoice["ES_US_NEURAL2_A"] = "es-US-Neural2-A";
    GoogleVoice["ES_US_NEURAL2_B"] = "es-US-Neural2-B";
    GoogleVoice["ES_US_NEURAL2_C"] = "es-US-Neural2-C";
    GoogleVoice["ES_US_NEWS_D"] = "es-US-News-D";
    GoogleVoice["ES_US_NEWS_E"] = "es-US-News-E";
    GoogleVoice["ES_US_NEWS_F"] = "es-US-News-F";
    GoogleVoice["ES_US_NEWS_G"] = "es-US-News-G";
    GoogleVoice["ES_US_POLYGLOT_1"] = "es-US-Polyglot-1";
    GoogleVoice["ES_US_STANDARD_A"] = "es-US-Standard-A";
    GoogleVoice["ES_US_STANDARD_B"] = "es-US-Standard-B";
    GoogleVoice["ES_US_STANDARD_C"] = "es-US-Standard-C";
    GoogleVoice["ES_US_STUDIO_B"] = "es-US-Studio-B";
    GoogleVoice["ES_US_WAVENET_A"] = "es-US-Wavenet-A";
    GoogleVoice["ES_US_WAVENET_B"] = "es-US-Wavenet-B";
    GoogleVoice["ES_US_WAVENET_C"] = "es-US-Wavenet-C";
    GoogleVoice["EU_ES_STANDARD_A"] = "eu-ES-Standard-A";
    GoogleVoice["FI_FI_STANDARD_A"] = "fi-FI-Standard-A";
    GoogleVoice["FI_FI_WAVENET_A"] = "fi-FI-Wavenet-A";
    GoogleVoice["FIL_PH_STANDARD_A"] = "fil-PH-Standard-A";
    GoogleVoice["FIL_PH_STANDARD_B"] = "fil-PH-Standard-B";
    GoogleVoice["FIL_PH_STANDARD_C"] = "fil-PH-Standard-C";
    GoogleVoice["FIL_PH_STANDARD_D"] = "fil-PH-Standard-D";
    GoogleVoice["FIL_PH_WAVENET_A"] = "fil-PH-Wavenet-A";
    GoogleVoice["FIL_PH_WAVENET_B"] = "fil-PH-Wavenet-B";
    GoogleVoice["FIL_PH_WAVENET_C"] = "fil-PH-Wavenet-C";
    GoogleVoice["FIL_PH_WAVENET_D"] = "fil-PH-Wavenet-D";
    GoogleVoice["FIL_PH_NEURAL2_A"] = "fil-ph-Neural2-A";
    GoogleVoice["FIL_PH_NEURAL2_D"] = "fil-ph-Neural2-D";
    GoogleVoice["FR_CA_NEURAL2_A"] = "fr-CA-Neural2-A";
    GoogleVoice["FR_CA_NEURAL2_B"] = "fr-CA-Neural2-B";
    GoogleVoice["FR_CA_NEURAL2_C"] = "fr-CA-Neural2-C";
    GoogleVoice["FR_CA_NEURAL2_D"] = "fr-CA-Neural2-D";
    GoogleVoice["FR_CA_STANDARD_A"] = "fr-CA-Standard-A";
    GoogleVoice["FR_CA_STANDARD_B"] = "fr-CA-Standard-B";
    GoogleVoice["FR_CA_STANDARD_C"] = "fr-CA-Standard-C";
    GoogleVoice["FR_CA_STANDARD_D"] = "fr-CA-Standard-D";
    GoogleVoice["FR_CA_WAVENET_A"] = "fr-CA-Wavenet-A";
    GoogleVoice["FR_CA_WAVENET_B"] = "fr-CA-Wavenet-B";
    GoogleVoice["FR_CA_WAVENET_C"] = "fr-CA-Wavenet-C";
    GoogleVoice["FR_CA_WAVENET_D"] = "fr-CA-Wavenet-D";
    GoogleVoice["FR_FR_NEURAL2_A"] = "fr-FR-Neural2-A";
    GoogleVoice["FR_FR_NEURAL2_B"] = "fr-FR-Neural2-B";
    GoogleVoice["FR_FR_NEURAL2_C"] = "fr-FR-Neural2-C";
    GoogleVoice["FR_FR_NEURAL2_D"] = "fr-FR-Neural2-D";
    GoogleVoice["FR_FR_NEURAL2_E"] = "fr-FR-Neural2-E";
    GoogleVoice["FR_FR_POLYGLOT_1"] = "fr-FR-Polyglot-1";
    GoogleVoice["FR_FR_STANDARD_A"] = "fr-FR-Standard-A";
    GoogleVoice["FR_FR_STANDARD_B"] = "fr-FR-Standard-B";
    GoogleVoice["FR_FR_STANDARD_C"] = "fr-FR-Standard-C";
    GoogleVoice["FR_FR_STANDARD_D"] = "fr-FR-Standard-D";
    GoogleVoice["FR_FR_STANDARD_E"] = "fr-FR-Standard-E";
    GoogleVoice["FR_FR_STUDIO_A"] = "fr-FR-Studio-A";
    GoogleVoice["FR_FR_STUDIO_D"] = "fr-FR-Studio-D";
    GoogleVoice["FR_FR_WAVENET_A"] = "fr-FR-Wavenet-A";
    GoogleVoice["FR_FR_WAVENET_B"] = "fr-FR-Wavenet-B";
    GoogleVoice["FR_FR_WAVENET_C"] = "fr-FR-Wavenet-C";
    GoogleVoice["FR_FR_WAVENET_D"] = "fr-FR-Wavenet-D";
    GoogleVoice["FR_FR_WAVENET_E"] = "fr-FR-Wavenet-E";
    GoogleVoice["GL_ES_STANDARD_A"] = "gl-ES-Standard-A";
    GoogleVoice["GU_IN_STANDARD_A"] = "gu-IN-Standard-A";
    GoogleVoice["GU_IN_STANDARD_B"] = "gu-IN-Standard-B";
    GoogleVoice["GU_IN_STANDARD_C"] = "gu-IN-Standard-C";
    GoogleVoice["GU_IN_STANDARD_D"] = "gu-IN-Standard-D";
    GoogleVoice["GU_IN_WAVENET_A"] = "gu-IN-Wavenet-A";
    GoogleVoice["GU_IN_WAVENET_B"] = "gu-IN-Wavenet-B";
    GoogleVoice["GU_IN_WAVENET_C"] = "gu-IN-Wavenet-C";
    GoogleVoice["GU_IN_WAVENET_D"] = "gu-IN-Wavenet-D";
    GoogleVoice["HE_IL_STANDARD_A"] = "he-IL-Standard-A";
    GoogleVoice["HE_IL_STANDARD_B"] = "he-IL-Standard-B";
    GoogleVoice["HE_IL_STANDARD_C"] = "he-IL-Standard-C";
    GoogleVoice["HE_IL_STANDARD_D"] = "he-IL-Standard-D";
    GoogleVoice["HE_IL_WAVENET_A"] = "he-IL-Wavenet-A";
    GoogleVoice["HE_IL_WAVENET_B"] = "he-IL-Wavenet-B";
    GoogleVoice["HE_IL_WAVENET_C"] = "he-IL-Wavenet-C";
    GoogleVoice["HE_IL_WAVENET_D"] = "he-IL-Wavenet-D";
    GoogleVoice["HI_IN_NEURAL2_A"] = "hi-IN-Neural2-A";
    GoogleVoice["HI_IN_NEURAL2_B"] = "hi-IN-Neural2-B";
    GoogleVoice["HI_IN_NEURAL2_C"] = "hi-IN-Neural2-C";
    GoogleVoice["HI_IN_NEURAL2_D"] = "hi-IN-Neural2-D";
    GoogleVoice["HI_IN_STANDARD_A"] = "hi-IN-Standard-A";
    GoogleVoice["HI_IN_STANDARD_B"] = "hi-IN-Standard-B";
    GoogleVoice["HI_IN_STANDARD_C"] = "hi-IN-Standard-C";
    GoogleVoice["HI_IN_STANDARD_D"] = "hi-IN-Standard-D";
    GoogleVoice["HI_IN_WAVENET_A"] = "hi-IN-Wavenet-A";
    GoogleVoice["HI_IN_WAVENET_B"] = "hi-IN-Wavenet-B";
    GoogleVoice["HI_IN_WAVENET_C"] = "hi-IN-Wavenet-C";
    GoogleVoice["HI_IN_WAVENET_D"] = "hi-IN-Wavenet-D";
    GoogleVoice["HU_HU_STANDARD_A"] = "hu-HU-Standard-A";
    GoogleVoice["HU_HU_WAVENET_A"] = "hu-HU-Wavenet-A";
    GoogleVoice["ID_ID_STANDARD_A"] = "id-ID-Standard-A";
    GoogleVoice["ID_ID_STANDARD_B"] = "id-ID-Standard-B";
    GoogleVoice["ID_ID_STANDARD_C"] = "id-ID-Standard-C";
    GoogleVoice["ID_ID_STANDARD_D"] = "id-ID-Standard-D";
    GoogleVoice["ID_ID_WAVENET_A"] = "id-ID-Wavenet-A";
    GoogleVoice["ID_ID_WAVENET_B"] = "id-ID-Wavenet-B";
    GoogleVoice["ID_ID_WAVENET_C"] = "id-ID-Wavenet-C";
    GoogleVoice["ID_ID_WAVENET_D"] = "id-ID-Wavenet-D";
    GoogleVoice["IS_IS_STANDARD_A"] = "is-IS-Standard-A";
    GoogleVoice["IT_IT_NEURAL2_A"] = "it-IT-Neural2-A";
    GoogleVoice["IT_IT_NEURAL2_C"] = "it-IT-Neural2-C";
    GoogleVoice["IT_IT_STANDARD_A"] = "it-IT-Standard-A";
    GoogleVoice["IT_IT_STANDARD_B"] = "it-IT-Standard-B";
    GoogleVoice["IT_IT_STANDARD_C"] = "it-IT-Standard-C";
    GoogleVoice["IT_IT_STANDARD_D"] = "it-IT-Standard-D";
    GoogleVoice["IT_IT_WAVENET_A"] = "it-IT-Wavenet-A";
    GoogleVoice["IT_IT_WAVENET_B"] = "it-IT-Wavenet-B";
    GoogleVoice["IT_IT_WAVENET_C"] = "it-IT-Wavenet-C";
    GoogleVoice["IT_IT_WAVENET_D"] = "it-IT-Wavenet-D";
    GoogleVoice["JA_JP_NEURAL2_B"] = "ja-JP-Neural2-B";
    GoogleVoice["JA_JP_NEURAL2_C"] = "ja-JP-Neural2-C";
    GoogleVoice["JA_JP_NEURAL2_D"] = "ja-JP-Neural2-D";
    GoogleVoice["JA_JP_STANDARD_A"] = "ja-JP-Standard-A";
    GoogleVoice["JA_JP_STANDARD_B"] = "ja-JP-Standard-B";
    GoogleVoice["JA_JP_STANDARD_C"] = "ja-JP-Standard-C";
    GoogleVoice["JA_JP_STANDARD_D"] = "ja-JP-Standard-D";
    GoogleVoice["JA_JP_WAVENET_A"] = "ja-JP-Wavenet-A";
    GoogleVoice["JA_JP_WAVENET_B"] = "ja-JP-Wavenet-B";
    GoogleVoice["JA_JP_WAVENET_C"] = "ja-JP-Wavenet-C";
    GoogleVoice["JA_JP_WAVENET_D"] = "ja-JP-Wavenet-D";
    GoogleVoice["KN_IN_STANDARD_A"] = "kn-IN-Standard-A";
    GoogleVoice["KN_IN_STANDARD_B"] = "kn-IN-Standard-B";
    GoogleVoice["KN_IN_STANDARD_C"] = "kn-IN-Standard-C";
    GoogleVoice["KN_IN_STANDARD_D"] = "kn-IN-Standard-D";
    GoogleVoice["KN_IN_WAVENET_A"] = "kn-IN-Wavenet-A";
    GoogleVoice["KN_IN_WAVENET_B"] = "kn-IN-Wavenet-B";
    GoogleVoice["KN_IN_WAVENET_C"] = "kn-IN-Wavenet-C";
    GoogleVoice["KN_IN_WAVENET_D"] = "kn-IN-Wavenet-D";
    GoogleVoice["KO_KR_NEURAL2_A"] = "ko-KR-Neural2-A";
    GoogleVoice["KO_KR_NEURAL2_B"] = "ko-KR-Neural2-B";
    GoogleVoice["KO_KR_NEURAL2_C"] = "ko-KR-Neural2-C";
    GoogleVoice["KO_KR_STANDARD_A"] = "ko-KR-Standard-A";
    GoogleVoice["KO_KR_STANDARD_B"] = "ko-KR-Standard-B";
    GoogleVoice["KO_KR_STANDARD_C"] = "ko-KR-Standard-C";
    GoogleVoice["KO_KR_STANDARD_D"] = "ko-KR-Standard-D";
    GoogleVoice["KO_KR_WAVENET_A"] = "ko-KR-Wavenet-A";
    GoogleVoice["KO_KR_WAVENET_B"] = "ko-KR-Wavenet-B";
    GoogleVoice["KO_KR_WAVENET_C"] = "ko-KR-Wavenet-C";
    GoogleVoice["KO_KR_WAVENET_D"] = "ko-KR-Wavenet-D";
    GoogleVoice["LT_LT_STANDARD_A"] = "lt-LT-Standard-A";
    GoogleVoice["LV_LV_STANDARD_A"] = "lv-LV-Standard-A";
    GoogleVoice["ML_IN_STANDARD_A"] = "ml-IN-Standard-A";
    GoogleVoice["ML_IN_STANDARD_B"] = "ml-IN-Standard-B";
    GoogleVoice["ML_IN_STANDARD_C"] = "ml-IN-Standard-C";
    GoogleVoice["ML_IN_STANDARD_D"] = "ml-IN-Standard-D";
    GoogleVoice["ML_IN_WAVENET_A"] = "ml-IN-Wavenet-A";
    GoogleVoice["ML_IN_WAVENET_B"] = "ml-IN-Wavenet-B";
    GoogleVoice["ML_IN_WAVENET_C"] = "ml-IN-Wavenet-C";
    GoogleVoice["ML_IN_WAVENET_D"] = "ml-IN-Wavenet-D";
    GoogleVoice["MR_IN_STANDARD_A"] = "mr-IN-Standard-A";
    GoogleVoice["MR_IN_STANDARD_B"] = "mr-IN-Standard-B";
    GoogleVoice["MR_IN_STANDARD_C"] = "mr-IN-Standard-C";
    GoogleVoice["MR_IN_WAVENET_A"] = "mr-IN-Wavenet-A";
    GoogleVoice["MR_IN_WAVENET_B"] = "mr-IN-Wavenet-B";
    GoogleVoice["MR_IN_WAVENET_C"] = "mr-IN-Wavenet-C";
    GoogleVoice["MS_MY_STANDARD_A"] = "ms-MY-Standard-A";
    GoogleVoice["MS_MY_STANDARD_B"] = "ms-MY-Standard-B";
    GoogleVoice["MS_MY_STANDARD_C"] = "ms-MY-Standard-C";
    GoogleVoice["MS_MY_STANDARD_D"] = "ms-MY-Standard-D";
    GoogleVoice["MS_MY_WAVENET_A"] = "ms-MY-Wavenet-A";
    GoogleVoice["MS_MY_WAVENET_B"] = "ms-MY-Wavenet-B";
    GoogleVoice["MS_MY_WAVENET_C"] = "ms-MY-Wavenet-C";
    GoogleVoice["MS_MY_WAVENET_D"] = "ms-MY-Wavenet-D";
    GoogleVoice["NB_NO_STANDARD_A"] = "nb-NO-Standard-A";
    GoogleVoice["NB_NO_STANDARD_B"] = "nb-NO-Standard-B";
    GoogleVoice["NB_NO_STANDARD_C"] = "nb-NO-Standard-C";
    GoogleVoice["NB_NO_STANDARD_D"] = "nb-NO-Standard-D";
    GoogleVoice["NB_NO_STANDARD_E"] = "nb-NO-Standard-E";
    GoogleVoice["NB_NO_WAVENET_A"] = "nb-NO-Wavenet-A";
    GoogleVoice["NB_NO_WAVENET_B"] = "nb-NO-Wavenet-B";
    GoogleVoice["NB_NO_WAVENET_C"] = "nb-NO-Wavenet-C";
    GoogleVoice["NB_NO_WAVENET_D"] = "nb-NO-Wavenet-D";
    GoogleVoice["NB_NO_WAVENET_E"] = "nb-NO-Wavenet-E";
    GoogleVoice["NL_BE_STANDARD_A"] = "nl-BE-Standard-A";
    GoogleVoice["NL_BE_STANDARD_B"] = "nl-BE-Standard-B";
    GoogleVoice["NL_BE_WAVENET_A"] = "nl-BE-Wavenet-A";
    GoogleVoice["NL_BE_WAVENET_B"] = "nl-BE-Wavenet-B";
    GoogleVoice["NL_NL_STANDARD_A"] = "nl-NL-Standard-A";
    GoogleVoice["NL_NL_STANDARD_B"] = "nl-NL-Standard-B";
    GoogleVoice["NL_NL_STANDARD_C"] = "nl-NL-Standard-C";
    GoogleVoice["NL_NL_STANDARD_D"] = "nl-NL-Standard-D";
    GoogleVoice["NL_NL_STANDARD_E"] = "nl-NL-Standard-E";
    GoogleVoice["NL_NL_WAVENET_A"] = "nl-NL-Wavenet-A";
    GoogleVoice["NL_NL_WAVENET_B"] = "nl-NL-Wavenet-B";
    GoogleVoice["NL_NL_WAVENET_C"] = "nl-NL-Wavenet-C";
    GoogleVoice["NL_NL_WAVENET_D"] = "nl-NL-Wavenet-D";
    GoogleVoice["NL_NL_WAVENET_E"] = "nl-NL-Wavenet-E";
    GoogleVoice["PA_IN_STANDARD_A"] = "pa-IN-Standard-A";
    GoogleVoice["PA_IN_STANDARD_B"] = "pa-IN-Standard-B";
    GoogleVoice["PA_IN_STANDARD_C"] = "pa-IN-Standard-C";
    GoogleVoice["PA_IN_STANDARD_D"] = "pa-IN-Standard-D";
    GoogleVoice["PA_IN_WAVENET_A"] = "pa-IN-Wavenet-A";
    GoogleVoice["PA_IN_WAVENET_B"] = "pa-IN-Wavenet-B";
    GoogleVoice["PA_IN_WAVENET_C"] = "pa-IN-Wavenet-C";
    GoogleVoice["PA_IN_WAVENET_D"] = "pa-IN-Wavenet-D";
    GoogleVoice["PL_PL_STANDARD_A"] = "pl-PL-Standard-A";
    GoogleVoice["PL_PL_STANDARD_B"] = "pl-PL-Standard-B";
    GoogleVoice["PL_PL_STANDARD_C"] = "pl-PL-Standard-C";
    GoogleVoice["PL_PL_STANDARD_D"] = "pl-PL-Standard-D";
    GoogleVoice["PL_PL_STANDARD_E"] = "pl-PL-Standard-E";
    GoogleVoice["PL_PL_WAVENET_A"] = "pl-PL-Wavenet-A";
    GoogleVoice["PL_PL_WAVENET_B"] = "pl-PL-Wavenet-B";
    GoogleVoice["PL_PL_WAVENET_C"] = "pl-PL-Wavenet-C";
    GoogleVoice["PL_PL_WAVENET_D"] = "pl-PL-Wavenet-D";
    GoogleVoice["PL_PL_WAVENET_E"] = "pl-PL-Wavenet-E";
    GoogleVoice["PT_BR_NEURAL2_A"] = "pt-BR-Neural2-A";
    GoogleVoice["PT_BR_NEURAL2_B"] = "pt-BR-Neural2-B";
    GoogleVoice["PT_BR_NEURAL2_C"] = "pt-BR-Neural2-C";
    GoogleVoice["PT_BR_STANDARD_A"] = "pt-BR-Standard-A";
    GoogleVoice["PT_BR_STANDARD_B"] = "pt-BR-Standard-B";
    GoogleVoice["PT_BR_STANDARD_C"] = "pt-BR-Standard-C";
    GoogleVoice["PT_BR_WAVENET_A"] = "pt-BR-Wavenet-A";
    GoogleVoice["PT_BR_WAVENET_B"] = "pt-BR-Wavenet-B";
    GoogleVoice["PT_BR_WAVENET_C"] = "pt-BR-Wavenet-C";
    GoogleVoice["PT_PT_STANDARD_A"] = "pt-PT-Standard-A";
    GoogleVoice["PT_PT_STANDARD_B"] = "pt-PT-Standard-B";
    GoogleVoice["PT_PT_STANDARD_C"] = "pt-PT-Standard-C";
    GoogleVoice["PT_PT_STANDARD_D"] = "pt-PT-Standard-D";
    GoogleVoice["PT_PT_WAVENET_A"] = "pt-PT-Wavenet-A";
    GoogleVoice["PT_PT_WAVENET_B"] = "pt-PT-Wavenet-B";
    GoogleVoice["PT_PT_WAVENET_C"] = "pt-PT-Wavenet-C";
    GoogleVoice["PT_PT_WAVENET_D"] = "pt-PT-Wavenet-D";
    GoogleVoice["RO_RO_STANDARD_A"] = "ro-RO-Standard-A";
    GoogleVoice["RO_RO_WAVENET_A"] = "ro-RO-Wavenet-A";
    GoogleVoice["RU_RU_STANDARD_A"] = "ru-RU-Standard-A";
    GoogleVoice["RU_RU_STANDARD_B"] = "ru-RU-Standard-B";
    GoogleVoice["RU_RU_STANDARD_C"] = "ru-RU-Standard-C";
    GoogleVoice["RU_RU_STANDARD_D"] = "ru-RU-Standard-D";
    GoogleVoice["RU_RU_STANDARD_E"] = "ru-RU-Standard-E";
    GoogleVoice["RU_RU_WAVENET_A"] = "ru-RU-Wavenet-A";
    GoogleVoice["RU_RU_WAVENET_B"] = "ru-RU-Wavenet-B";
    GoogleVoice["RU_RU_WAVENET_C"] = "ru-RU-Wavenet-C";
    GoogleVoice["RU_RU_WAVENET_D"] = "ru-RU-Wavenet-D";
    GoogleVoice["RU_RU_WAVENET_E"] = "ru-RU-Wavenet-E";
    GoogleVoice["SK_SK_STANDARD_A"] = "sk-SK-Standard-A";
    GoogleVoice["SK_SK_WAVENET_A"] = "sk-SK-Wavenet-A";
    GoogleVoice["SR_RS_STANDARD_A"] = "sr-RS-Standard-A";
    GoogleVoice["SV_SE_STANDARD_A"] = "sv-SE-Standard-A";
    GoogleVoice["SV_SE_STANDARD_B"] = "sv-SE-Standard-B";
    GoogleVoice["SV_SE_STANDARD_C"] = "sv-SE-Standard-C";
    GoogleVoice["SV_SE_STANDARD_D"] = "sv-SE-Standard-D";
    GoogleVoice["SV_SE_STANDARD_E"] = "sv-SE-Standard-E";
    GoogleVoice["SV_SE_WAVENET_A"] = "sv-SE-Wavenet-A";
    GoogleVoice["SV_SE_WAVENET_B"] = "sv-SE-Wavenet-B";
    GoogleVoice["SV_SE_WAVENET_C"] = "sv-SE-Wavenet-C";
    GoogleVoice["SV_SE_WAVENET_D"] = "sv-SE-Wavenet-D";
    GoogleVoice["SV_SE_WAVENET_E"] = "sv-SE-Wavenet-E";
    GoogleVoice["TA_IN_STANDARD_A"] = "ta-IN-Standard-A";
    GoogleVoice["TA_IN_STANDARD_B"] = "ta-IN-Standard-B";
    GoogleVoice["TA_IN_STANDARD_C"] = "ta-IN-Standard-C";
    GoogleVoice["TA_IN_STANDARD_D"] = "ta-IN-Standard-D";
    GoogleVoice["TA_IN_WAVENET_A"] = "ta-IN-Wavenet-A";
    GoogleVoice["TA_IN_WAVENET_B"] = "ta-IN-Wavenet-B";
    GoogleVoice["TA_IN_WAVENET_C"] = "ta-IN-Wavenet-C";
    GoogleVoice["TA_IN_WAVENET_D"] = "ta-IN-Wavenet-D";
    GoogleVoice["TE_IN_STANDARD_A"] = "te-IN-Standard-A";
    GoogleVoice["TE_IN_STANDARD_B"] = "te-IN-Standard-B";
    GoogleVoice["TH_TH_NEURAL2_C"] = "th-TH-Neural2-C";
    GoogleVoice["TH_TH_STANDARD_A"] = "th-TH-Standard-A";
    GoogleVoice["TR_TR_STANDARD_A"] = "tr-TR-Standard-A";
    GoogleVoice["TR_TR_STANDARD_B"] = "tr-TR-Standard-B";
    GoogleVoice["TR_TR_STANDARD_C"] = "tr-TR-Standard-C";
    GoogleVoice["TR_TR_STANDARD_D"] = "tr-TR-Standard-D";
    GoogleVoice["TR_TR_STANDARD_E"] = "tr-TR-Standard-E";
    GoogleVoice["TR_TR_WAVENET_A"] = "tr-TR-Wavenet-A";
    GoogleVoice["TR_TR_WAVENET_B"] = "tr-TR-Wavenet-B";
    GoogleVoice["TR_TR_WAVENET_C"] = "tr-TR-Wavenet-C";
    GoogleVoice["TR_TR_WAVENET_D"] = "tr-TR-Wavenet-D";
    GoogleVoice["TR_TR_WAVENET_E"] = "tr-TR-Wavenet-E";
    GoogleVoice["UK_UA_STANDARD_A"] = "uk-UA-Standard-A";
    GoogleVoice["UK_UA_WAVENET_A"] = "uk-UA-Wavenet-A";
    GoogleVoice["UR_IN_STANDARD_A"] = "ur-IN-Standard-A";
    GoogleVoice["UR_IN_STANDARD_B"] = "ur-IN-Standard-B";
    GoogleVoice["UR_IN_WAVENET_A"] = "ur-IN-Wavenet-A";
    GoogleVoice["UR_IN_WAVENET_B"] = "ur-IN-Wavenet-B";
    GoogleVoice["VI_VN_NEURAL2_A"] = "vi-VN-Neural2-A";
    GoogleVoice["VI_VN_NEURAL2_D"] = "vi-VN-Neural2-D";
    GoogleVoice["VI_VN_STANDARD_A"] = "vi-VN-Standard-A";
    GoogleVoice["VI_VN_STANDARD_B"] = "vi-VN-Standard-B";
    GoogleVoice["VI_VN_STANDARD_C"] = "vi-VN-Standard-C";
    GoogleVoice["VI_VN_STANDARD_D"] = "vi-VN-Standard-D";
    GoogleVoice["VI_VN_WAVENET_A"] = "vi-VN-Wavenet-A";
    GoogleVoice["VI_VN_WAVENET_B"] = "vi-VN-Wavenet-B";
    GoogleVoice["VI_VN_WAVENET_C"] = "vi-VN-Wavenet-C";
    GoogleVoice["VI_VN_WAVENET_D"] = "vi-VN-Wavenet-D";
    GoogleVoice["YUE_HK_STANDARD_A"] = "yue-HK-Standard-A";
    GoogleVoice["YUE_HK_STANDARD_B"] = "yue-HK-Standard-B";
    GoogleVoice["YUE_HK_STANDARD_C"] = "yue-HK-Standard-C";
    GoogleVoice["YUE_HK_STANDARD_D"] = "yue-HK-Standard-D";
})(GoogleVoice || (exports.GoogleVoice = GoogleVoice = {}));
const GoogleVoiceDetails = [
    {
        name: GoogleVoice.AF_ZA_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.AF_ZA,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.AM_ET_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.AM_ET,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.AM_ET_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.AM_ET,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.AM_ET_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.AM_ET,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.AM_ET_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.AM_ET,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.AR_XA_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.AR_XA,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.AR_XA_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.AR_XA,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.AR_XA_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.AR_XA,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.AR_XA_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.AR_XA,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.AR_XA_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.AR_XA,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.AR_XA_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.AR_XA,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.AR_XA_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.AR_XA,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.AR_XA_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.AR_XA,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.BG_BG_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.BG_BG,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.BN_IN_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.BN_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.BN_IN_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.BN_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.BN_IN_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.BN_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.BN_IN_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.BN_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.BN_IN_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.BN_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.BN_IN_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.BN_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.BN_IN_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.BN_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.BN_IN_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.BN_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.CA_ES_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.CA_ES,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.CMN_CN_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.CMN_CN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.CMN_CN_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.CMN_CN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.CMN_CN_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.CMN_CN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.CMN_CN_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.CMN_CN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.CMN_CN_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.CMN_CN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.CMN_CN_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.CMN_CN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.CMN_CN_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.CMN_CN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.CMN_CN_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.CMN_CN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.CMN_TW_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.CMN_TW,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.CMN_TW_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.CMN_TW,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.CMN_TW_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.CMN_TW,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.CMN_TW_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.CMN_TW,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.CMN_TW_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.CMN_TW,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.CMN_TW_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.CMN_TW,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.CS_CZ_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.CS_CZ,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.CS_CZ_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.CS_CZ,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.DA_DK_NEURAL2_D,
        displayName: "Neural2 D",
        languageCode: types_1.VoiceLanguage.DA_DK,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.DA_DK_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.DA_DK,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.DA_DK_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.DA_DK,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.DA_DK_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.DA_DK,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.DA_DK_STANDARD_E,
        displayName: "Standard E",
        languageCode: types_1.VoiceLanguage.DA_DK,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.DA_DK_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.DA_DK,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.DA_DK_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.DA_DK,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.DA_DK_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.DA_DK,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.DA_DK_WAVENET_E,
        displayName: "Wavenet E",
        languageCode: types_1.VoiceLanguage.DA_DK,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.DE_DE_NEURAL2_A,
        displayName: "Neural2 A",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.DE_DE_NEURAL2_B,
        displayName: "Neural2 B",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.DE_DE_NEURAL2_C,
        displayName: "Neural2 C",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.DE_DE_NEURAL2_D,
        displayName: "Neural2 D",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.DE_DE_NEURAL2_F,
        displayName: "Neural2 F",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.DE_DE_POLYGLOT_1,
        displayName: "Polyglot 1",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.DE_DE_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.DE_DE_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.DE_DE_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.DE_DE_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.DE_DE_STANDARD_E,
        displayName: "Standard E",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.DE_DE_STANDARD_F,
        displayName: "Standard F",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.DE_DE_STUDIO_B,
        displayName: "Studio B",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.DE_DE_STUDIO_C,
        displayName: "Studio C",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.DE_DE_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.DE_DE_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.DE_DE_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.DE_DE_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.DE_DE_WAVENET_E,
        displayName: "Wavenet E",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.DE_DE_WAVENET_F,
        displayName: "Wavenet F",
        languageCode: types_1.VoiceLanguage.DE_DE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EL_GR_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.EL_GR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EL_GR_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.EL_GR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_AU_NEURAL2_A,
        displayName: "Neural2 A",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_AU_NEURAL2_B,
        displayName: "Neural2 B",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_AU_NEURAL2_C,
        displayName: "Neural2 C",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_AU_NEURAL2_D,
        displayName: "Neural2 D",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_AU_NEWS_E,
        displayName: "News E",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_AU_NEWS_F,
        displayName: "News F",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_AU_NEWS_G,
        displayName: "News G",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_AU_POLYGLOT_1,
        displayName: "Polyglot 1",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_AU_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_AU_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_AU_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_AU_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_AU_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_AU_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_AU_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_AU_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.EN_AU,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_GB_NEURAL2_A,
        displayName: "Neural2 A",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_GB_NEURAL2_B,
        displayName: "Neural2 B",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_GB_NEURAL2_C,
        displayName: "Neural2 C",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_GB_NEURAL2_D,
        displayName: "Neural2 D",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_GB_NEURAL2_F,
        displayName: "Neural2 F",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_GB_NEWS_G,
        displayName: "News G",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_GB_NEWS_H,
        displayName: "News H",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_GB_NEWS_I,
        displayName: "News I",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_GB_NEWS_J,
        displayName: "News J",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_GB_NEWS_K,
        displayName: "News K",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_GB_NEWS_L,
        displayName: "News L",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_GB_NEWS_M,
        displayName: "News M",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_GB_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_GB_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_GB_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_GB_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_GB_STANDARD_F,
        displayName: "Standard F",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_GB_STUDIO_B,
        displayName: "Studio B",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_GB_STUDIO_C,
        displayName: "Studio C",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_GB_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_GB_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_GB_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_GB_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_GB_WAVENET_F,
        displayName: "Wavenet F",
        languageCode: types_1.VoiceLanguage.EN_GB,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_IN_NEURAL2_A,
        displayName: "Neural2 A",
        languageCode: types_1.VoiceLanguage.EN_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_IN_NEURAL2_B,
        displayName: "Neural2 B",
        languageCode: types_1.VoiceLanguage.EN_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_IN_NEURAL2_C,
        displayName: "Neural2 C",
        languageCode: types_1.VoiceLanguage.EN_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_IN_NEURAL2_D,
        displayName: "Neural2 D",
        languageCode: types_1.VoiceLanguage.EN_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_IN_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.EN_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_IN_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.EN_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_IN_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.EN_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_IN_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.EN_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_IN_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.EN_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_IN_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.EN_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_IN_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.EN_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_IN_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.EN_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_CASUAL_K,
        displayName: "Casual K",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_JOURNEY_D,
        displayName: "Journey D",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_JOURNEY_F,
        displayName: "Journey F",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_JOURNEY_O,
        displayName: "Journey O",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_NEURAL2_A,
        displayName: "Neural2 A",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_NEURAL2_C,
        displayName: "Neural2 C",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_NEURAL2_D,
        displayName: "Neural2 D",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_NEURAL2_E,
        displayName: "Neural2 E",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_NEURAL2_F,
        displayName: "Neural2 F",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_NEURAL2_G,
        displayName: "Neural2 G",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_NEURAL2_H,
        displayName: "Neural2 H",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_NEURAL2_I,
        displayName: "Neural2 I",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_NEURAL2_J,
        displayName: "Neural2 J",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_NEWS_K,
        displayName: "News K",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_NEWS_L,
        displayName: "News L",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_NEWS_N,
        displayName: "News N",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_POLYGLOT_1,
        displayName: "Polyglot 1",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_STANDARD_E,
        displayName: "Standard E",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_STANDARD_F,
        displayName: "Standard F",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_STANDARD_G,
        displayName: "Standard G",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_STANDARD_H,
        displayName: "Standard H",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_STANDARD_I,
        displayName: "Standard I",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_STANDARD_J,
        displayName: "Standard J",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_STUDIO_O,
        displayName: "Studio O",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_STUDIO_Q,
        displayName: "Studio Q",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_WAVENET_E,
        displayName: "Wavenet E",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_WAVENET_F,
        displayName: "Wavenet F",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_WAVENET_G,
        displayName: "Wavenet G",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_WAVENET_H,
        displayName: "Wavenet H",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.EN_US_WAVENET_I,
        displayName: "Wavenet I",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EN_US_WAVENET_J,
        displayName: "Wavenet J",
        languageCode: types_1.VoiceLanguage.EN_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ES_ES_NEURAL2_A,
        displayName: "Neural2 A",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ES_ES_NEURAL2_B,
        displayName: "Neural2 B",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ES_ES_NEURAL2_C,
        displayName: "Neural2 C",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ES_ES_NEURAL2_D,
        displayName: "Neural2 D",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ES_ES_NEURAL2_E,
        displayName: "Neural2 E",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ES_ES_NEURAL2_F,
        displayName: "Neural2 F",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ES_ES_POLYGLOT_1,
        displayName: "Polyglot 1",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ES_ES_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ES_ES_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ES_ES_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ES_ES_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ES_ES_STUDIO_C,
        displayName: "Studio C",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ES_ES_STUDIO_F,
        displayName: "Studio F",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ES_ES_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ES_ES_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ES_ES_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.ES_ES,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ES_US_NEURAL2_A,
        displayName: "Neural2 A",
        languageCode: types_1.VoiceLanguage.ES_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ES_US_NEURAL2_B,
        displayName: "Neural2 B",
        languageCode: types_1.VoiceLanguage.ES_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ES_US_NEURAL2_C,
        displayName: "Neural2 C",
        languageCode: types_1.VoiceLanguage.ES_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ES_US_NEWS_D,
        displayName: "News D",
        languageCode: types_1.VoiceLanguage.ES_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ES_US_NEWS_E,
        displayName: "News E",
        languageCode: types_1.VoiceLanguage.ES_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ES_US_NEWS_F,
        displayName: "News F",
        languageCode: types_1.VoiceLanguage.ES_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ES_US_NEWS_G,
        displayName: "News G",
        languageCode: types_1.VoiceLanguage.ES_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ES_US_POLYGLOT_1,
        displayName: "Polyglot 1",
        languageCode: types_1.VoiceLanguage.ES_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ES_US_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.ES_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ES_US_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.ES_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ES_US_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.ES_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ES_US_STUDIO_B,
        displayName: "Studio B",
        languageCode: types_1.VoiceLanguage.ES_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ES_US_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.ES_US,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ES_US_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.ES_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ES_US_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.ES_US,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.EU_ES_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.EU_ES,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FI_FI_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.FI_FI,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FI_FI_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.FI_FI,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FIL_PH_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.FIL_PH,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FIL_PH_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.FIL_PH,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FIL_PH_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.FIL_PH,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FIL_PH_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.FIL_PH,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FIL_PH_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.FIL_PH,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FIL_PH_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.FIL_PH,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FIL_PH_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.FIL_PH,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FIL_PH_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.FIL_PH,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FIL_PH_NEURAL2_A,
        displayName: "Neural2 A",
        languageCode: types_1.VoiceLanguage.FIL_PH,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FIL_PH_NEURAL2_D,
        displayName: "Neural2 D",
        languageCode: types_1.VoiceLanguage.FIL_PH,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FR_CA_NEURAL2_A,
        displayName: "Neural2 A",
        languageCode: types_1.VoiceLanguage.FR_CA,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FR_CA_NEURAL2_B,
        displayName: "Neural2 B",
        languageCode: types_1.VoiceLanguage.FR_CA,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FR_CA_NEURAL2_C,
        displayName: "Neural2 C",
        languageCode: types_1.VoiceLanguage.FR_CA,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FR_CA_NEURAL2_D,
        displayName: "Neural2 D",
        languageCode: types_1.VoiceLanguage.FR_CA,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FR_CA_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.FR_CA,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FR_CA_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.FR_CA,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FR_CA_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.FR_CA,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FR_CA_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.FR_CA,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FR_CA_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.FR_CA,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FR_CA_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.FR_CA,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FR_CA_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.FR_CA,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FR_CA_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.FR_CA,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FR_FR_NEURAL2_A,
        displayName: "Neural2 A",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FR_FR_NEURAL2_B,
        displayName: "Neural2 B",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FR_FR_NEURAL2_C,
        displayName: "Neural2 C",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FR_FR_NEURAL2_D,
        displayName: "Neural2 D",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FR_FR_NEURAL2_E,
        displayName: "Neural2 E",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FR_FR_POLYGLOT_1,
        displayName: "Polyglot 1",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FR_FR_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FR_FR_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FR_FR_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FR_FR_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FR_FR_STANDARD_E,
        displayName: "Standard E",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FR_FR_STUDIO_A,
        displayName: "Studio A",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FR_FR_STUDIO_D,
        displayName: "Studio D",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FR_FR_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FR_FR_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FR_FR_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.FR_FR_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.FR_FR_WAVENET_E,
        displayName: "Wavenet E",
        languageCode: types_1.VoiceLanguage.FR_FR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.GL_ES_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.GL_ES,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.GU_IN_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.GU_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.GU_IN_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.GU_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.GU_IN_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.GU_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.GU_IN_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.GU_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.GU_IN_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.GU_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.GU_IN_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.GU_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.GU_IN_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.GU_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.GU_IN_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.GU_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.HE_IL_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.HE_IL,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.HE_IL_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.HE_IL,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.HE_IL_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.HE_IL,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.HE_IL_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.HE_IL,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.HE_IL_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.HE_IL,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.HE_IL_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.HE_IL,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.HE_IL_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.HE_IL,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.HE_IL_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.HE_IL,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.HI_IN_NEURAL2_A,
        displayName: "Neural2 A",
        languageCode: types_1.VoiceLanguage.HI_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.HI_IN_NEURAL2_B,
        displayName: "Neural2 B",
        languageCode: types_1.VoiceLanguage.HI_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.HI_IN_NEURAL2_C,
        displayName: "Neural2 C",
        languageCode: types_1.VoiceLanguage.HI_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.HI_IN_NEURAL2_D,
        displayName: "Neural2 D",
        languageCode: types_1.VoiceLanguage.HI_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.HI_IN_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.HI_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.HI_IN_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.HI_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.HI_IN_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.HI_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.HI_IN_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.HI_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.HI_IN_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.HI_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.HI_IN_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.HI_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.HI_IN_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.HI_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.HI_IN_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.HI_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.HU_HU_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.HU_HU,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.HU_HU_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.HU_HU,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ID_ID_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.ID_ID,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ID_ID_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.ID_ID,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ID_ID_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.ID_ID,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ID_ID_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.ID_ID,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ID_ID_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.ID_ID,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ID_ID_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.ID_ID,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ID_ID_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.ID_ID,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ID_ID_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.ID_ID,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.IS_IS_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.IS_IS,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.IT_IT_NEURAL2_A,
        displayName: "Neural2 A",
        languageCode: types_1.VoiceLanguage.IT_IT,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.IT_IT_NEURAL2_C,
        displayName: "Neural2 C",
        languageCode: types_1.VoiceLanguage.IT_IT,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.IT_IT_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.IT_IT,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.IT_IT_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.IT_IT,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.IT_IT_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.IT_IT,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.IT_IT_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.IT_IT,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.IT_IT_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.IT_IT,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.IT_IT_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.IT_IT,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.IT_IT_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.IT_IT,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.IT_IT_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.IT_IT,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.JA_JP_NEURAL2_B,
        displayName: "Neural2 B",
        languageCode: types_1.VoiceLanguage.JA_JP,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.JA_JP_NEURAL2_C,
        displayName: "Neural2 C",
        languageCode: types_1.VoiceLanguage.JA_JP,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.JA_JP_NEURAL2_D,
        displayName: "Neural2 D",
        languageCode: types_1.VoiceLanguage.JA_JP,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.JA_JP_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.JA_JP,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.JA_JP_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.JA_JP,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.JA_JP_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.JA_JP,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.JA_JP_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.JA_JP,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.JA_JP_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.JA_JP,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.JA_JP_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.JA_JP,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.JA_JP_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.JA_JP,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.JA_JP_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.JA_JP,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.KN_IN_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.KN_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.KN_IN_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.KN_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.KN_IN_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.KN_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.KN_IN_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.KN_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.KN_IN_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.KN_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.KN_IN_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.KN_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.KN_IN_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.KN_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.KN_IN_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.KN_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.KO_KR_NEURAL2_A,
        displayName: "Neural2 A",
        languageCode: types_1.VoiceLanguage.KO_KR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.KO_KR_NEURAL2_B,
        displayName: "Neural2 B",
        languageCode: types_1.VoiceLanguage.KO_KR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.KO_KR_NEURAL2_C,
        displayName: "Neural2 C",
        languageCode: types_1.VoiceLanguage.KO_KR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.KO_KR_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.KO_KR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.KO_KR_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.KO_KR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.KO_KR_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.KO_KR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.KO_KR_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.KO_KR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.KO_KR_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.KO_KR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.KO_KR_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.KO_KR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.KO_KR_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.KO_KR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.KO_KR_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.KO_KR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.LT_LT_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.LT_LT,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.LV_LV_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.LV_LV,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ML_IN_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.ML_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ML_IN_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.ML_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ML_IN_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.ML_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ML_IN_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.ML_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ML_IN_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.ML_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ML_IN_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.ML_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.ML_IN_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.ML_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.ML_IN_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.ML_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.MR_IN_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.MR_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.MR_IN_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.MR_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.MR_IN_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.MR_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.MR_IN_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.MR_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.MR_IN_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.MR_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.MR_IN_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.MR_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.MS_MY_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.MS_MY,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.MS_MY_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.MS_MY,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.MS_MY_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.MS_MY,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.MS_MY_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.MS_MY,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.MS_MY_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.MS_MY,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.MS_MY_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.MS_MY,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.MS_MY_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.MS_MY,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.MS_MY_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.MS_MY,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.NB_NO_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.NB_NO,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.NB_NO_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.NB_NO,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.NB_NO_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.NB_NO,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.NB_NO_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.NB_NO,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.NB_NO_STANDARD_E,
        displayName: "Standard E",
        languageCode: types_1.VoiceLanguage.NB_NO,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.NB_NO_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.NB_NO,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.NB_NO_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.NB_NO,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.NB_NO_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.NB_NO,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.NB_NO_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.NB_NO,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.NB_NO_WAVENET_E,
        displayName: "Wavenet E",
        languageCode: types_1.VoiceLanguage.NB_NO,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.NL_BE_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.NL_BE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.NL_BE_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.NL_BE,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.NL_BE_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.NL_BE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.NL_BE_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.NL_BE,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.NL_NL_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.NL_NL,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.NL_NL_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.NL_NL,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.NL_NL_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.NL_NL,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.NL_NL_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.NL_NL,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.NL_NL_STANDARD_E,
        displayName: "Standard E",
        languageCode: types_1.VoiceLanguage.NL_NL,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.NL_NL_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.NL_NL,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.NL_NL_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.NL_NL,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.NL_NL_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.NL_NL,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.NL_NL_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.NL_NL,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.NL_NL_WAVENET_E,
        displayName: "Wavenet E",
        languageCode: types_1.VoiceLanguage.NL_NL,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PA_IN_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.PA_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PA_IN_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.PA_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.PA_IN_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.PA_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PA_IN_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.PA_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.PA_IN_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.PA_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PA_IN_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.PA_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.PA_IN_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.PA_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PA_IN_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.PA_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.PL_PL_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.PL_PL,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PL_PL_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.PL_PL,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.PL_PL_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.PL_PL,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.PL_PL_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.PL_PL,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PL_PL_STANDARD_E,
        displayName: "Standard E",
        languageCode: types_1.VoiceLanguage.PL_PL,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PL_PL_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.PL_PL,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PL_PL_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.PL_PL,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.PL_PL_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.PL_PL,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.PL_PL_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.PL_PL,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PL_PL_WAVENET_E,
        displayName: "Wavenet E",
        languageCode: types_1.VoiceLanguage.PL_PL,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PT_BR_NEURAL2_A,
        displayName: "Neural2 A",
        languageCode: types_1.VoiceLanguage.PT_BR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PT_BR_NEURAL2_B,
        displayName: "Neural2 B",
        languageCode: types_1.VoiceLanguage.PT_BR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.PT_BR_NEURAL2_C,
        displayName: "Neural2 C",
        languageCode: types_1.VoiceLanguage.PT_BR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PT_BR_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.PT_BR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PT_BR_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.PT_BR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.PT_BR_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.PT_BR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PT_BR_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.PT_BR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PT_BR_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.PT_BR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.PT_BR_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.PT_BR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PT_PT_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.PT_PT,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PT_PT_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.PT_PT,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.PT_PT_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.PT_PT,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.PT_PT_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.PT_PT,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PT_PT_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.PT_PT,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.PT_PT_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.PT_PT,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.PT_PT_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.PT_PT,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.PT_PT_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.PT_PT,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.RO_RO_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.RO_RO,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.RO_RO_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.RO_RO,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.RU_RU_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.RU_RU,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.RU_RU_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.RU_RU,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.RU_RU_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.RU_RU,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.RU_RU_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.RU_RU,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.RU_RU_STANDARD_E,
        displayName: "Standard E",
        languageCode: types_1.VoiceLanguage.RU_RU,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.RU_RU_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.RU_RU,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.RU_RU_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.RU_RU,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.RU_RU_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.RU_RU,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.RU_RU_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.RU_RU,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.RU_RU_WAVENET_E,
        displayName: "Wavenet E",
        languageCode: types_1.VoiceLanguage.RU_RU,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.SK_SK_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.SK_SK,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.SK_SK_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.SK_SK,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.SR_RS_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.SR_RS,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.SV_SE_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.SV_SE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.SV_SE_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.SV_SE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.SV_SE_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.SV_SE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.SV_SE_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.SV_SE,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.SV_SE_STANDARD_E,
        displayName: "Standard E",
        languageCode: types_1.VoiceLanguage.SV_SE,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.SV_SE_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.SV_SE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.SV_SE_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.SV_SE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.SV_SE_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.SV_SE,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.SV_SE_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.SV_SE,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.SV_SE_WAVENET_E,
        displayName: "Wavenet E",
        languageCode: types_1.VoiceLanguage.SV_SE,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.TA_IN_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.TA_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.TA_IN_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.TA_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.TA_IN_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.TA_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.TA_IN_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.TA_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.TA_IN_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.TA_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.TA_IN_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.TA_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.TA_IN_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.TA_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.TA_IN_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.TA_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.TE_IN_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.TE_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.TE_IN_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.TE_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.TH_TH_NEURAL2_C,
        displayName: "Neural2 C",
        languageCode: types_1.VoiceLanguage.TH_TH,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.TH_TH_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.TH_TH,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.TR_TR_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.TR_TR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.TR_TR_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.TR_TR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.TR_TR_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.TR_TR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.TR_TR_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.TR_TR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.TR_TR_STANDARD_E,
        displayName: "Standard E",
        languageCode: types_1.VoiceLanguage.TR_TR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.TR_TR_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.TR_TR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.TR_TR_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.TR_TR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.TR_TR_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.TR_TR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.TR_TR_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.TR_TR,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.TR_TR_WAVENET_E,
        displayName: "Wavenet E",
        languageCode: types_1.VoiceLanguage.TR_TR,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.UK_UA_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.UK_UA,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.UK_UA_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.UK_UA,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.UR_IN_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.UR_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.UR_IN_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.UR_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.UR_IN_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.UR_IN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.UR_IN_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.UR_IN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.VI_VN_NEURAL2_A,
        displayName: "Neural2 A",
        languageCode: types_1.VoiceLanguage.VI_VN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.VI_VN_NEURAL2_D,
        displayName: "Neural2 D",
        languageCode: types_1.VoiceLanguage.VI_VN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.VI_VN_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.VI_VN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.VI_VN_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.VI_VN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.VI_VN_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.VI_VN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.VI_VN_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.VI_VN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.VI_VN_WAVENET_A,
        displayName: "Wavenet A",
        languageCode: types_1.VoiceLanguage.VI_VN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.VI_VN_WAVENET_B,
        displayName: "Wavenet B",
        languageCode: types_1.VoiceLanguage.VI_VN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.VI_VN_WAVENET_C,
        displayName: "Wavenet C",
        languageCode: types_1.VoiceLanguage.VI_VN,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.VI_VN_WAVENET_D,
        displayName: "Wavenet D",
        languageCode: types_1.VoiceLanguage.VI_VN,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.YUE_HK_STANDARD_A,
        displayName: "Standard A",
        languageCode: types_1.VoiceLanguage.YUE_HK,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.YUE_HK_STANDARD_B,
        displayName: "Standard B",
        languageCode: types_1.VoiceLanguage.YUE_HK,
        gender: types_1.VoiceGender.MALE
    },
    {
        name: GoogleVoice.YUE_HK_STANDARD_C,
        displayName: "Standard C",
        languageCode: types_1.VoiceLanguage.YUE_HK,
        gender: types_1.VoiceGender.FEMALE
    },
    {
        name: GoogleVoice.YUE_HK_STANDARD_D,
        displayName: "Standard D",
        languageCode: types_1.VoiceLanguage.YUE_HK,
        gender: types_1.VoiceGender.MALE
    }
];
exports.GoogleVoiceDetails = GoogleVoiceDetails;
