// @fonoster/common/dist/tts/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoiceLanguage = exports.VoiceGender = void 0;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
var VoiceLanguage;
(function (VoiceLanguage) {
    // Base language codes
    VoiceLanguage["AR"] = "ar";
    VoiceLanguage["AF"] = "af";
    VoiceLanguage["AM"] = "am";
    VoiceLanguage["AS"] = "as";
    VoiceLanguage["AZ"] = "az";
    VoiceLanguage["BG"] = "bg";
    VoiceLanguage["BN"] = "bn";
    VoiceLanguage["BS"] = "bs";
    VoiceLanguage["CA"] = "ca";
    VoiceLanguage["CS"] = "cs";
    VoiceLanguage["CY"] = "cy";
    VoiceLanguage["DA"] = "da";
    VoiceLanguage["DE"] = "de";
    VoiceLanguage["EL"] = "el";
    VoiceLanguage["EN"] = "en";
    VoiceLanguage["ES"] = "es";
    VoiceLanguage["ET"] = "et";
    VoiceLanguage["EU"] = "eu";
    VoiceLanguage["FA"] = "fa";
    VoiceLanguage["FI"] = "fi";
    VoiceLanguage["FIL"] = "fil";
    VoiceLanguage["FR"] = "fr";
    VoiceLanguage["GA"] = "ga";
    VoiceLanguage["GL"] = "gl";
    VoiceLanguage["GU"] = "gu";
    VoiceLanguage["HE"] = "he";
    VoiceLanguage["HI"] = "hi";
    VoiceLanguage["HR"] = "hr";
    VoiceLanguage["HU"] = "hu";
    VoiceLanguage["HY"] = "hy";
    VoiceLanguage["ID"] = "id";
    VoiceLanguage["IS"] = "is";
    VoiceLanguage["IT"] = "it";
    VoiceLanguage["JA"] = "ja";
    VoiceLanguage["JV"] = "jv";
    VoiceLanguage["KA"] = "ka";
    VoiceLanguage["KK"] = "kk";
    VoiceLanguage["KM"] = "km";
    VoiceLanguage["KN"] = "kn";
    VoiceLanguage["KO"] = "ko";
    VoiceLanguage["LO"] = "lo";
    VoiceLanguage["LT"] = "lt";
    VoiceLanguage["LV"] = "lv";
    VoiceLanguage["MK"] = "mk";
    VoiceLanguage["ML"] = "ml";
    VoiceLanguage["MN"] = "mn";
    VoiceLanguage["MR"] = "mr";
    VoiceLanguage["MS"] = "ms";
    VoiceLanguage["MT"] = "mt";
    VoiceLanguage["MY"] = "my";
    VoiceLanguage["NB"] = "nb";
    VoiceLanguage["NE"] = "ne";
    VoiceLanguage["NL"] = "nl";
    VoiceLanguage["OR"] = "or";
    VoiceLanguage["PA"] = "pa";
    VoiceLanguage["PL"] = "pl";
    VoiceLanguage["PS"] = "ps";
    VoiceLanguage["PT"] = "pt";
    VoiceLanguage["RO"] = "ro";
    VoiceLanguage["RU"] = "ru";
    VoiceLanguage["SI"] = "si";
    VoiceLanguage["SK"] = "sk";
    VoiceLanguage["SL"] = "sl";
    VoiceLanguage["SO"] = "so";
    VoiceLanguage["SQ"] = "sq";
    VoiceLanguage["SR"] = "sr";
    VoiceLanguage["SU"] = "su";
    VoiceLanguage["SV"] = "sv";
    VoiceLanguage["SW"] = "sw";
    VoiceLanguage["TA"] = "ta";
    VoiceLanguage["TE"] = "te";
    VoiceLanguage["TH"] = "th";
    VoiceLanguage["TR"] = "tr";
    VoiceLanguage["UK"] = "uk";
    VoiceLanguage["UR"] = "ur";
    VoiceLanguage["UZ"] = "uz";
    // Regional variants
    VoiceLanguage["AR_XA"] = "ar-XA";
    VoiceLanguage["AF_ZA"] = "af-ZA";
    VoiceLanguage["AM_ET"] = "am-ET";
    VoiceLanguage["AR_AE"] = "ar-AE";
    VoiceLanguage["AR_BH"] = "ar-BH";
    VoiceLanguage["AR_DZ"] = "ar-DZ";
    VoiceLanguage["AR_EG"] = "ar-EG";
    VoiceLanguage["AR_IQ"] = "ar-IQ";
    VoiceLanguage["AR_JO"] = "ar-JO";
    VoiceLanguage["AR_KW"] = "ar-KW";
    VoiceLanguage["AR_LB"] = "ar-LB";
    VoiceLanguage["AR_LY"] = "ar-LY";
    VoiceLanguage["AR_MA"] = "ar-MA";
    VoiceLanguage["AR_OM"] = "ar-OM";
    VoiceLanguage["AR_QA"] = "ar-QA";
    VoiceLanguage["AR_SA"] = "ar-SA";
    VoiceLanguage["AR_SY"] = "ar-SY";
    VoiceLanguage["AR_TN"] = "ar-TN";
    VoiceLanguage["AR_YE"] = "ar-YE";
    VoiceLanguage["AS_IN"] = "as-IN";
    VoiceLanguage["AZ_AZ"] = "az-AZ";
    VoiceLanguage["BG_BG"] = "bg-BG";
    VoiceLanguage["BN_BD"] = "bn-BD";
    VoiceLanguage["BN_IN"] = "bn-IN";
    VoiceLanguage["BS_BA"] = "bs-BA";
    VoiceLanguage["CA_ES"] = "ca-ES";
    VoiceLanguage["CMN_CN"] = "cmn-CN";
    VoiceLanguage["CMN_TW"] = "cmn-TW";
    VoiceLanguage["CS_CZ"] = "cs-CZ";
    VoiceLanguage["CY_GB"] = "cy-GB";
    VoiceLanguage["DA_DK"] = "da-DK";
    VoiceLanguage["DE_AT"] = "de-AT";
    VoiceLanguage["DE_CH"] = "de-CH";
    VoiceLanguage["DE_DE"] = "de-DE";
    VoiceLanguage["EL_GR"] = "el-GR";
    VoiceLanguage["EN_AU"] = "en-AU";
    VoiceLanguage["EN_CA"] = "en-CA";
    VoiceLanguage["EN_GB"] = "en-GB";
    VoiceLanguage["EN_HK"] = "en-HK";
    VoiceLanguage["EN_IE"] = "en-IE";
    VoiceLanguage["EN_IN"] = "en-IN";
    VoiceLanguage["EN_KE"] = "en-KE";
    VoiceLanguage["EN_NG"] = "en-NG";
    VoiceLanguage["EN_NZ"] = "en-NZ";
    VoiceLanguage["EN_PH"] = "en-PH";
    VoiceLanguage["EN_SG"] = "en-SG";
    VoiceLanguage["EN_TZ"] = "en-TZ";
    VoiceLanguage["EN_US"] = "en-US";
    VoiceLanguage["EN_ZA"] = "en-ZA";
    VoiceLanguage["ES_AR"] = "es-AR";
    VoiceLanguage["ES_BO"] = "es-BO";
    VoiceLanguage["ES_CL"] = "es-CL";
    VoiceLanguage["ES_CO"] = "es-CO";
    VoiceLanguage["ES_CR"] = "es-CR";
    VoiceLanguage["ES_CU"] = "es-CU";
    VoiceLanguage["ES_DO"] = "es-DO";
    VoiceLanguage["ES_EC"] = "es-EC";
    VoiceLanguage["ES_ES"] = "es-ES";
    VoiceLanguage["ES_GQ"] = "es-GQ";
    VoiceLanguage["ES_GT"] = "es-GT";
    VoiceLanguage["ES_HN"] = "es-HN";
    VoiceLanguage["ES_MX"] = "es-MX";
    VoiceLanguage["ES_NI"] = "es-NI";
    VoiceLanguage["ES_PA"] = "es-PA";
    VoiceLanguage["ES_PE"] = "es-PE";
    VoiceLanguage["ES_PR"] = "es-PR";
    VoiceLanguage["ES_PY"] = "es-PY";
    VoiceLanguage["ES_SV"] = "es-SV";
    VoiceLanguage["ES_US"] = "es-US";
    VoiceLanguage["ES_UY"] = "es-UY";
    VoiceLanguage["ES_VE"] = "es-VE";
    VoiceLanguage["ES_419"] = "es-419";
    VoiceLanguage["ET_EE"] = "et-EE";
    VoiceLanguage["EU_ES"] = "eu-ES";
    VoiceLanguage["FA_IR"] = "fa-IR";
    VoiceLanguage["FI_FI"] = "fi-FI";
    VoiceLanguage["FIL_PH"] = "fil-PH";
    VoiceLanguage["FR_BE"] = "fr-BE";
    VoiceLanguage["FR_CA"] = "fr-CA";
    VoiceLanguage["FR_CH"] = "fr-CH";
    VoiceLanguage["FR_FR"] = "fr-FR";
    VoiceLanguage["GA_IE"] = "ga-IE";
    VoiceLanguage["GL_ES"] = "gl-ES";
    VoiceLanguage["GU_IN"] = "gu-IN";
    VoiceLanguage["HE_IL"] = "he-IL";
    VoiceLanguage["HI_IN"] = "hi-IN";
    VoiceLanguage["HR_HR"] = "hr-HR";
    VoiceLanguage["HU_HU"] = "hu-HU";
    VoiceLanguage["HY_AM"] = "hy-AM";
    VoiceLanguage["ID_ID"] = "id-ID";
    VoiceLanguage["IS_IS"] = "is-IS";
    VoiceLanguage["IT_IT"] = "it-IT";
    VoiceLanguage["JA_JP"] = "ja-JP";
    VoiceLanguage["JV_ID"] = "jv-ID";
    VoiceLanguage["KA_GE"] = "ka-GE";
    VoiceLanguage["KK_KZ"] = "kk-KZ";
    VoiceLanguage["KM_KH"] = "km-KH";
    VoiceLanguage["KN_IN"] = "kn-IN";
    VoiceLanguage["KO_KR"] = "ko-KR";
    VoiceLanguage["LO_LA"] = "lo-LA";
    VoiceLanguage["LT_LT"] = "lt-LT";
    VoiceLanguage["LV_LV"] = "lv-LV";
    VoiceLanguage["MK_MK"] = "mk-MK";
    VoiceLanguage["ML_IN"] = "ml-IN";
    VoiceLanguage["MN_MN"] = "mn-MN";
    VoiceLanguage["MR_IN"] = "mr-IN";
    VoiceLanguage["MS_MY"] = "ms-MY";
    VoiceLanguage["MT_MT"] = "mt-MT";
    VoiceLanguage["MY_MM"] = "my-MM";
    VoiceLanguage["NB_NO"] = "nb-NO";
    VoiceLanguage["NE_NP"] = "ne-NP";
    VoiceLanguage["NL_BE"] = "nl-BE";
    VoiceLanguage["NL_NL"] = "nl-NL";
    VoiceLanguage["OR_IN"] = "or-IN";
    VoiceLanguage["PA_IN"] = "pa-IN";
    VoiceLanguage["PL_PL"] = "pl-PL";
    VoiceLanguage["PS_AF"] = "ps-AF";
    VoiceLanguage["PT_BR"] = "pt-BR";
    VoiceLanguage["PT_PT"] = "pt-PT";
    VoiceLanguage["RO_RO"] = "ro-RO";
    VoiceLanguage["RU_RU"] = "ru-RU";
    VoiceLanguage["SI_LK"] = "si-LK";
    VoiceLanguage["SK_SK"] = "sk-SK";
    VoiceLanguage["SL_SI"] = "sl-SI";
    VoiceLanguage["SO_SO"] = "so-SO";
    VoiceLanguage["SQ_AL"] = "sq-AL";
    VoiceLanguage["SR_LATN_RS"] = "sr-Latn-RS";
    VoiceLanguage["SR_RS"] = "sr-RS";
    VoiceLanguage["SU_ID"] = "su-ID";
    VoiceLanguage["SV_SE"] = "sv-SE";
    VoiceLanguage["SW_KE"] = "sw-KE";
    VoiceLanguage["SW_TZ"] = "sw-TZ";
    VoiceLanguage["TA_IN"] = "ta-IN";
    VoiceLanguage["TA_LK"] = "ta-LK";
    VoiceLanguage["TA_MY"] = "ta-MY";
    VoiceLanguage["TA_SG"] = "ta-SG";
    VoiceLanguage["TE_IN"] = "te-IN";
    VoiceLanguage["TH_TH"] = "th-TH";
    VoiceLanguage["TR_TR"] = "tr-TR";
    VoiceLanguage["UK_UA"] = "uk-UA";
    VoiceLanguage["UR_IN"] = "ur-IN";
    VoiceLanguage["UR_PK"] = "ur-PK";
    VoiceLanguage["UZ_UZ"] = "uz-UZ";
    VoiceLanguage["VI_VN"] = "vi-VN";
    VoiceLanguage["WUU_CN"] = "wuu-CN";
    VoiceLanguage["YUE_CN"] = "yue-CN";
    VoiceLanguage["YUE_HK"] = "yue-HK";
    VoiceLanguage["ZH_CN"] = "zh-CN";
    VoiceLanguage["ZH_CN_GUANGXI"] = "zh-CN-guangxi";
    VoiceLanguage["ZH_CN_HENAN"] = "zh-CN-henan";
    VoiceLanguage["ZH_CN_LIAONING"] = "zh-CN-liaoning";
    VoiceLanguage["ZH_CN_SHAANXI"] = "zh-CN-shaanxi";
    VoiceLanguage["ZH_CN_SHANDONG"] = "zh-CN-shandong";
    VoiceLanguage["ZH_CN_SICHUAN"] = "zh-CN-sichuan";
    VoiceLanguage["ZH_HK"] = "zh-HK";
    VoiceLanguage["ZH_TW"] = "zh-TW";
    VoiceLanguage["ZU_ZA"] = "zu-ZA";
})(VoiceLanguage || (exports.VoiceLanguage = VoiceLanguage = {}));
var VoiceGender;
(function (VoiceGender) {
    VoiceGender[VoiceGender["MALE"] = 0] = "MALE";
    VoiceGender[VoiceGender["FEMALE"] = 1] = "FEMALE";
    VoiceGender[VoiceGender["NEUTRAL"] = 2] = "NEUTRAL";
})(VoiceGender || (exports.VoiceGender = VoiceGender = {}));
