// @fonoster/common/dist/identity/createAuthInterceptor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createAuthInterceptor = createAuthInterceptor;
/**
 * Copyright (C) 2025 by Fonoster Inc (https://fonoster.com)
 * http://github.com/fonoster/fonoster
 *
 * This file is part of Fonoster
 *
 * Licensed under the MIT License (the "License");
 * you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *    https://opensource.org/licenses/MIT
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const logger_1 = require("@fonoster/logger");
const decodeToken_1 = require("./decodeToken");
const errors_1 = require("./errors");
const getAccessKeyIdFromCall_1 = require("./getAccessKeyIdFromCall");
const getTokenFromCall_1 = require("./getTokenFromCall");
const hasAccess_1 = require("./hasAccess");
const isValidToken_1 = require("./isValidToken");
const roles_1 = require("./roles");
const tokenHasAccessKeyId_1 = require("./tokenHasAccessKeyId");
const logger = (0, logger_1.getLogger)({ service: "common", filePath: __filename });
/**
 * This function is a gRPC interceptor that checks if the request is valid
 * and if the user has the right permissions to access the resource. When
 * validating the request, the function will check if the request is in the
 * skip list, if the token is valid and if the role is allowed by the RBAC.
 *
 * @param {string} identityPublicKey - The public key to validate the token
 * @param {string[]} publicPath - The list of public paths
 * @return {Function} - The gRPC interceptor
 */
function createAuthInterceptor(identityPublicKey, publicPath) {
    /**
     * Inner function that will be called by the gRPC server.
     *
     * @param {object} methodDefinition - The method definition
     * @param {string} methodDefinition.path - The path of the gRPC method
     * @param {ServerInterceptingCall} call - The call object
     * @return {ServerInterceptingCall} - The modified call object
     */
    return (methodDefinition, call) => {
        const { path } = methodDefinition;
        const accessKeyId = (0, getAccessKeyIdFromCall_1.getAccessKeyIdFromCall)(call);
        logger.verbose("intercepting api call to path", { accessKeyId, path });
        if (publicPath.includes(methodDefinition.path)) {
            logger.verbose("passing auth control to edge function", { path });
            return call;
        }
        const token = (0, getTokenFromCall_1.getTokenFromCall)(call);
        logger.verbose("validating token", { accessKeyId, path });
        if (!(0, isValidToken_1.isValidToken)(token, identityPublicKey)) {
            return (0, errors_1.unauthenticatedError)(call);
        }
        const decodedToken = (0, decodeToken_1.decodeToken)(token);
        logger.verbose("checking access for accessKeyId", {
            accessKeyId,
            path,
            hasAccess: (0, hasAccess_1.hasAccess)(decodedToken, path),
            pathIsWorkspacePath: roles_1.workspaceResourceAccess.includes(path),
            tokenHasAccessKeyId: (0, tokenHasAccessKeyId_1.tokenHasAccessKeyId)(token, accessKeyId)
        });
        if (!(0, hasAccess_1.hasAccess)(decodedToken, path) ||
            (roles_1.workspaceResourceAccess.includes(path) &&
                !(0, tokenHasAccessKeyId_1.tokenHasAccessKeyId)(token, accessKeyId)) ||
            (roles_1.workspaceResourceOwnerOrAdminAccess.includes(path) &&
                !(0, tokenHasAccessKeyId_1.tokenHasAccessKeyId)(token, accessKeyId))) {
            return (0, errors_1.permissionDeniedError)(call);
        }
        return call;
    };
}
